---
title: "Analysis for MUSMUS manuscript"
output: html_notebook
author: "Eveliina Hanski"
date: 20/05/2024 - 22/08/2024
---

# PART 1

**PART 1: GUT MICROBIOTA VARIATION IN LAB VS WILD HOUSE MICE**

***Step 1: Select data***

```{r}

phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds")

table(sample_data(phy)$Project)
skok <- subset_samples(phy, Project=="Skokholm")
unique(sample_data(skok)$Animal_ID)
phy <- subset_samples(phy, Sample_type!="soil") # 1231

# Re-assign age categories as described in Hanski et al 2024 Molecular Ecology (also found in EpiAge_code.Rmd file on my computer)

Skok <- data.frame(sample_data(skok))
Skok$IMR <- rownames(Skok)
unique(Skok$Reprod)
active <-  Skok %>% filter(str_detect(Reprod, "Pregnant|PREG|Pregnant?|PREG?|Nipples|NIPP|Perforate|PERF|Lactating|Testis large|TL|Testis small|TS"))
Skok$act <- FALSE
Skok$act[Skok$trapdate%in%active$trapdate] <- TRUE
table(Skok$Reprod, Skok$act)
Skok$new_age[Skok$Body_mass_g>20] <- "Adult"
Skok$new_age[Skok$Body_mass_g<=15 & Skok$act==FALSE] <- "Juvenile"
Skok$new_age[Skok$Body_mass_g<=15 & Skok$act==TRUE] <- "Sub-adult"
Skok$new_age[Skok$Body_mass_g>15 & Skok$Body_mass_g<=20] <- "Sub-adult"
table(Skok$new_age) # 117 juv, 291 SA, 424 adult
same <- Skok[which(Skok$Age==Skok$new_age),]
Skok$Age <- NA
Skok$Age <- Skok$new_age
Skok <- data.frame(Skok)
rownames(Skok) <- Skok$IMR
Skok <- Skok[,c(1:88)]
sample_data(skok) <- Skok

other <- subset_samples(phy, Project!="Skokholm")

phy <- merge_phyloseq(other, skok) # 1231 samples

table(sample_data(phy)$Source, sample_data(phy)$Project) # "Wild"    "Unclear" "Lab"  - the unclears are Espelette and Cologne mice and these are wild mice, so I will update that.
sample_data(phy)$Source[sample_data(phy)$Source=="Unclear"] <- "Wild"
unique(sample_data(phy)$Project)
unique(sample_data(phy)$Age) # "Adult"     "JUV"       "A"         "Juvenile"  NA          "SA"        ""          "Sub-adult" "adult"     "juvenile" 
sample_data(phy)$Age[is.na(sample_data(phy)$Age)] <- "unknown"
table(sample_data(phy)$Age, sample_data(phy)$Project)
phy # 1231 samples

# Excluding juvenile and sub-adults, so that I will have just adults left, as I don't want to look at age dynamics here.
phy <- subset_samples(phy, Age!="JUV") # 1231
phy <- subset_samples(phy, Age!="Juvenile") # 1107
phy <- subset_samples(phy, Age!="SA") # 1103
phy <- subset_samples(phy, Age!="Sub-adult") # 812
phy <- subset_samples(phy, Age!="juvenile") # 809 not all mice have age assigned, so excluding this way and next looking at body mass and chronological age (where applicable)

table(sample_data(phy)$Age, sample_data(phy)$Project) # only adults and unknown age left. Out of the wild mice, for Faroe, Espelette and Cologne I don't have further age-specific (eg body mass) information that I could use to exclude other than adults. For Midway Atoll, I only had age categories and have already excluded mice not ranked as adults. Wytham and Skok mice have been filtered so that only mice ranked as adults are left. Now I need to exclude lab mice that are not adults, say under 3mo of age.
data <- as.data.frame(sample_data(phy))
data <- data[which(data$Source=="Lab"),] # 195
data$Collection_date <- strptime(as.character(data$Collection_date), "%d/%m/%Y")
data$Collection_date <- format(data$Collection_date, "%Y-%m-%d")
data$Birthdate <- strptime(as.character(data$Birthdate), "%d/%m/%Y")
data$Birthdate <- format(data$Birthdate, "%Y-%m-%d")
# NOTE: For some reason I don't have birthdate for a big chunk of these lab mice. I have the body mass for many of those though, so I can use body mass to exclude non-adults. I can filter out eg mice under 20g of body mass, as that is a good threshold for excluding mice under <12 wk of age https://www.jax.org/jax-mice-and-services/strain-data-sheet-pages/body-weight-chart-000664
lab_bodymass <- data[which(data$Body_mass_g>=20),]
lab_bodymass <- as.list(unique(lab_bodymass$trapdate)) # 108
`%notin%` <- Negate(`%in%`)
data <- data[which(data$trapdate%notin%lab_bodymass),] # 76 
data$age <- difftime(data$Collection_date, data$Birthdate, units = "days")
check <- data[,c("Birthdate", "Collection_date", "age")] # 76
lab_age <- data[which(data$age>=30),] # 38
lab_age <- as.list(unique(lab_age$trapdate)) # 38
lab <- c(lab_bodymass, lab_age) # 146
lab_nothanks <- data[which(data$trapdate%notin%lab),] # 38
lab_nothanks <- as.list(unique(lab_nothanks$trapdate)) # 38
phy <- subset_samples(phy, trapdate%notin%lab_nothanks) # 771

# OK, now to the best of my knowledge I only have adult samples. I will include both females and males and just control for this. Other things I want to homogenise (wild mice): season, reproductive status
sample_data(phy)$month <- str_sub(sample_data(phy)$Collection_date, start = 4, end = 5)
table(sample_data(phy)$month, sample_data(phy)$Project) # no collection date for Espelette, Cologne and Faroe samples, so can't do any seasonal (or reproductive status exclusion for those samples). For lab samples I don't have reproductive status data (but I shouldn't have any preggos etc). For Skok, Wytham, Midway atoll and Isle of May mice I have seasonal (and reprod data). Starting with seasonal exclusion. For all these populations, I have samples from August-November, so I will ony include these months. I could also limit to Sept-Nov but then isle of may sample size would go down from 3 to 2.
samples <- sample_data(phy)
samples$month <- as.numeric(samples$month)
samples <- samples[which(samples$Source!="Lab"),]
samples_x <- samples[which(samples$month>07),] # exclude samples collection between january-july
unique(samples_x$Reprod) # Excluding mice that have signs of pregnancy, nipples, lactation, perforate, testis large, testis large+ recorded in their reprod.
samples_x <- data.frame(samples_x)
write.csv(samples_x, "MUSMUS/MUSMUS_samples_x.csv")

active <- samples_x %>%
  filter(str_detect(Reprod, "Nipples|Pregnant|PREG|PREG?|NIPP|Nipples|Lact|Lactating|PERF|Perforate|TL|Testis large|Testis large+")) # 67
samples_x <- samples_x[which(samples_x$trapdate%notin%active$trapdate),] # 270 wild samples left
table(samples_x$Project) # 3x isle of may, 15x midway atoll, 245x skok, 7x wytham
samples_nothanks <- samples[which(samples$trapdate%notin%samples_x$trapdate),] # 335 - sounds right
unique(samples_nothanks$Project) # I want to exclude Espelette, Cologne and Faroe samples from here since I don't know the collection month of these samples.
samples_nothanks <- samples_nothanks[which(samples_nothanks$Project!="Espelette"),]
samples_nothanks <- samples_nothanks[which(samples_nothanks$Project!="Cologne"),]
samples_nothanks <- samples_nothanks[which(samples_nothanks$Project!="Faroe"),]
unique(samples_nothanks$Project) # done, 339 samples to exclude left
samples_nothanks <- as.list(unique(samples_nothanks$trapdate))

phy <- subset_samples(phy, trapdate%notin%samples_nothanks) 
table(sample_data(phy)$Project)

# So now I have excluded other than adult mice, reproductively active mice as well as samples from other than aug-nov months (wild mice, where sampling month/reprod status/age known).

imr_11102021_P1toP4 <- read.csv("16s_V4V5_11102021_P1toP4.csv")
imr_11102021_P1toP4$MiSeq_batch <- "imr_11102021_P1toP4"
imr_11102021_P5toP6 <- read.csv("16s_V4V5_11102021_P5toP6.csv")
imr_11102021_P5toP6$MiSeq_batch <- "imr_11102021_P5toP6"
imr_21062021_run1 <- read.csv("16s_V4V5_21062021_run1.csv")
imr_21062021_run1$MiSeq_batch <- "imr_21062021_run1"
imr_21062021_run2 <- read.csv("16s_V4V5_21062021_run2.csv")
imr_21062021_run2$MiSeq_batch <- "imr_21062021_run2"
imr_25032022_run1 <- read.csv("16s_V4V5_25032022_run1.csv")
imr_25032022_run1$MiSeq_batch <- "imr_25032022_run1"
miseq <- rbind(imr_11102021_P1toP4, imr_11102021_P5toP6, imr_21062021_run1, imr_21062021_run2, imr_25032022_run1)
miseq <- miseq %>%
  mutate(Sample_ID = sub("_.*", "", Sample_ID))
miseq <- miseq[!duplicated(miseq),] # 1560

sampledata <- data.frame(sample_data(phy))
sampledata$IMR_code <- rownames(sampledata)
sampledata <- merge(sampledata, miseq, by.x="IMR_code", by.y="Sample_ID", all.x=T, all.y=F)
rownames(sampledata) <- sampledata$IMR_code
sample_data(phy) <- data.frame(sampledata)
table(sample_data(phy)$Project, sample_data(phy)$MiSeq_batch)
# Remove Skok samples from imr_11102021_P5toP6 and imr_21062021_run2 since from those batches there are only skok samples (and I have skok samples in other batches too that will be included). Isle of May samples are all from imr_21062021_run1 in which there are also Skok samples but no other lab/wild samples. C57BL/6 samples (king's college?) are all from imr_25032022_run1 and there are no skok samples or any other lab/wild from that batch, but need to keep those samples in.
phy <- subset_samples(phy, MiSeq_batch!="imr_11102021_P5toP6" & MiSeq_batch!="imr_21062021_run2")
table(sample_data(phy)$Project, sample_data(phy)$MiSeq_batch) # samples from 3 MiSeq batches.

saveRDS(phy, "MUSMUS/musmus_MainlandIsland_samples.RDS")

```

***Step 2: Add phylogenetic tree***

```{r}

phy <- readRDS("MUSMUS/musmus_MainlandIsland_samples.RDS") # 46326 taxa

# filter out singletons and doubletons separately for each population/colony
unique(sample_data(phy)$Project) # "Skokholm"     "Isle_of_May"  "Espelette"    "Faroe"        "Midway_Atoll" "Cologne"      "Wytham"       "Lab"          "BMS"          "C57BL_6" 
skok <- subset_samples(phy, Project=="Skokholm")
skok <- filter_taxa(skok, function(x) max(x) > 2, TRUE)
Isle_of_May <- subset_samples(phy, Project=="Isle_of_May")
Isle_of_May <- filter_taxa(Isle_of_May, function(x) max(x) > 2, TRUE)
Espelette <- subset_samples(phy, Project=="Espelette")
Espelette <- filter_taxa(Espelette, function(x) max(x) > 2, TRUE)
Faroe <- subset_samples(phy, Project=="Faroe")
Faroe <- filter_taxa(Faroe, function(x) max(x) > 2, TRUE)
Midway_Atoll <- subset_samples(phy, Project=="Midway_Atoll")
Midway_Atoll <- filter_taxa(Midway_Atoll, function(x) max(x) > 2, TRUE)
Cologne <- subset_samples(phy, Project=="Cologne")
Cologne <- filter_taxa(Cologne, function(x) max(x) > 2, TRUE)
Wytham <- subset_samples(phy, Project=="Wytham")
Wytham <- filter_taxa(Wytham, function(x) max(x) > 2, TRUE)
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_Date <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sep="-")
unique(sample_data(lab)$Project_Date) # "Lab-Kennedy"           "BMS-04/12/2020"        "C57BL_6-BMS_18102021"  "C57BL_6-Kings_college" - removing singletons and doubletons within each of these.
Lab_Kennedy <- subset_samples(lab, Project_Date=="Lab-Kennedy")
Lab_Kennedy <- filter_taxa(Lab_Kennedy, function(x) max(x) > 2, TRUE)
BMS_04 <- subset_samples(lab, Project_Date=="BMS-04/12/2020")
BMS_04 <- filter_taxa(BMS_04, function(x) max(x) > 2, TRUE)
BMS_18 <- subset_samples(lab, Project_Date=="C57BL_6-BMS_18102021")
BMS_18 <- filter_taxa(BMS_18, function(x) max(x) > 2, TRUE)
King <- subset_samples(lab, Project_Date=="C57BL_6-Kings_college")
King <- filter_taxa(King, function(x) max(x) > 2, TRUE)

phy <- merge_phyloseq(skok, Isle_of_May, Espelette, Faroe, Midway_Atoll, Cologne, Wytham, Lab_Kennedy, BMS_04, BMS_18, King) # 7949 taxa, 326 samples

saveRDS(phy, "MUSMUS/labVSwild_noSingDoub.RDS")
sequences <- as.list(colnames(otu_table(phy))) 
saveRDS(sequences, "MUSMUS/MUSMUS_phylotree/labVSwild_sequences.rds")

# ADD A PHYLOGENETIC TREE, BUILT OVER CLUSTER, as in folder MUSMUS_phylotree:

phy <- readRDS("MUSMUS/labVSwild_noSingDoub.RDS")
tree <- readRDS("MUSMUS/MUSMUS_phylotree/MUSMUS_phylotree.rds")
tree_Align <- tree$tree
sampledata <- sample_data(phy)
otu <- otu_table(phy, taxa_are_rows = TRUE)
taxa <- tax_table(phy)
phylo <- phyloseq(otu, taxa, sampledata, tree_Align) # put phyloseq together with tree

# See if tree is rooted
library(ape)
is.rooted(phy_tree(phylo)) # FALSE
phy_tree(phylo) <- root(phy_tree(phylo), sample(taxa_names(phylo), 1), resolve.root = TRUE)
is.rooted(phy_tree(phylo)) # TRUE

# Remove Spike-In taxa:
phy <- phylo
phy <- subset_taxa(phy, Genus!="Imtechella")
phy <- subset_taxa(phy, Genus!="Allobacillus")

saveRDS(phy, "MUSMUS/musmus_labVSwild_final_PS.RDS")

```

***Step 3: Convert data to compositional***

```{r}

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
phy <- microbiome::transform(phy, "compositional")
saveRDS(phy, "MUSMUS/musmus_labVSwild_final_PS.RDS")

```

***Step 4: Investigate influence of storage buffer***

```{r}

# 15 Skokholm samples were stored in four ways: without buffer, in ethanol, in RNAlater or in Shield. All aliquots from a given sample were extracted in the same DNA extraction batch.

phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds")
unique(sample_data(phy)$Buffer) # "Shield"   NA         "S"        "R"        "NB"       "E"        "PBS"      "RNAlater"
phy <- subset_samples(phy, Project=="Skokholm")
id <- data.frame(sample_data(phy))
id <- id[which(id$Buffer=="NB"),] # 15
phy <- subset_samples(phy, trapdate%in%id$trapdate) # 61 samples - should be 60?
table(sample_data(phy)$trapdate) # trapdate 605-19/07/2019 appears five times in the data. Check which one is from a different extraction batch and exclude that
who <- data.frame(sample_data(phy)) # trapdate 605-19/07/2019 has been processed four times in extraction batch 4, and then once in extraction batch 18 with eb_code 18_21 --> remove this last one, it must be a typo.
phy <- subset_samples(phy, EB_code!="18_21") # 60 samples

# Convert to compositional
phy <- microbiome::transform(phy, "compositional")

# Check if samples were sequenced in the same sequencing run
imr_11102021_P1toP4 <- read.csv("16s_V4V5_11102021_P1toP4.csv")
imr_11102021_P1toP4$MiSeq_batch <- "imr_11102021_P1toP4"
imr_11102021_P5toP6 <- read.csv("16s_V4V5_11102021_P5toP6.csv")
imr_11102021_P5toP6$MiSeq_batch <- "imr_11102021_P5toP6"
imr_21062021_run1 <- read.csv("16s_V4V5_21062021_run1.csv")
imr_21062021_run1$MiSeq_batch <- "imr_21062021_run1"
imr_21062021_run2 <- read.csv("16s_V4V5_21062021_run2.csv")
imr_21062021_run2$MiSeq_batch <- "imr_21062021_run2"
imr_25032022_run1 <- read.csv("16s_V4V5_25032022_run1.csv")
imr_25032022_run1$MiSeq_batch <- "imr_25032022_run1"
miseq <- rbind(imr_11102021_P1toP4, imr_11102021_P5toP6, imr_21062021_run1, imr_21062021_run2, imr_25032022_run1)
miseq <- miseq %>%
  mutate(Sample_ID = sub("_.*", "", Sample_ID))
miseq <- miseq[!duplicated(miseq),] # 1560

sampledata <- data.frame(sample_data(phy))
sampledata$IMR_code <- rownames(sampledata)
sampledata <- merge(sampledata, miseq, by.x="IMR_code", by.y="Sample_ID", all.x=T, all.y=F)
rownames(sampledata) <- sampledata$IMR_code
sample_data(phy) <- data.frame(sampledata)
table(sample_data(phy)$Project, sample_data(phy)$MiSeq_batch) # All buffer samples are from same sequencing run: imr_21062021_run1

# How much of variation does sample identity explain? Use PERMANOVA to investigate.

sample_data(phy)$extraction_batch <- str_sub(sample_data(phy)$EB_code, start = 1, end = 2)

# Jaccard

M4 <- data.frame(sample_data(phy))
D <- phyloseq::distance(phy, method = "jaccard", type = "samples", binary=T)

adonis2(D ~ M4$read_count + M4$Buffer + M4$trapdate, na.rm=TRUE,
        by="margin")
# read count R2=0.00572, F=1.1611, p=0.250
# buffer R2=0.02082, F=1.4081  p=0.019
# trapdate R2=0.75664, F=10.9650, p=0.001

beta <- betadisper(D, M4$trapdate)
permutest(beta) # F=2.4052, p=0.014

# Aitchison

M4 <- data.frame(sample_data(phy))
phy_clr <- microbiome::transform(phy, "clr")
D <- phyloseq::distance(phy_clr, method = "euclidean", type = "samples")

adonis2(D ~ M4$read_count + M4$Buffer + M4$trapdate, na.rm=TRUE,
        by="margin")
# read count R2=0.00572, F=1.1611, p=0.242
# buffer R2=0.02082, F=1.4081, p=0.028
# trapdate R2=0.75664, F=10.9650, p=0.001

beta <- betadisper(D, M4$trapdate)
permutest(beta) # F=2.4052, p=0.011

beta <- betadisper(D, M4$Buffer)
permutest(beta) # F=0.1489, p=0.94

# Do samples cluster by sample identity or sample buffer?
# Jaccard
ord <- ordinate(phy, "PCoA", "jaccard", binary=T)
plot <- plot_ordination(phy, ord, color="trapdate") +
                geom_point(size=3, alpha=.7, stroke=0) +
  theme_classic() + 
  labs(title = "Jaccard",
       x = "PCo1 (12.9%)",
       y = "PCo2 (9.1%)",
       color = "") +
  theme(axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        title = element_text(size=20),
        legend.text = element_text(size = 5),
        legend.position = "none")  + geom_polygon(aes(fill=trapdate), alpha=.7)
ggsave("MUSMUS/MUSMUS_figures/Buffer_PCoA_jaccard.png", width=6, height=6)

# Aitchison
phy_clr <- microbiome::transform(phy, "clr")
ord <- ordinate(phy_clr, "PCoA", "euclidean")
plot <- plot_ordination(phy_clr, ord, color="trapdate") +
                geom_point(size=3, alpha=.7, stroke=0) +
  theme_classic() + 
  labs(title = "Aitchison",
       x = "PCo1 (15.2%)",
       y = "PCo2 (10.0%)",
       color = "") +
  theme(axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        title = element_text(size=20),
        legend.text = element_text(size = 5),
        legend.position = "none")  + geom_polygon(aes(fill=trapdate), alpha=.7)
ggsave("MUSMUS/MUSMUS_figures/Buffer_PCoA_aitchison.png", width=6, height=6)

```

***Step5: Add read count info for lab mouse samples***

```{r}

# For some reason hadn't saved read count data for lab mouse samples, adding here
raw <- readRDS("IMR_16s_phyloseq.rds")
sample_data(raw)$read_count_add <- sample_sums(raw)
key <- data.frame(sample_data(raw))
key$IMR_code <- rownames(key)
key <- key[,c("IMR_code", "read_count_add")]

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
df <- data.frame(sample_data(phy))
df <- merge(df, key, by="IMR_code", all=F)
check <- df[,c("Source","read_count", "read_count_add")] # looks good
df$read_count <- ifelse(is.na(df$read_count), df$read_count_add, df$read_count)
df <- df[,c(1:93)]
rownames(df) <- df$IMR_code
sample_data(phy) <- df
saveRDS(phy, "MUSMUS/musmus_labVSwild_final_PS.RDS")

```

***Step6: Investigate read counts across all samples included in the study***

```{r}

mussamples <- read.csv("MUSMUS_sampleIDs.csv", header=F)
phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds")
sample_data(phy)$IMR_code <- rownames(sample_data(phy))

phy <- subset_samples(phy, IMR_code%in%mussamples$V1)
phy # 864 samples

# For some reason hadn't saved read count data for lab mouse samples, adding here
raw <- readRDS("IMR_16s_phyloseq.rds")
sample_data(raw)$read_count_add <- sample_sums(raw)
key <- data.frame(sample_data(raw))
key$IMR_code <- rownames(key)
key <- key[,c("IMR_code", "read_count_add")]

df <- data.frame(sample_data(phy))
df <- merge(df, key, by="IMR_code", all=F)
check <- df[,c("Source","read_count", "read_count_add")] # looks good
df$read_count <- ifelse(is.na(df$read_count), df$read_count_add, df$read_count)
metadata <- df

# Compute mean and standard deviation for Wild samples
wild_reads <- metadata$read_count[metadata$Source == "Wild"]
wild_mean <- mean(wild_reads, na.rm = TRUE)
wild_range <- range(wild_reads, na.rm = TRUE)

# Compute mean and standard deviation for Lab samples
lab_reads <- metadata$read_count[metadata$Source == "Lab"]
lab_mean <- mean(lab_reads, na.rm = TRUE)
lab_range <- range(lab_reads, na.rm = TRUE)

# Print results
cat(sprintf("Wild mice - Mean read count: %.2f, range: %.2f\n", wild_mean, wild_range))
cat(sprintf("Lab mice - Mean read count: %.2f, range: %.2f\n", lab_mean, lab_range)) 

```

***Q1: Do wild mice have a higher alpha diversity than lab mice?***

```{r}

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")

### Do wild and lab mice have statistically significantly different alpha diversity? Using Wilcoxon tests to test this.
alphadiv <- data.frame(sample_data(phy))

## SHANNON
u.obs <- wilcox.test(Shannon_asy ~ Source, data=alphadiv)
u.obs # W = 5619, p-value < 2.2e-16

## ESTIMATED ASV RICHNESS
u.obs <- wilcox.test(Richness_asy ~ Source, data=alphadiv)
u.obs # W = 3714, p-value < 2.2e-16

# Plot:
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)

unique(sample_data(phy)$Project_ID)
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="Skokholm"] <- "7"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="Isle_of_May"] <- "6"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="Espelette"] <- "3"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="Faroe"] <- "5"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="Midway_Atoll"] <- "4"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="Cologne"] <- "1"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="Wytham"] <- "2"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="Lab-Kennedy-C57BL/6"] <- "12"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="Lab-Kennedy-SKG"] <- "13"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg"] <- "8"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="BMS-04/12/2020-PdgfraCreERT2"] <- "9"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="C57BL_6-BMS_18102021-C57BL/6"] <- "10"
sample_data(phy)$Project_nr[sample_data(phy)$Project_ID=="C57BL_6-Kings_college-C57BL/6"] <- "11"
sample_data(phy)$Project_nr <- as.numeric(as.character(sample_data(phy)$Project_nr))
meta <- data.frame(sample_data(phy))

meta2 <- meta
meta2$Project_ID[meta2$Source=="Wild"] <- "Wild"
meta2$Project_nr[meta2$Source=="Wild"] <- 7.5
meta2$Project_ID[meta2$Source=="Lab"] <- "Lab"
meta2$Project_nr[meta2$Source=="Lab"] <- 14
meta <- rbind(meta, meta2)

meta$Color <- ifelse(meta$Project_ID == "Lab", "#C38D94", 
                     ifelse(meta$Project_ID == "Wild", "#096B72", "black"))

meta <- meta %>%
  mutate(Color = case_when(
    Project_ID == "Lab" ~ "#C38D94",
    Project_ID == "Wild" ~ "#096B72",
    TRUE ~ "black"
  ),
  Label = case_when(
    Project_nr == 7.5 ~ "W",
    Project_nr == 14 ~ "L",
    TRUE ~ as.character(Project_nr)
  ))

plot <- ggplot(data = meta, aes(x = reorder(Label, Project_nr), y = Richness_asy, fill = Project_ID, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb", "BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg" = "#ffb1bc", "BMS-04/12/2020-PdgfraCreERT2" = "#ff8cbf", "C57BL_6-BMS_18102021-C57BL/6" = "#ff4e71", "C57BL_6-Kings_college-C57BL/6" = "#bb2545", "Lab-Kennedy-C57BL/6" = "#78061f", "Lab-Kennedy-SKG" = "#ab7777", "Lab" = "white", "Wild" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("ASV richness") +
  rremove("legend.title") +
    theme(axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 35),
        legend.position = "none",
        axis.text.x = element_text(face=c("plain", "plain", "plain", "plain", "plain", "plain", "plain", "bold",
                                          "plain", "plain", "plain", "plain", "plain", "plain", "bold"), size=20), 
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() + scale_y_continuous(n.breaks=6) 
p <- plot + geom_signif(comparisons = list(c("W", "L")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_ASVrichness.png", width=6.5, height=6)

plot <- ggplot(data = meta, aes(x = reorder(Label, Project_nr), y = Shannon_asy, fill = Project_ID, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb", "BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg" = "#ffb1bc", "BMS-04/12/2020-PdgfraCreERT2" = "#ff8cbf", "C57BL_6-BMS_18102021-C57BL/6" = "#ff4e71", "C57BL_6-Kings_college-C57BL/6" = "#bb2545", "Lab-Kennedy-C57BL/6" = "#78061f", "Lab-Kennedy-SKG" = "#ab7777", "Lab" = "white", "Wild" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("Shannon diversity") +
  rremove("legend.title") +
    theme(axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 35),
        legend.position = "none",
        axis.text.x = element_text(face=c("plain", "plain", "plain", "plain", "plain", "plain", "plain", "bold",
                                          "plain", "plain", "plain", "plain", "plain", "plain", "bold"), size=20), 
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() + scale_y_continuous(n.breaks=6) 
p <- plot + geom_signif(comparisons = list(c("W", "L")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_Shannon.png", width=6.5, height=6)

```

***Q2: How many shared and unique ASVs do lab and wild mice in this data have, and what proportion of wild and lab ASVs are detected in >90% wild/lab samples?***

```{r}

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")

# First include all samples
wild <- subset_samples(phy, Source=="Wild")
wild  <- prune_taxa(taxa_sums(wild) > 0, wild)
wild_list <- taxa_names(wild)
lab <- subset_samples(phy, Source=="Lab")
lab  <- prune_taxa(taxa_sums(lab) > 0, lab)
lab_list <- taxa_names(lab)
venn_list <- list("Wild (n=436)"=wild_list, "Lab (n=146)"=lab_list) 
library(eulerr)
euler <- euler(venn_list)
plot <- plot(euler,
             fills=c("#69A56B", "#69A2B0", "cadetblue4"),
             alpha=.6,
             key=TRUE,
             quantities=TRUE,
             edges=FALSE,
             fontsize=60,
             size=60,
             main="Unique and shared ASVs") # 6006 unique ASVs in wild, 1116 unique ASVs in lab. 812 shared ASVs.
ggsave("MUSMUS/MUSMUS_figures/LabWild_allASVs.png", plot, width=6, height=6)

# Then include equal sample size. Take 100 wild and 100 lab mouse samples
set.seed(8765)
w <- data.frame(sample_data(subset_samples(phy, Source=="Wild")))
w <- sample_n(w, 100)
l <- data.frame(sample_data(subset_samples(phy, Source=="Lab")))
l <- sample_n(l, 100)
w_l <- rbind(w, l)
phy_subset <- subset_samples(phy, EB_code%in%w_l$EB_code)
table(sample_data(phy_subset)$Source) #

wild <- subset_samples(phy_subset, Source=="Wild")
wild  <- prune_taxa(taxa_sums(wild) > 0, wild)
wild_list <- taxa_names(wild)
lab <- subset_samples(phy_subset, Source=="Lab")
lab  <- prune_taxa(taxa_sums(lab) > 0, lab)
lab_list <- taxa_names(lab)
venn_list <- list("Wild (n=100)"=wild_list, "Lab (n=100)"=lab_list) 
library(eulerr)
euler <- euler(venn_list)
plot <- plot(euler,
             fills=c("#69A56B", "#69A2B0", "cadetblue4"),
             alpha=.6,
             key=TRUE,
             quantities=TRUE,
             edges=FALSE,
             fontsize=60,
             size=60,
             main="Unique and shared ASVs") # In this random cross-sec subset of the data (100 wild, 100 lab): 4194 unique ASVs in wild, 839 unique ASVs in lab. 702 shared ASVs.
ggsave("MUSMUS/MUSMUS_figures/LabWild_subsetSamplesASVs.png", plot, width=6, height=6)

# How many unique ASVs across all wild samples in this dataset?

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")

sample_data(phy)$Sample_ID <- rownames(sample_data(phy))
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)

# Subset the phyloseq object by Source
wild_phy <- subset_samples(phy, Source == "Wild")
lab_phy <- subset_samples(phy, Source == "Lab")

# Prune taxa to include only those present in each subset
wild_phy <- prune_taxa(taxa_sums(wild_phy) > 0, wild_phy) # 6818 taxa, 180 samples
lab_phy <- prune_taxa(taxa_sums(lab_phy) > 0, lab_phy) # 1928 taxa, 146 samples

# How many wild asvs present in >90% of wild samples?
wild_check <- wild_phy
otu_table(wild_check) <- t(otu_table(wild_check))
otu_table(wild_check)[otu_table(wild_check)>0] <- 1
wild_check <- filter_taxa(wild_check, function(x) sum(x > 0) > (0.9*length(x)), TRUE) # 1 taxon
head(tax_table(wild_check)) # It is Ligilactobacillus from family Lactobacillaceae

# How many wild asvs present in >90% of lab samples?
lab_check <- lab_phy
otu_table(lab_check) <- t(otu_table(lab_check))
otu_table(lab_check)[otu_table(lab_check)>0] <- 1
lab_check <- filter_taxa(lab_check, function(x) sum(x > 0) > (0.9*length(x)), TRUE) # 16 taxon
head(tax_table(lab_check))
# 10x family Muribaculaceae (1x genus Muribaculum, others are unknown genus)
# 1x Oscillospiraceae, Colidextribacter
# 1x Lactobacillaceae, Ligilactobacillus
# 1x Desulfovibrionaceae, Desulfovibrio
# 1x Bacteroidaceae, Bacteroides
# 1x Akkermansiaceae, Akkermansia

# Is the Ligilactobacillus ASV same in lab and wild?
taxa <- data.frame(tax_table(wild_check))
taxa$ASV <- rownames(taxa)
save <- taxa
taxa <- data.frame(tax_table(lab_check))
taxa$ASV <- rownames(taxa)
taxa <- taxa[which(taxa$ASV%in%save$ASV),] # Yep, same ASV. 

```

***Q3: Do wild mice have a higher pairwise microbial dissimilarity than lab mice?***

```{r}

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")

sample_data(phy)$Sample_ID <- rownames(sample_data(phy))
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)

# Aitchison
phy_clr <- microbiome::transform(phy, "clr")
D <- phyloseq::distance(phy_clr, method = "euclidean", type = "samples")
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(phy))
sd$Mouse <- rownames(sd)
sd$Sample_ID <- rownames(sd)
sd$Animal_ID <- ifelse(is.na(sd$Animal_ID), sd$Sample_ID, sd$Animal_ID)
sd <- data.frame(sd[,c("Source", "Mouse", "Project_ID", "Sample_ID")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID", "mouse2_source", "mouse2_project", "mouse2_ID")
distance <- na.omit(distance)
distance <- distance[distance$mouse1!=distance$mouse2, ] 
distance <- distance[distance$mouse1_ID!=distance$mouse2_ID, ]
distance$Source <- paste(distance$mouse1_source, distance$mouse2_source, sep="-")
distance <- distance[which(distance$Source!="Wild-Lab"),]
distance <- distance[which(distance$Source!="Lab-Wild"),]
distance$Project <- paste(distance$mouse1_project, distance$mouse2_project, sep="-")
same <- distance[which(distance$mouse1_project==distance$mouse2_project),]

# Wilcoxon test to check if differences between bins (within source) are significant
#Get observed u-statistic
u.obs<-wilcox.test(BC_distance ~ mouse2_source, data=same)
u.obs # W = 76817832, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"BC_distance"
for (i in 1:Nperm) {
  d<-same[,!names(same) %in% lose]
  d$dist.rand<-sample(same$BC_distance, length(same$BC_distance), replace=F)
  x<-wilcox.test(dist.rand ~ mouse2_source, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat<u.obs$statistic, 1, 0)
perm1$result 
p<-mean(perm1$result)
p # 0

mean_A<-mean(same$BC_distance[same$mouse2_source=="Wild"]) # 97.52436
mean_B<-mean(same$BC_distance[same$mouse2_source=="Lab"]) # 55.41994
diff<-mean_A-mean_B # 42.10442
pval<-2*p
pval # final permutational Wilcoxon rank sum test p-value = 0

# Plotting
same$Project_nr <- NA
same2 <- same
same2$mouse2_project[same2$mouse2_source=="Wild"] <- "Wild"
same2$Project_nr[same2$mouse2_source=="Wild"] <- 7.5
same2$mouse2_project[same2$mouse2_source=="Lab"] <- "Lab"
same2$Project_nr[same2$mouse2_source=="Lab"] <- 14
same <- rbind(same, same2)

same$Project_nr[same$mouse2_project=="Skokholm"] <- "7"
same$Project_nr[same$mouse2_project=="Isle_of_May"] <- "6"
same$Project_nr[same$mouse2_project=="Espelette"] <- "3"
same$Project_nr[same$mouse2_project=="Faroe"] <- "5"
same$Project_nr[same$mouse2_project=="Midway_Atoll"] <- "4"
same$Project_nr[same$mouse2_project=="Cologne"] <- "1"
same$Project_nr[same$mouse2_project=="Wytham"] <- "2"
same$Project_nr[same$mouse2_project=="Lab-Kennedy-C57BL/6"] <- "12"
same$Project_nr[same$mouse2_project=="Lab-Kennedy-SKG"] <- "13"
same$Project_nr[same$mouse2_project=="BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg"] <- "8"
same$Project_nr[same$mouse2_project=="BMS-04/12/2020-PdgfraCreERT2"] <- "9"
same$Project_nr[same$mouse2_project=="C57BL_6-BMS_18102021-C57BL/6"] <- "10"
same$Project_nr[same$mouse2_project=="C57BL_6-Kings_college-C57BL/6"] <- "11"
same$Project_nr <- as.numeric(as.character(same$Project_nr))

same$Color <- ifelse(same$mouse2_project == "Lab", "#C38D94", 
                     ifelse(same$mouse2_project == "Wild", "#096B72", "black"))

same <- same %>%
  mutate(Color = case_when(
    mouse2_project == "Lab" ~ "#C38D94",
    mouse2_project == "Wild" ~ "#096B72",
    TRUE ~ "black"
  ),
  Label = case_when(
    Project_nr == 7.5 ~ "W",
    Project_nr == 14 ~ "L",
    TRUE ~ as.character(Project_nr)
  ))

plot <- ggplot(data = same, aes(x = reorder(Label, Project_nr), y = BC_distance, fill = mouse2_project, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb", "BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg" = "#ffb1bc", "BMS-04/12/2020-PdgfraCreERT2" = "#ff8cbf", "C57BL_6-BMS_18102021-C57BL/6" = "#ff4e71", "C57BL_6-Kings_college-C57BL/6" = "#bb2545", "Lab-Kennedy-C57BL/6" = "#78061f", "Lab-Kennedy-SKG" = "#ab7777", "Lab" = "white", "Wild" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("Aitchison dissimilarity") +
  rremove("legend.title") +
    theme(axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 35),
        legend.position = "none",
        axis.text.x = element_text(face=c("plain", "plain", "plain", "plain", "plain", "plain", "plain", "bold",
                                          "plain", "plain", "plain", "plain", "plain", "plain", "bold"), size=20), 
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() + scale_y_continuous(n.breaks=6) 
p <- plot + geom_signif(comparisons = list(c("W", "L")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_Aitchison.png", width=6.5, height=6)

# Jaccard
D <- phyloseq::distance(phy, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(phy))
sd$Mouse <- rownames(sd)
sd$Sample_ID <- rownames(sd)
sd$Animal_ID <- ifelse(is.na(sd$Animal_ID), sd$Sample_ID, sd$Animal_ID)
sd <- data.frame(sd[,c("Source", "Mouse", "Project_ID", "Sample_ID")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID", "mouse2_source", "mouse2_project", "mouse2_ID")
distance <- na.omit(distance)
distance <- distance[distance$mouse1!=distance$mouse2, ] 
distance <- distance[distance$mouse1_ID!=distance$mouse2_ID, ]
distance$Source <- paste(distance$mouse1_source, distance$mouse2_source, sep="-")
distance <- distance[which(distance$Source!="Wild-Lab"),]
distance <- distance[which(distance$Source!="Lab-Wild"),]
distance$Project <- paste(distance$mouse1_project, distance$mouse2_project, sep="-")
same <- distance[which(distance$mouse1_project==distance$mouse2_project),]

# Average Jaccard for a wild-wild vs lab-lab pair (only same population/colony/strain pairs considered)
same$mouse1_source==same$mouse2_source # TRUE
same$mouse1_project==same$mouse2_project # TRUE
w <- same[which(same$mouse1_source=="Wild"),]
mean(w$BC_distance) # 0.8224184 --> around 17.8% of ASVs shared
sd(w$BC_distance) # 0.06526965
l <- same[which(same$mouse1_source=="Lab"),]
mean(l$BC_distance) # 0.6084406 --> around 39.2% of ASVs shared
sd(l$BC_distance) # 0.1154309

# Wilcoxon test to check if differences between bins (within source) are significant
#Get observed u-statistic
u.obs<-wilcox.test(BC_distance ~ mouse2_source, data=same)
u.obs # W = 5063894, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"BC_distance"
for (i in 1:Nperm) {
  d<-same[,!names(same) %in% lose]
  d$dist.rand<-sample(same$BC_distance, length(same$BC_distance), replace=F)
  x<-wilcox.test(dist.rand ~ mouse2_source, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat<u.obs$statistic, 1, 0)
perm1$result 
p<-mean(perm1$result)
p # 0

mean_A<-mean(same$BC_distance[same$mouse2_source=="Wild"]) # 0.8224184
mean_B<-mean(same$BC_distance[same$mouse2_source=="Lab"]) # 0.6084406
diff<-mean_A-mean_B # 0.2139778
pval<-2*p
pval # final permutational Wilcoxon rank sum test p-value = 0

# Plotting
same$Project_nr <- NA
same2 <- same
same2$mouse2_project[same2$mouse2_source=="Wild"] <- "Wild"
same2$Project_nr[same2$mouse2_source=="Wild"] <- 7.5
same2$mouse2_project[same2$mouse2_source=="Lab"] <- "Lab"
same2$Project_nr[same2$mouse2_source=="Lab"] <- 14
same <- rbind(same, same2)

same$Project_nr[same$mouse2_project=="Skokholm"] <- "7"
same$Project_nr[same$mouse2_project=="Isle_of_May"] <- "6"
same$Project_nr[same$mouse2_project=="Espelette"] <- "3"
same$Project_nr[same$mouse2_project=="Faroe"] <- "5"
same$Project_nr[same$mouse2_project=="Midway_Atoll"] <- "4"
same$Project_nr[same$mouse2_project=="Cologne"] <- "1"
same$Project_nr[same$mouse2_project=="Wytham"] <- "2"
same$Project_nr[same$mouse2_project=="Lab-Kennedy-C57BL/6"] <- "12"
same$Project_nr[same$mouse2_project=="Lab-Kennedy-SKG"] <- "13"
same$Project_nr[same$mouse2_project=="BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg"] <- "8"
same$Project_nr[same$mouse2_project=="BMS-04/12/2020-PdgfraCreERT2"] <- "9"
same$Project_nr[same$mouse2_project=="C57BL_6-BMS_18102021-C57BL/6"] <- "10"
same$Project_nr[same$mouse2_project=="C57BL_6-Kings_college-C57BL/6"] <- "11"
same$Project_nr <- as.numeric(as.character(same$Project_nr))

same$Color <- ifelse(same$mouse2_project == "Lab", "#C38D94", 
                     ifelse(same$mouse2_project == "Wild", "#096B72", "black"))

same <- same %>%
  mutate(Color = case_when(
    mouse2_project == "Lab" ~ "#C38D94",
    mouse2_project == "Wild" ~ "#096B72",
    TRUE ~ "black"
  ),
  Label = case_when(
    Project_nr == 7.5 ~ "W",
    Project_nr == 14 ~ "L",
    TRUE ~ as.character(Project_nr)
  ))

plot <- ggplot(data = same, aes(x = reorder(Label, Project_nr), y = BC_distance, fill = mouse2_project, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb", "BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg" = "#ffb1bc", "BMS-04/12/2020-PdgfraCreERT2" = "#ff8cbf", "C57BL_6-BMS_18102021-C57BL/6" = "#ff4e71", "C57BL_6-Kings_college-C57BL/6" = "#bb2545", "Lab-Kennedy-C57BL/6" = "#78061f", "Lab-Kennedy-SKG" = "#ab7777", "Lab" = "white", "Wild" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("Jaccard distance") +
  rremove("legend.title") +
    theme(axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 35),
        legend.position = "none",
        axis.text.x = element_text(face=c("plain", "plain", "plain", "plain", "plain", "plain", "plain", "bold",
                                          "plain", "plain", "plain", "plain", "plain", "plain", "bold"), size=20), 
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() + scale_y_continuous(n.breaks=6) 
p <- plot + geom_signif(comparisons = list(c("W", "L")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_Jaccard.png", width=6.5, height=6)

# Also checking what proportion of ASVs are shared between lab mouse pairs from the same facility (regardless of which strain)

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
sample_data(phy)$Sample_ID <- rownames(sample_data(phy))
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- lab
unique(sample_data(phy)$Project_ID) # change colony to facility
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Lab-Kennedy-C57BL/6"] <- "Facility C"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Lab-Kennedy-SKG"] <- "Facility C"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg"] <- "Facility A"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="BMS-04/12/2020-PdgfraCreERT2"] <- "Facility A"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="C57BL_6-BMS_18102021-C57BL/6"] <- "Facility C"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="C57BL_6-Kings_college-C57BL/6"] <- "Facility B"

# Jaccard
D <- phyloseq::distance(phy, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(phy))
sd$Mouse <- rownames(sd)
sd$Sample_ID <- rownames(sd)
sd$Animal_ID <- ifelse(is.na(sd$Animal_ID), sd$Sample_ID, sd$Animal_ID)
sd <- data.frame(sd[,c("Source", "Mouse", "Project_ID", "Sample_ID")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID", "mouse2_source", "mouse2_project", "mouse2_ID")
distance <- na.omit(distance)
distance <- distance[distance$mouse1!=distance$mouse2, ] 
distance <- distance[distance$mouse1_ID!=distance$mouse2_ID, ]
distance$Source <- paste(distance$mouse1_source, distance$mouse2_source, sep="-")
distance <- distance[which(distance$Source!="Wild-Lab"),]
distance <- distance[which(distance$Source!="Lab-Wild"),]
distance$Project <- paste(distance$mouse1_project, distance$mouse2_project, sep="-")
same <- distance[which(distance$mouse1_project==distance$mouse2_project),]

# Average Jaccard for a lab-lab pair (only same facility pairs considered)
same$mouse1_source==same$mouse2_source # TRUE
same$mouse1_project==same$mouse2_project # TRUE
w <- same[which(same$mouse1_source=="Wild"),]
l <- same[which(same$mouse1_source=="Lab"),]
mean(l$BC_distance) # 0.6140092 --> around 38.6% of ASVs shared
sd(l$BC_distance) # 0.1188582

```

***Q4: Do wild mice have more unique functional pathways than lab mice?***

```{r prep_files_for_PICRUSt2}

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
sequences <- as.list(colnames(otu_table(phy))) 
lw_sequences <- data.frame(sequences)
first_row <- lw_sequences[1, ]
lw_sequences <- as.list(lw_sequences)
library(seqinr)
write.fasta(sequences = lw_sequences, names = lw_sequences, file.out = "MUSMUS/MUSMUS_PICRUSt2/LabWild_seqs.fna") # FASTA header is same as sequence, i.e. using sequences as otu id's.

# Make a biom formatted otu table
phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
otu_table(phy) <- t(otu_table(phy))
library(biomformat)
b2 <- 
    make_biom(
        data = otu_table(phy)
    )
write_biom(b2, 'MUSMUS/MUSMUS_PICRUSt2/lw_relOtu.biom')

```

# Run on terminal:

# Activate
$ conda activate picrust2

# Run PICrust
$ picrust2_pipeline.py -s MUSMUS/MUSMUS_PICRUSt2/LabWild_seqs.fna -i MUSMUS/MUSMUS_PICRUSt2/lw_relOtu.biom -o MUSMUS/MUSMUS_PICRUSt2/picrust2_out_pipeline -p 1

```{r plot_unique_pathways}

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)

sd <- sample_data(phy)
key.x <- sd[,c("Project_ID", "Sample_ID")]
key.x$Sample_ID <- rownames(key.x)

pathways <- read.table(file = "MUSMUS/MUSMUS_PICRUSt2/picrust2_out_pipeline/pathways_out/path_abun_unstrat.tsv.gz", sep = '\t', header = TRUE)
rownames(pathways) <- pathways[,c(1)]
pathways <- pathways[,c(2:327)]
library(funrar)
pathways <- t(pathways)
pathways_rel <- make_relative(pathways)

# Heatmapping
phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
sd <- as.data.frame(sample_data(phy))
meta <- sd[,c("Source", "Project")]
meta$Sample_ID <- rownames(meta)
meta <- meta[,c(3,1,2)]
key <- data.frame(meta[,c("Sample_ID", "Source", "Project")])

lab <- meta[which(meta$Source=="Lab"),]
wild <- meta[which(meta$Source=="Wild"),]
mat <- data.frame(pathways_rel)
mat$Sample_ID <- rownames(mat)

mat_lab <- mat[which(mat$Sample_ID%in%lab$Sample_ID),]
mat_lab <- merge(mat_lab, key, by="Sample_ID", all.x=TRUE, all.y=FALSE)
mat_lab <- mat_lab[,c(2:416)]
names(mat_lab) <- substring(names(mat_lab), 1,100)
mat_lab <- mat_lab %>% select_if(colSums(.) != 0) # Remove pathways not found in lab mice at any level. 312 pathways left
mat_lab <- as.matrix(mat_lab)
heatmap(mat_lab, Rowv=NA, labCol = FALSE)

mat_wild <- mat[which(mat$Sample_ID%in%wild$Sample_ID),]
mat_wild <- merge(mat_wild, key, by="Sample_ID", all.x=TRUE, all.y=FALSE)
mat_wild <- mat_wild[,c(2:416)]
names(mat_wild) <- substring(names(mat_wild), 1,100)
mat_wild <- mat_wild %>% select_if(colSums(.) != 0) # Remove pathways not found in wild mice at any level. 415 pathways left
mat_wild <- as.matrix(mat_wild)
heatmap(mat_wild, Rowv=NA, labCol = FALSE)

# Making a heatmap with both, sorted by source

mat_both <- merge(mat, key, by="Sample_ID", all.x=TRUE, all.y=FALSE)
mat_both$sourceID <- paste(mat_both$Source, mat_both$Project, mat_both$Sample_ID, sep="      ")
rownames(mat_both) <- mat_both$sourceID
mat_both <- mat_both[order(mat_both$Source, mat_both$Project),] 
mat_both <- mat_both[,c(2:416)]
mat_both <- as.matrix(mat_both)
mat_both <- data.frame(t(mat_both))
mat_both$mean <- data.frame(rowMeans(mat_both)) # 415 pathways
mat_both1 <- mat_both[which(mat_both$mean>0.01),] # 1 pathway
mat_both <- mat_both[which(mat_both$mean>0.005),] # 91 pathways
mat_both <- mat_both[,c(1:327)]
mat_both <- as.matrix(t(mat_both))
heatmap(mat_both, Rowv=NA,labCol = TRUE, cexRow=.5)

# Number of functional pathways in wild vs lab mice
wl <- merge(mat, key, by="Sample_ID", all.x = TRUE, all.y=FALSE)
wl <- wl[,c(1,417,418,2:416)]
wl_qual <- wl %>% mutate_if(is.numeric, ~1 * (. > 0)) # converted to qualitative (presence/absence)
wl_qual$total_pathways <- rowSums(wl_qual[,4:418])
summary(wl_qual$total_pathways) # 287 pathways on average, ranges 241-393

id_paths <- wl_qual[,c("Sample_ID", "total_pathways")]
sd <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
sd <- data.frame(sample_data(sd))
sd$Sample_ID <- rownames(sd)
sd <- sd[,c("Sample_ID", "Shannon_asy", "Richness_asy")]
sd <- merge(sd, id_paths, by="Sample_ID", all=FALSE)
summary(lm(total_pathways ~ Shannon_asy, data=sd))

key.x <- data.frame(key.x)
merged <- merge(wl_qual, key.x, by="Sample_ID", all.x=TRUE, all.y=FALSE) # this key comes from R chunk with code for lab vs wild gut microbiome in general

## Wilcoxon test to see if number of unique pathways is significantly different between lab and wild
u.obs <- wilcox.test(total_pathways ~ Source, data=merged)
u.obs # W = 1558, p-value < 2.2e-16

merged$Project_nr[merged$Project_ID=="Skokholm"] <- "7"
merged$Project_nr[merged$Project_ID=="Isle_of_May"] <- "6"
merged$Project_nr[merged$Project_ID=="Espelette"] <- "3"
merged$Project_nr[merged$Project_ID=="Faroe"] <- "5"
merged$Project_nr[merged$Project_ID=="Midway_Atoll"] <- "4"
merged$Project_nr[merged$Project_ID=="Cologne"] <- "1"
merged$Project_nr[merged$Project_ID=="Wytham"] <- "2"
merged$Project_nr[merged$Project_ID=="Lab-Kennedy-C57BL/6"] <- "12"
merged$Project_nr[merged$Project_ID=="Lab-Kennedy-SKG"] <- "13"
merged$Project_nr[merged$Project_ID=="BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg"] <- "8"
merged$Project_nr[merged$Project_ID=="BMS-04/12/2020-PdgfraCreERT2"] <- "9"
merged$Project_nr[merged$Project_ID=="C57BL_6-BMS_18102021-C57BL/6"] <- "10"
merged$Project_nr[merged$Project_ID=="C57BL_6-Kings_college-C57BL/6"] <- "11"
merged$Project_nr <- as.numeric(as.character(merged$Project_nr))
meta <- data.frame(merged)

meta2 <- meta
meta2$Project_ID[meta2$Source=="Wild"] <- "Wild"
meta2$Project_nr[meta2$Source=="Wild"] <- 7.5
meta2$Project_ID[meta2$Source=="Lab"] <- "Lab"
meta2$Project_nr[meta2$Source=="Lab"] <- 14
meta <- rbind(meta, meta2)

meta$Color <- ifelse(meta$Project_ID == "Lab", "#C38D94", 
                     ifelse(meta$Project_ID == "Wild", "#096B72", "black"))

meta <- meta %>%
  mutate(Color = case_when(
    Project_ID == "Lab" ~ "#C38D94",
    Project_ID == "Wild" ~ "#096B72",
    TRUE ~ "black"
  ),
  Label = case_when(
    Project_nr == 7.5 ~ "W",
    Project_nr == 14 ~ "L",
    TRUE ~ as.character(Project_nr)
  ))

plot <- ggplot(data = meta, aes(x = reorder(Label, Project_nr), y = total_pathways, fill = Project_ID, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb", "BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg" = "#ffb1bc", "BMS-04/12/2020-PdgfraCreERT2" = "#ff8cbf", "C57BL_6-BMS_18102021-C57BL/6" = "#ff4e71", "C57BL_6-Kings_college-C57BL/6" = "#bb2545", "Lab-Kennedy-C57BL/6" = "#78061f", "Lab-Kennedy-SKG" = "#ab7777", "Lab" = "white", "Wild" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("Nr. functional pathways") +
  rremove("legend.title") +
  theme(axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 35),
        legend.position = "none",
        axis.text.x = element_text(face=c("plain", "plain", "plain", "plain", "plain", "plain", "plain", "bold",
                                          "plain", "plain", "plain", "plain", "plain", "plain", "bold"), size=20), 
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() + scale_y_continuous(n.breaks=6) 
p <- plot + geom_signif(comparisons = list(c("W", "L")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_nrPathways.png", width=6.5, height=6)

# How many pathways are found in both lab and wild mice, and how many are unique to one or the other?

dim(wl_qual) # 419 variables -> 415 pathways

wild <- wl_qual[which(wl_qual$Source=="Wild"),]
wild <- wild[,c(4:418)]
wild <- wild %>% select_if(colSums(.) != 0) 
wild_list <- colnames(wild) # 415 - all pathways that are found in lab are also found in the wild

lab <- wl_qual[which(wl_qual$Source=="Lab"),]
lab <- lab[,c(4:418)]
lab <- lab %>% select_if(colSums(.) != 0) 
lab_list <- colnames(lab) # 312 - not all pathways found in wild are found in lab

```

***Q5: Do wild and lab mouse samples cluster separate on principal coordinate analyses?***

```{r}

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
sample_data(phy)$Sample_ID <- rownames(sample_data(phy))
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)

# Jaccard
ord <- ordinate(phy, "PCoA", "jaccard", binary=T)
plot <- plot_ordination(phy, ord, color="Project_ID") +
                geom_point(size = 6, alpha=.9, stroke=0) +
  scale_color_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb", "BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg" = "#ffb1bc", "BMS-04/12/2020-PdgfraCreERT2" = "#ff8cbf", "C57BL_6-BMS_18102021-C57BL/6" = "#ff4e71", "C57BL_6-Kings_college-C57BL/6" = "#bb2545", "Lab-Kennedy-C57BL/6" = "#78061f", "Lab-Kennedy-SKG" = "#ab7777")) +
  theme_classic() + 
  labs(title = "Jaccard",
       x = "PCo1 (17.7%)",
       y = "PCo1 (5.0%)",
       color = "") +
  theme(axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        title = element_text(size=20),
        legend.text = element_text(size = 5),
        legend.position = "none")  #+
  stat_ellipse(type = "norm", linetype = 2)
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_PCoA_jaccard.png", width=6, height=6)

# Aitchison
phy_clr <- microbiome::transform(phy, "clr")
ord <- ordinate(phy_clr, "PCoA", "euclidean")
plot <- plot_ordination(phy, ord, color="Project_ID") +
                geom_point(size = 6, alpha=.9, stroke=0) +
  scale_color_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb", "BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg" = "#ffb1bc", "BMS-04/12/2020-PdgfraCreERT2" = "#ff8cbf", "C57BL_6-BMS_18102021-C57BL/6" = "#ff4e71", "C57BL_6-Kings_college-C57BL/6" = "#bb2545", "Lab-Kennedy-C57BL/6" = "#78061f", "Lab-Kennedy-SKG" = "#ab7777")) +
  theme_classic() + 
  labs(title = "Aitchison",
       x = "PCo1 (15.9%)",
       y = "PCo1 (4.5%)",
       color = "") +
  theme(axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        title = element_text(size=20),
        legend.text = element_text(size = 5),
        legend.position = "none")  #+
  stat_ellipse(type = "norm", linetype = 2)
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_PCoA_aitchison.png", width=6, height=6)
  
# UniFrac
ord <- ordinate(phy, "PCoA", "unifrac")
plot <- plot_ordination(phy, ord, color="Project_ID") +
                geom_point(size = 6, alpha=.9, stroke=0) +
  scale_color_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb", "BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg" = "#ffb1bc", "BMS-04/12/2020-PdgfraCreERT2" = "#ff8cbf", "C57BL_6-BMS_18102021-C57BL/6" = "#ff4e71", "C57BL_6-Kings_college-C57BL/6" = "#bb2545", "Lab-Kennedy-C57BL/6" = "#78061f", "Lab-Kennedy-SKG" = "#ab7777")) +
  theme_classic() + 
  labs(title = "UniFrac",
       x = "PCo1 (27.4%)",
       y = "PCo1 (7.8%)",
       color = "") +
  theme(axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        title = element_text(size=20),
        legend.text = element_text(size = 5),
        legend.position = "none")  #+
  stat_ellipse(type = "norm", linetype = 2)
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_PCoA_unifrac.png", width=6, height=6)
 
# Weighted UniFrac
ord <- ordinate(phy, "PCoA", "wunifrac")
plot <- plot_ordination(phy, ord, color="Project_ID") +
                geom_point(size = 6, alpha=.9, stroke=0) +
  scale_color_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb", "BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg" = "#ffb1bc", "BMS-04/12/2020-PdgfraCreERT2" = "#ff8cbf", "C57BL_6-BMS_18102021-C57BL/6" = "#ff4e71", "C57BL_6-Kings_college-C57BL/6" = "#bb2545", "Lab-Kennedy-C57BL/6" = "#78061f", "Lab-Kennedy-SKG" = "#ab7777")) +
  theme_classic() + 
  labs(title = "Weighted UniFrac",
       x = "PCo1 (44.6%)",
       y = "PCo1 (17.0%)",
       color = "") +
  theme(axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        title = element_text(size=20),
        legend.text = element_text(size = 5),
        legend.position = "none")  #+
  stat_ellipse(type = "norm", linetype = 2)
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_PCoA_wunifrac.png", width=6, height=6)
 
```

***Q6: Does strain or facility predict gut microbial variation in lab mice?***

```{r lab_PERMANOVA}

set.seed(86754)
phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- lab
sample_data(phy)$Facility <- "Facility A"
sample_data(phy)$Facility[sample_data(phy)$Date=="Kennedy"] <- "Facility C"
sample_data(phy)$Facility[sample_data(phy)$Date=="Kings_college"] <- "Facility B"
table(sample_data(phy)$Facility)
# Facility A Facility B Facility C 
#       25        112          9
table(sample_data(phy)$Strain)
#  C57BL/6 CCSP_rtTAtg/(tetO)75FTPBtg              PdgfraCreERT2                        SKG 
#     133                          4                          6                          3 

# Jaccard

M4 <- data.frame(sample_data(phy))
D <- phyloseq::distance(phy, method = "jaccard", type = "samples", binary=T)

adonis2(D ~ M4$read_count + M4$Strain + M4$Facility, na.rm=TRUE,
        by="margin")
# read count R2=0.01254, F=2.4461, p=0.002
# strain R2=0.08461, F=5.5027, p=0.001
# facility R2=0.16046, F=15.6528, p=0.001

adonis2(D ~ M4$read_count + M4$Strain, na.rm=TRUE,
        by="margin")
# strain R2=0.11422, F=6.1502, p=0.001

adonis2(D ~ M4$read_count + M4$Facility, na.rm=TRUE,
        by="margin")
# facility R2=0.19007, F=16.9308, p=0.001

beta <- betadisper(D, M4$Strain)
permutest(beta) # F=6.2008, p=0.003
beta <- betadisper(D, M4$Facility)
permutest(beta) # F=42.08, p=0.001

# Aitchison

M4 <- data.frame(sample_data(phy))
phy_clr <- microbiome::transform(phy, "clr")
D <- phyloseq::distance(phy_clr, method = "euclidean", type = "samples")

adonis2(D ~ M4$read_count + M4$Strain + M4$Facility, na.rm=TRUE,
        by="margin")
# read count R2=0.00942, F=1.9318, p=0.014
# strain R2=0.11098, F=7.5868, p=0.001
# facility R2=0.17871, F=18.3251, p=0.001

adonis2(D ~ M4$read_count + M4$Strain, na.rm=TRUE,
        by="margin")
# strain R2=0.13424, F=7.3666, p=0.001

adonis2(D ~ M4$read_count + M4$Facility, na.rm=TRUE,
        by="margin")
# facility R2=0.20197, F=18.1803, p=0.001

beta <- betadisper(D, M4$Strain)
permutest(beta) # F=1.1231, p=0.294
beta <- betadisper(D, M4$Facility)
permutest(beta) # F=91.143, p=0.001

```

***Plot taxonomic compositions at phylum and family level***

```{r}

library(microbiome)
library(dplyr)
library(hrbrthemes)
library(gcookbook)
library(tidyverse)
library(RColorBrewer)

nb.cols <- 29
mycolors <- colorRampPalette(brewer.pal(8, "Set3"))(nb.cols)
set.seed(1) 
mixed <- sample(mycolors)

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)

sd <- sample_data(phy)
key.x <- sd[,c("Project_ID", "Sample_ID")]
key.x$Sample_ID <- rownames(key.x)

unique(sample_data(phy)$Project_ID)
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Skokholm"] <- "7"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Isle_of_May"] <- "6"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Espelette"] <- "3"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Faroe"] <- "5"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Midway_Atoll"] <- "4"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Cologne"] <- "1"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Wytham"] <- "2"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Lab-Kennedy-C57BL/6"] <- "12"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Lab-Kennedy-SKG"] <- "13"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg"] <- "8"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="BMS-04/12/2020-PdgfraCreERT2"] <- "9"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="C57BL_6-BMS_18102021-C57BL/6"] <- "10"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="C57BL_6-Kings_college-C57BL/6"] <- "11"

grouped <- merge_samples(phy, "Project_ID")
sample_data(grouped)$Project_ID <- as.numeric(as.character(sample_data(grouped)$Project_ID))

# Phylum
pseq <- grouped %>%
  aggregate_taxa(level = "Phylum")
pseq <- aggregate_rare(pseq, level = "Phylum", detection = 1/100, prevalence = 1/100)
p <- plot_composition(pseq,
                      sample.sort = "Project_ID",
                      x.label = "Project_ID") +
  scale_fill_manual(values=mixed, name="Phylum") +
  theme_minimal() +
  #guides(fill=guide_legend(nrow=30,byrow=TRUE)) + 
  labs(fill = "Phylum", x = "", y = "Relative abundance",
       title = "") +
  theme(axis.text.x = element_text(size=3),
        legend.text = element_text(size=10),
        legend.title = element_text(size=15),
        axis.text.y = element_text(size=20),
        axis.title.y = element_text(size=20),
        plot.title=element_text(hjust=0.5),
        axis.ticks.x=element_blank())
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_phyla.png", width=6.2, height=6)
 
# Family
pseq <- grouped %>%
  aggregate_taxa(level = "Family")
pseq <- aggregate_rare(pseq, level = "Family", detection = 3/100, prevalence = 5/100)
p <- plot_composition(pseq,
                      sample.sort = "Project_ID",
                      x.label = "Project_ID") +
  scale_fill_manual(values=mixed, name="Family") +
  theme_minimal() +
  #guides(fill=guide_legend(nrow=30,byrow=TRUE)) + 
  labs(x = "", y = "Relative abundance",
       title = "", fill = "Family") +
  theme(axis.text.x = element_text(size=3),
        legend.text = element_text(size=10),
        legend.title = element_text(size=15),
        axis.text.y = element_text(size=20),
        axis.title.y = element_text(size=20),
        plot.title=element_text(hjust=0.5),
        axis.ticks.x=element_blank())
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_families.png", width=6.2, height=6)

# Muribaculaceae mean relative abundance in lab mice
taxa <- psmelt(lab)
taxa <- taxa[,c("Sample", "Family", "Abundance")]
taxa$sample_fam <- paste(taxa$Sample, taxa$Family, sep="-")
taxa <- taxa[,c("sample_fam", "Abundance")]
taxa <- as.data.table(taxa)[, lapply(.SD, sum), by = .(x = tolower(sample_fam))]
taxa$family <- sub("[^-]+", '', taxa$x)

Muribaculaceae <- taxa[which(taxa$family=="-muribaculaceae"),]
mean(Muribaculaceae$Abundance) # 0.4667427
sd(Muribaculaceae$Abundance) # 0.1412914
 
```

***Q7: Are there specific taxa indicative of source?***

```{r random_forest}

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")

library(randomForest)
set.seed(321)

predictors <- otu_table(phy)
dim(predictors) 
source <- as.factor(sample_data(phy)$Source) 
data <- data.frame(source, predictors)

mouse_source <- randomForest(source ~ ., data = data, ntree=10000, importance=T)
print(mouse_source)

## look at SV importance (mean decrease gini)
mouse_source_RF <- importance(mouse_source)
mouse_source_RF_dat <- data.frame(predictors = rownames(mouse_source_RF), mouse_source_RF)
mouse_source_RF_sort <- arrange(mouse_source_RF_dat, desc(MeanDecreaseGini)) # mean decrease gini = IncNodePurity --> measures how much the model error increases when a particular variable is randomly permuted or shuffled
mouse_source_RF_sort$predictors <- factor(mouse_source_RF_sort$predictors, levels = mouse_source_RF_sort$predictors)
mouse_source_RF_sub <- mouse_source_RF_sort[1:30, ]

## coloured by family
tax_tab <- tax_table(phy)
colnames(mouse_source_RF_sub) <- c('ASV_name', "Lab", "Wild", 'MeanDecreaseAccuracy', 'MeanDecreaseGini')
tax_SV_fam <- tax_tab[,5]
tax_SV_fam <- as.data.frame(tax_SV_fam)
tax_SV_fam$ASV_name <- rownames(tax_SV_fam)
phy_plusfam <- merge(mouse_source_RF_sub, tax_SV_fam, by='ASV_name', all.x = T)
phy_plusfam_sort <- arrange(phy_plusfam, desc(MeanDecreaseGini))

# Plot
top6 <- as.list(as.character(phy_plusfam_sort$ASV_name[c(1:10)]))
phy_top6 <- subset_taxa(phy, taxa_names(phy)%in%top6)

phy_top6.melt <- psmelt(phy_top6)

#calculate mean and se of abundances per source per ASV
library(plyr)
phy_top6_sum <-  ddply(phy_top6.melt, c("OTU", "Source"), summarise,
                       N   = length(Abundance),
                       mean = mean(Abundance) ,
                       sd  = sd(Abundance),
                       se  = sd / sqrt(N))
phy_top6_sum
colnames(phy_top6_sum)[colnames(phy_top6_sum)=="mean"] <- "Abundance"
#add in families
phy_top6_sum$Family <- phy_top6.melt$Family[match(phy_top6_sum$OTU, phy_top6.melt$OTU)]
phy_top6_sum$Genus <- phy_top6.melt$Genus[match(phy_top6_sum$OTU, phy_top6.melt$OTU)]
phy_top6_sum$Species <- phy_top6.melt$Species[match(phy_top6_sum$OTU, phy_top6.melt$OTU)]
phy_top6_sum$Fam_gen <- paste(phy_top6_sum$Family, phy_top6_sum$Genus, sep=" ")

#plot barplot showing mean and se per season per asv
#phy_top6_sum$Source[phy_top6_sum$Source=="Wild"] <- "W"
#phy_top6_sum$Source[phy_top6_sum$Source=="Lab"] <- "L"
p2 <- ggplot(phy_top6_sum, aes(x=Source, y=Abundance)) +
  geom_bar(stat = "identity",aes(fill=Fam_gen)) +
  geom_errorbar(data=phy_top6_sum,
                aes(ymax=Abundance+se, ymin=Abundance),
                width=0.5) +
  facet_wrap(~OTU, scales = "free_y", nrow = 1) +
  theme_minimal() +
  theme(axis.text.x = element_text(size=22, angle = 45),
        axis.text.y = element_text(size=30), legend.position = "none",
        axis.title = element_text(size=45), panel.spacing = unit(2, "lines")) +
  ggtitle("") + 
  labs(title = "",
       x = "",
       y = "Relative abundance",
       color = "") +
scale_fill_manual("legend", values = c("Muribaculaceae Other" = "#DCD6F7", "Muribaculaceae Muribaculum" = "#A6B1E1", "Desulfovibrionaceae Desulfovibrio" = "#B4869F", "Akkermansiaceae Akkermansia" = "#4E4C67"))
p <- p2 + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/LabWild_RandomForestTop10.png", width=24, height=12)

# Plot the relative abundance of those lab-indicative taxa separated by strain and facility
lab <- subset_samples(phy, Source=="Lab")
rf_otu <- data.frame(otu_table(lab))
asv_list <- as.list(phy_top6_sum$OTU)
if (!is.character(asv_list)) {
  asv_list <- as.character(asv_list)
}
rf_otu <- rf_otu %>% select(all_of(asv_list))
colnames(rf_otu)
names(rf_otu)[names(rf_otu) == "ACAGAGGTCTCAAGCGTTGTTCGGAATCACTGGGCGTAAAGCGTGCGTAGGCTGTTTCGTAAGTCGTGTGTGAAAGGCGCGGGCTCAACCCGCGGACGGCACATGATACTGCGAGACTAGAGTAATGGAGGGGGAACCGGAATTCTCGGTGTAGCAGTGAAATGCGTAGATATCGAGAGGAACACTCGTGGCGAAGGCGGGTTCCTGGACATTAACTGACGCTGAGGCACGAAGGCCAGGGGAGCGAAAGGGATTAGATACCCCTGTAGTCCTGGCAGTAAACGGTGCACGCTTGGTGTGCGGGGAATCGACCCCCTGCGTGCCGGAGCTAACGCGTTAAGCGTGCCGCCTGGGGAGTACGGTCGCAAGATTA"] <- "Akkermansiaceae, Akkermansia"
names(rf_otu)[names(rf_otu) == "ACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGCAGGCGGAGGTGCAAGTCAGCGGTCAAATTGCGGGGCTCAACCCCGTACTGCCGTTGAAACTGCATCCCTTGAGTGCGCGAGAAGTATGCGGAATGCGTGGTGTAGCGGTGAAATGCATAGATATCACGCAGAACTCCGATTGCGAAGGCAGCATACCGGCGCGCAACTGACGCTCATGCACGAAAGCGTGGGTATCGAACAGGATTAGATACCCTGGTAGTCCACGCAGTAAACGATGAACACTAACTGTTCGGCCAGAATGATGGCTGAGTGGTACAGCGAAAGCGTTAAGTGTTCCACCTGGGGAGTACGCCGGCAACGGTG"] <- "Muribaculaceae, other 1"
names(rf_otu)[names(rf_otu) == "ACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGCAGGCGGTCTGTTAAGTCAGCGGTCAAAGCCCGGGGCTCAACCCCGGCCCGCCGTTGAAACTGGCAGTCTCGAGTTGGAGAGAAGTATGCGGAATGCGCGGTGTAGCGGTGAAATGCATAGATATCGCGCAGAACTCCGATTGCGAAGGCAGCATGCCGGCTCCACACTGACGCTGAGGCACGAAAGCGTGGGTATCGAACAGGATTAGATACCCTGGTAGTCCACGCAGTAAACGATGAATGCTGGCTGTCCGGGGGGACTGACCCCTGGGTGGCGAAGCGAAAGCGATAAGCATTCCACCTGGGGAGTACGCCGGCAACGGTG"] <- "Muribaculaceae, other 2"
names(rf_otu)[names(rf_otu) == "ACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGAATATCAAGTCAGCGGTAAAAATTCGGGGCTCAACCCCGTCGTGCCGTTGAAACTGATGTTCTTGAGTGGGCGAGAAGTATGCGGAATGCGTGGTGTAGCGGTGAAATGCATAGATATCACGCAGAACTCCGATTGCGAAGGCAGCATACCGGCGCCCAACTGACGCTGAAGCACGAAAGCGTGGGTATCGAACAGGATTAGATACCCTGGTAGTCCACGCAGTAAACGATGAATGCTAATTGTCCGGAGGGATTGACCTCTGGGTGATACAGCGAAAGCGTTAAGCATTCCACCTGGGGAGTACGCCGGCAACGGTG"] <- "Muribaculaceae, Muribaculum"
names(rf_otu)[names(rf_otu) == "ACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGACTGTCAAGTCAGCGGTAAAATACGGGGGCTCAACCTCCGCCCGCCGTTGAAACTGACGGTCTTGAGTGGGCGAGAAGTATGCGGAATGCGTGGTGTAGCGGTGAAATGCATAGATATCACGCAGAACTCCGATTGCGAAGGCAGCATACCGGCGCCCAACTGACGCTGAAGCACGAAAGCGTGGGTATCGAACAGGATTAGATACCCTGGTAGTCCACGCAGTAAACGATGAATACTAGCTGTCCGGGGCGAATGAGCCCTGGGTGGCACAGCGAAAGCGTTAAGTATTCCACCTGGGGAGTACGCCGGCAACGGTG"] <- "Muribaculaceae, other 3"
names(rf_otu)[names(rf_otu) == "ACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGAGTGTCAAGTCAGCGGTAAAATTTCGGGGCTCAACCCCGTCGTGCCGTTGAAACTGACACCCTTGAGTGAGTGAGAAGTAAGCGGAATGCGTGGTGTAGCGGTGAAATGCATAGATATCACGCAGAACGCCGATTGCGAAGGCAGCTTACCGGCGCTCAACTGACGCTGAGGCACGAAAGTGCGGGTATCGAACAGGATTAGATACCCTGGTAGTCCGCACAGTAAACGATGAATGCTAGATGTCCGGGGCGAATGAGCTTTGGGTGTCACAGCGAAAGCGTTAAGCATTCCACCTGGGGAGTACGCCGGCAACGGTG"] <- "Muribaculaceae, other 4"
names(rf_otu)[names(rf_otu) == "ACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGAGTGTCAAGTCAGCGGTAAAATTTCGGGGCTCAACCCCGTCGTGCCGTTGAAACTGACACCCTTGAGTGAGTGAGAAGTAAGCGGAATGCGTGGTGTAGCGGTGAAATGCATAGATATCACGCAGAACGCCGATTGCGAAGGCAGCTTACCGGCGCTCAACTGACGCTGAGGCACGAAAGTGCGGGTATCGAACAGGATTAGATACCCTGGTAGTCCGCACAGTAAACGATGAATGCTAGATGTCCGGGGCGAATGAGTCCTGGGTGTCACAGCGAAAGCGTTAAGCATTCCACCTGGGGAGTACGCCGGCAACGGTG"] <- "Muribaculaceae, other 5"
names(rf_otu)[names(rf_otu) == "ACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGAGTGTCAAGTCAGCGGTAAAATTTCGGGGCTCAACCCCGTCGTGCCGTTGAAACTGACGCCCTTGAGTGAGTGAGAAGTAAGCGGAATGCGTGGTGTAGCGGTGAAATGCATAGATATCACGCAGAACGCCGATTGCGAAGGCAGCTTACCGGCGCTCAACTGACGCTGAGGCACGAAAGTGCGGGTATCGAACAGGATTAGATACCCTGGTAGTCCGCACAGTAAACGATGAATGCTAGATGTCCGGGGCGAATGAGTCCTGGGTGTCACAGCGAAAGCGTTAAGCATTCCACCTGGGGAGTACGCCGGCAACGGTG"] <- "Muribaculaceae, other 6"
names(rf_otu)[names(rf_otu) == "ACGGAGGATGCGAGCGTTATCCGGATTTATTGGGTTTAAAGGGTGCGTAGGCGGATCGTTAAGTCAGTGGTCAAATTGAGGGGCTCAACCCCTTCCCGCCATTGAAACTGGCGATCTTGAGTGGAAGAGAAGTATGCGGAATGCGTGGTGTAGCGGTGAAATGCATAGATATCACGCAGAACCCCGATTGCGAAGGCAGCATGCCGGCTTCCGACTGACGCTGAAGCACGAAAGCGTGGGGATCGAACAGGATTAGATACCCTGGTAGTCCACGCGGTAAACGATGAGCGCTAGGTGTCCGGGGGGAATGGCCCCTGGGTGCCACAGCGAAAGCGTTAAGCGCTCCACCTGGGGAGTACGCCGGCAACGGTG"] <- "Muribaculaceae, other 7"
names(rf_otu)[names(rf_otu) == "ACGGAGGGCGCGAGCGTTAATCGGAATTACTGGGCGTAAAGCGCGCGTAGGCTGTTGTGTAAGTTGGGGGTGAAATCCCACGGCTCAACCGTGGAACAGCCCTTGATACTGCGCGACTTGAATCCGGGAGAGGGTGGCGGAATTCCAGGTGTAGGAGTGAAATCCGTAGATATCTGGAGGAACATCAGTGGCGAAGGCGGCCACCTGGACCGGTATTGACGCTGAGGCGCGAAAGCGTGGGGAGCAAACAGGATTAGATACCCTGGTAGTCCACGCTGTAAACGATGGATGCTGGATGTCGGGGAGTAATCCCCGGTGTCGTAGCCAACGCGATAAGCATCCCGCCTGGGGAGTACGGTCGCAAGGCTG"] <- "Desulfovibrionaceae, Desulfovibrio"
rf_otu$Sample_ID <- rownames(rf_otu)
rf_otu2 <- melt(rf_otu, id.vars=c("Sample_ID"))
colnames(rf_otu2) <- c("Sample_ID", "Taxon", "Abundance")
meta <- data.frame(sample_data(lab))
meta2 <- merge(meta, rf_otu2, by.x="IMR_code", by.y="Sample_ID", all=T)
unique(meta2$Date)
meta2$Facility[meta2$Date=="Kings_college"] <- "Facility B"
meta2$Facility[meta2$Date=="BMS_18102021"] <- "Facility A"
meta2$Facility[meta2$Date=="Kennedy"] <- "Facility C"
meta2$Facility[meta2$Date=="04/12/2020"] <- "Facility A"
meta2$strain_facility <- paste(meta2$Strain, meta2$Facility, sep=", ")
meta2$strain_facility[meta2$strain_facility=="C57BL/6, Facility A"] <- "w C57BL/6, Facility A"

#calculate mean and se of abundances per source per ASV
labx <-  ddply(meta2, c("Taxon", "strain_facility"), summarise,
                       N   = length(Abundance),
                       mean = mean(Abundance) ,
                       sd  = sd(Abundance),
                       se  = sd / sqrt(N))

labx$Facility <- gsub(".*, ","", labx$strain_facility)
labx$Facility_num[labx$Facility=="Facility A"] <- 1
labx$Facility_num[labx$Facility=="Facility B"] <- 2
labx$Facility_num[labx$Facility=="Facility C"] <- 3
labx$Strain <- sub(",.*", "", labx$strain_facility)

p2 <- ggplot(labx, aes(x=reorder(strain_facility, Facility_num), y=mean)) +
  geom_bar(stat = "identity",aes(fill=Taxon)) +
  geom_errorbar(data=labx,
                aes(ymax=mean+se, ymin=mean),
                width=0.5) +
  facet_wrap(~Taxon, scales = "free_y", nrow = 3) +
  theme_minimal() +
  theme(axis.text.x = element_text(size=10, angle = 90),
        axis.text.y = element_text(size=15), legend.position = "none",
        axis.title = element_text(size=20), panel.spacing = unit(2, "lines")) +
  ggtitle("") + 
  labs(title = "",
       x = "",
       y = "Relative abundance",
       color = "") +
scale_fill_manual("legend", values = c("Muribaculaceae, other 1" = "#DCD6F7", "Muribaculaceae, other 2" = "#DCD6F7", "Muribaculaceae, other 3" = "#DCD6F7", "Muribaculaceae, other 4" = "#DCD6F7", "Muribaculaceae, other 5" = "#DCD6F7", "Muribaculaceae, other 6" = "#DCD6F7", "Muribaculaceae, other 7" = "#DCD6F7", "Muribaculaceae, Muribaculum" = "#A6B1E1", "Desulfovibrionaceae, Desulfovibrio" = "#B4869F", "Akkermansiaceae, Akkermansia" = "#4E4C67"))
p <- p2 + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/Lab_RandomForestTop10.png", width=12, height=14)

# Other plot - currently not used
colourCount = length(unique(phy_plusfam_sort$Family))
getPalette = colorRampPalette(brewer.pal(15, "Paired"))

long <- phy_plusfam_sort[,c("Lab", "Wild", "MeanDecreaseGini", "Family")]
long <- gather(long, Source, MeanDecreaseGini, Lab:Wild, factor_key=TRUE)

p1 <- ggplot(phy_plusfam_sort, aes(x = ASV_name, y = MeanDecreaseGini, fill=Family)) +
  geom_bar(stat = "identity") +
  xlab("Amplicon sequence variant") +
  theme_bw() +
  theme(axis.text=element_text(size=14), axis.title=element_text(size=16),
        axis.text.x = element_blank(),
        strip.text = element_text(size=16),
        legend.text = element_text(size=14),
        legend.title = element_text(size=16)) +
  scale_fill_manual(values = getPalette(colourCount))

```

```{r brms_drop_model}

# Create dyadic dataframe with source, project id and read depth difference

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
sample_data(phy)$Sample_ID <- rownames(sample_data(phy))
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)

df <- data.frame(sample_data(phy))
df <- df[,c("IMR_code", "read_count", "Source", "Project_ID")]
D <- phyloseq::distance(phy, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
dyadic <- merge(D_melt, df, by.x="Var1", by.y="IMR_code")
colnames(dyadic) <- c("Sample1", "Sample2", "Jaccard_dissimilarity", "Sample1_readcount", "Sample1_source", "Sample1_project")
dyadic <- merge(dyadic, df, by.x="Sample2", by.y="IMR_code")
colnames(dyadic) <- c("Sample2", "Sample1", "Jaccard_dissimilarity", "Sample1_readcount", "Sample1_source", "Sample1_project", "Sample2_readcount", "Sample2_source", "Sample2_project")
dyadic$readcount_diff <- abs(dyadic$Sample1_readcount-dyadic$Sample2_readcount)
dyadic$source_pair <- paste(dyadic$Sample1_source, dyadic$Sample2_source, sep="-")
dyadic$source_pair[dyadic$source_pair=="Lab-Wild"] <- "Wild-Lab"
dyadic <- dyadic[which(dyadic$source_pair!="Wild-Lab"),]
dyadic$project_pair <- paste(dyadic$Sample1_project, dyadic$Sample2_project, sep="-")
dyadic <- dyadic[which(dyadic$Sample1!=dyadic$Sample2),]
dyadic <- dyadic[,c("Sample1", "Sample2", "Jaccard_dissimilarity", "source_pair", "project_pair", "readcount_diff")]
dyadic$sample_pair <- paste(dyadic$Sample1, dyadic$Sample2, sep="-")
hist(dyadic$Jaccard_dissimilarity) # right leaning
hist(dyadic$readcount_diff) # left leaning
# -> use beta distribution
saveRDS(dyadic, "MUSMUS/MUSMUS_brm_dropmodels/MUSMUS_drop_LW/musmus_lab_wild_dyadic_data.RDS")

# Run on cluster: see MUSMUS_lab_wild_dropmodel.R

```

```{r taxa_prevalence}

# Check prevalence and population level mean relative abundance of Akkermansiaceae, Streptococcaceae, Enterobacteriaceae, and Muribaculaceae

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
sample_data(phy)$Sample_ID <- rownames(sample_data(phy))
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)

df <- data.frame(sample_data(phy))
key <- df[,c("Project_ID", "Source")]

# Prevalence across Project_IDs
families <- c("Akkermansiaceae", "Streptococcaceae", "Enterobacteriaceae", "Muribaculaceae")
project_ids <- unique(sample_data(phy)$Project_ID)
# Initialize an empty data frame to store the results
# Initialize an empty data frame to store the results
results <- data.frame(Project_ID = character(), Family = character(), Prevalence = numeric(), stringsAsFactors = FALSE)

# Loop over each Project_ID and Family
for (proj_id in project_ids) {
  for (family in families) {
    # Subset the phyloseq object by Project_ID
    project <- subset_samples(phy, Project_ID == proj_id)
    
    # Get the total number of samples in this Project_ID
    total_samples <- nrow(sample_data(project))
    
    # Subset the phyloseq object by Family
    project <- subset_taxa(project, Family == family)
    
    # Prune samples with zero abundance in the family
    project <- tryCatch(
      prune_samples(sample_sums(project) > 0, project),
      error = function(e) NULL
    )
    
    # Check if the pruning resulted in a valid phyloseq object
    if (!is.null(project) && nsamples(project) > 0) {
      # Calculate prevalence
      prevalence <- nrow(sample_data(project)) / total_samples
    } else {
      # If no samples remain or pruning resulted in an invalid object, set prevalence to 0
      prevalence <- 0
    }
    
    # Store the result
    results <- rbind(results, data.frame(Project_ID = proj_id, Family = family, Prevalence = prevalence))
  }
}

results <- merge(results, key, by="Project_ID", all=F)
library(dplyr)
results <- results %>%
  group_by(Family, Source) %>%
  mutate(Source_prevalence = mean(Prevalence)) %>%
  ungroup()

# Mean relative abundance
phy <- merge_samples(phy, group = "Project_ID") # 13 samples
phy <- microbiome::transform(phy, "compositional")
family <- tax_glom(phy, taxrank = "Family")
family <- psmelt(family)

Akkermansiaceae <- family[which(family$Family=="Akkermansiaceae"),]
Akkermansiaceae <- merge(Akkermansiaceae, key, by.x="Sample", by.y="Project_ID", all.x = T, all.y=F)
lab <- Akkermansiaceae[which(Akkermansiaceae$Source.y=="Lab"),]
summary(lab$Abundance)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.00000 0.01660 0.01660 0.01729 0.01660 0.05289 
wild <- Akkermansiaceae[which(Akkermansiaceae$Source.y=="Wild"),]
summary(wild$Abundance)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.0000000 0.0003311 0.0017599 0.0011863 0.0017599 0.0018994 

Streptococcaceae <- family[which(family$Family=="Streptococcaceae"),]
Streptococcaceae <- merge(Streptococcaceae, key, by.x="Sample", by.y="Project_ID", all.x = T, all.y=F)
lab <- Streptococcaceae[which(Streptococcaceae$Source.y=="Lab"),]
summary(lab$Abundance)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.000e+00 3.678e-05 3.678e-05 4.052e-05 3.678e-05 5.375e-04 
wild <- Streptococcaceae[which(Streptococcaceae$Source.y=="Wild"),]
summary(wild$Abundance)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.0002655 0.0047960 0.0047960 0.0253120 0.0299001 0.1671228 

Enterobacteriaceae <- family[which(family$Family=="Enterobacteriaceae"),]
Enterobacteriaceae <- merge(Enterobacteriaceae, key, by.x="Sample", by.y="Project_ID", all.x = T, all.y=F)
lab <- Enterobacteriaceae[which(Enterobacteriaceae$Source.y=="Lab"),]
summary(lab$Abundance)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.000e+00 0.000e+00 0.000e+00 1.575e-06 0.000e+00 1.533e-05 
wild <- Enterobacteriaceae[which(Enterobacteriaceae$Source.y=="Wild"),]
summary(wild$Abundance)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.000000 0.005812 0.025982 0.015680 0.025982 0.025982

Muribaculaceae <- family[which(family$Family=="Muribaculaceae"),]
Muribaculaceae <- merge(Muribaculaceae, key, by.x="Sample", by.y="Project_ID", all.x = T, all.y=F)
lab <- Muribaculaceae[which(Muribaculaceae$Source.y=="Lab"),]
summary(lab$Abundance)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#  0.3450  0.4636  0.4636  0.4667  0.4636  0.6820 
wild <- Muribaculaceae[which(Muribaculaceae$Source.y=="Wild"),]
summary(wild$Abundance)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.05724 0.10496 0.17684 0.14459 0.17684 0.32723

# Check prevalence and abundance of Muribaculum intestinale in lab vs wild
phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
lab <- subset_samples(phy, Source=="Lab") # 146
lab <- subset_taxa(lab, Species=="intestinale")
lab <- prune_samples(sample_sums(lab)>0, lab) # 145
otu <- data.frame(otu_table(lab))
summary(otu) # min 0.0008727, mean 0.0232647, median 0.0242612, max 0.0622270

wild <- subset_samples(phy, Source=="Wild") # 180
wild <- subset_taxa(wild, Species=="intestinale")
wild <- prune_samples(sample_sums(wild)>0, wild) # 26
otu <- data.frame(otu_table(wild))
summary(otu) # min 0.0001645, mean 0.0032499, median 0.0019358, max 0.0110378

phy <- merge_samples(phy, group = "Source") # 2 samples
phy <- microbiome::transform(phy, "compositional")
species <- tax_glom(phy, taxrank = "Species")

```

***Q8: Is microbial turnover rate faster in wild than lab mice?***

```{r turnover_Jaccard}

phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds") # Select data for longitudinal analysis. I have longitudinal samples only from Skokholm and King's College mice. I will exclude females showing signs of pregnancy (as for the rest of this study), but otherwise will keep all samples.
phy <- subset_samples(phy, Buffer!="R" & Buffer!="NB" & Buffer!="E" & Buffer!="PBS" & Buffer!="RNAlater")

phy <- subset_taxa(phy, Genus!="Imtechella")
phy <- subset_taxa(phy, Genus!="Allobacillus")
phy <- microbiome::transform(phy, "compositional")

samples_x <- subset_samples(phy, Project=="Skokholm") # 903 samples
samples_x <- data.frame(sample_data(samples_x))
active <- samples_x %>%
  filter(str_detect(Reprod, "Nipples|Pregnant|PREG|PREG?|NIPP|Nipples|Lact|Lactating|PERF|Perforate|TL|Testis large|Testis large+")) # 138
`%notin%` <- Negate(`%in%`)
samples_x <- samples_x[which(samples_x$trapdate%notin%active$trapdate),] # 765
samples_x <- subset_samples(phy, trapdate%in%samples_x$trapdate) # 765
samples_y <- subset_samples(phy, Date=="Kings_college") # 123
phy <- merge_phyloseq(samples_x, samples_y)

sample_data(phy)$datedate <- sample_data(phy)$Collection_date
sample_data(phy)$datedate <- strptime(as.character(sample_data(phy)$datedate), "%d/%m/%Y")
sample_data(phy)$datedate <- format(sample_data(phy)$datedate, "%Y-%m-%d") 
sample_data(phy)$Birthdate <- strptime(as.character(sample_data(phy)$Birthdate), "%d/%m/%Y")
sample_data(phy)$Birthdate <- format(sample_data(phy)$Birthdate, "%Y-%m-%d") 
sample_data(phy)$age <- difftime(sample_data(phy)$datedate, sample_data(phy)$Birthdate, units="days")

mice <- data.frame(sample_data(phy))
mice$Sample_ID <- rownames(mice)
long <- mice %>% group_by(Animal_ID) %>% filter(n()>1) # 729 - Select mice sampled longitudinally
table(long$Project)
#  C57BL_6 Skokholm 
#    106      623 
unique(long$Animal_ID) # exclude skok mice with no true animal id
long <- long[which(long$Animal_ID!="NA"),]
long <- long[which(long$Animal_ID!="SK0"),] 
duplicated(long$trapdate[long$Source=="Wild"],) # One is true
skok <- long[which(long$Source=="Wild"),]
dup <- skok[duplicated(skok$trapdate),]
dup <- skok[which(skok$trapdate%in%dup$trapdate),] # 4 - exclude these samples as not sure why duplicated
long <- long[which(long$trapdate%notin%dup$trapdate),] # 654

table(long$Source) # 99 lab (King's college, C57BL/6), 555 wild (Skokholm)
table(long$Animal_ID) # 2-12 samples each

check_lab <- long[which(long$Source=="Lab"),]
table(check_lab$Animal_ID) # 5-7
length(unique(check_lab$Animal_ID)) # 16

check_wild <- long[which(long$Source=="Wild"),]
table(check_wild$Animal_ID) # 2-12
length(unique(check_wild$Animal_ID)) # 179

time <- phy
sample_data(time)$Sample_ID <- rownames(sample_data(time))
sample_data(time)$include[sample_data(time)$Sample_ID%in%long$Sample_ID] <- "yes"
time <- subset_samples(time, include=="yes")
table(sample_data(time)$Project) # 99 lab, 555 wild - correct

# For lab mice, add cage density
# C00072610 C00079971 C00087205 C00090961 C00091454 C00096262 C00103059 C00103824 C00103825 C00115156 C00115157 C00116613 C00127404 
#       2         3         2         1         2         2         3         5         3         3         3         2         2 
sample_data(time)$cage_density <- 2
sample_data(time)$cage_density[sample_data(time)$Cage=="C00079971" | sample_data(time)$Cage=="C00103059" | sample_data(time)$Cage=="C00103825" | sample_data(time)$Cage=="C00115156" | sample_data(time)$Cage=="C00115157"] <- 3
sample_data(time)$cage_density[sample_data(time)$Cage=="C00103824"] <- 5
sample_data(time)$cage_density[sample_data(time)$Cage=="C00090961"] <- 1
sample_data(time)$cage_density[sample_data(time)$Source=="Wild"] <- NA
table(sample_data(time)$cage_density)
#  2  3  5 
# 12 53 34 

# Jaccard
D <- phyloseq::distance(time, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Mouse <- rownames(sd)
sd$newdate <- strptime(as.character(sd$Collection_date), "%d/%m/%Y") # here use current format
sd$Date <- format(sd$newdate, "%Y-%m-%d") 
sd <- sd[order(sd$Animal_ID, sd$Date, decreasing = F),]
sd$Time_point <- as.numeric(unlist(lapply(split(sd, sd$Animal_ID), function(x) { rank(x$Date) })))
sd <- data.frame(sd[,c("Source", "Mouse", "Animal_ID", "Date", "Time_point", "cage_density")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp", "mouse1_cage_density")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp", "mouse1_cage_density","mouse2_source", "mouse2_id", "mouse2_date", "mouse2_tp", "mouse2_cage_density")
distance <- na.omit(distance)
self <- distance[which(distance$mouse1_id==distance$mouse2_id),] # only self comparisons
first_tp <- sd[which(sd$Time_point==1),]
first_tp <- first_tp[,c("Animal_ID", "Date")]
self <- merge(self, first_tp, by.x="mouse1_id", by.y="Animal_ID", all.x=FALSE, all.y=FALSE)
self$added_days <- difftime(self$mouse2_date, self$Date, units="days") # counting how many days added to first sample
self <- self[which(self$added_days>0),]
self$added_days_step <- difftime(self$mouse2_date, self$mouse1_date, units="days")
self <- self[which(self$added_days_step>0),]
#self <- self[which(self$added_days_step>=6),]
self$tp <- paste(self$mouse1_tp, self$mouse2_tp, sep="-")
#self <- self[which(self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="2-3" | self$tp=="3-4" | self$tp=="4-5" | self$tp=="5-6" | self$tp=="6-7" | self$tp=="7-8" | self$tp=="8-9" | self$tp=="9-10"),]
self <- self[which(self$mouse1_tp!=self$mouse2_tp),]
self$added_days <- as.numeric(self$added_days)
self$added_days_step <- as.numeric(self$added_days_step) # 1079

wild <- self[which(self$mouse1_source=="Wild"),]
lab <- self[which(self$mouse1_source=="Lab"),]

range(wild$added_days) # 1-332
range(lab$added_days) # 9-197

# Focus on <=200 sampling window

wild <- wild[which(wild$added_days<=200),]
lab <- lab[which(lab$added_days<=200),]

# Cage density counts
lab_5 <- subset(lab, mouse1_cage_density == 5)
check <- lab_5[,c("mouse1_id", "mouse1")]
check <- check[!duplicated(check),]
table(check$mouse1_id) # 5 mice, 5-6 samples each
lab_3 <- subset(lab, mouse1_cage_density == 3)
check <- lab_3[,c("mouse1_id", "mouse1")]
check <- check[!duplicated(check),]
table(check$mouse1_id) # 9 mice, 4-6 samples each
lab_2 <- subset(lab, mouse1_cage_density == 2)
check <- lab_2[,c("mouse1_id", "mouse1")]
check <- check[!duplicated(check),]
table(check$mouse1_id) # 2 mice, 5 samples each

set.seed(6435)

### WILD
wild$aitch_sim <- wild$BC_distance
plot(wild$added_days_step, wild$aitch_sim) # Linear plateau relationship. Try fit a quadratic plateau model.
wildX <- wild

wildX$aitch_sim <- wildX$aitch_sim
wildY <- wildX[, c("added_days_step", "aitch_sim")]
nlsfit(wildY, model=4) # AIC -1.924682e+03

wildX$aitch_sim_log <- log(wildX$aitch_sim)
wildY <- wildX[, c("added_days_step", "aitch_sim_log")]
nlsfit(wildY, model=4) # AIC -1.270170e+03 -> use non-log transformed

wildX <- wild
wildX$aitch_sim <- wildX$aitch_sim
wildY <- wildX[, c("added_days_step", "aitch_sim")]
nlsfit(wildY, model=4)
# coefficient a                   5.534339e-01
# coefficient b                   2.180959e-02
# coefficient c                  -6.085500e-04
# adjusted r-squared              4.941000e-01
# AIC                            -1.924682e+03
# BIC                            -1.906223e+03
# maximum or minimum value for y  7.488404e-01
# critical point in x             1.791932e+01
range(wildY$aitch_sim) # 0.3185841 0.9132791
nlsfit(wildY, model=1) # AIC -1.590595e+03 --> delta AIC is thus 334.087

dta.nls <- nls(aitch_sim ~ (a + b * added_days_step + c * I(added_days_step^2)) * (added_days_step <= -0.5 * b/c) + (a + I(-b^2/(4 * c))) * (added_days_step > -0.5 * b/c), wildY, start=list(a=0.5534339,b=0.02180959,c=as.numeric(-0.0006085500)))
wildY$line <- predict(dta.nls)
plot <- ggplot(wildY, aes(x=added_days_step, y=aitch_sim)) +
  annotate("segment", x = 17.91932, 
           xend = 17.91932, y = 0, 
           yend = 1,
           colour = "gray50",size=1,linetype = "dashed")+
  geom_point(size=2, colour = "#096B72",alpha=.1) +
  xlab("Days elapsed") + ylab("Jaccard dissimilarity") + 
  theme_minimal() +
  theme(axis.title = element_text(size = 25), axis.text = element_text(size = 20),  legend.text = element_text(size=15)) +
  geom_line(aes(y =line),size=1,color='#096B72') + ylim(0,1) + xlim(0,200) + ggtitle("")
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/Wild_turnover_Jaccard.png", width=6.2, height=6)

### LAB
# Assign values to the variables and plot initial relationship
lab$aitch_sim <- lab$BC_distance
plot(lab$added_days_step, lab$aitch_sim) # Linear relationship

# Initial linear model
mdl.1 <- lm(aitch_sim ~ added_days_step, data = lab)
summary(mdl.1) #  F-statistic: 391.5 on 1 and 254 DF,  p-value: < 2.2e-16, Adjusted R-squared:  0.6049 
plot(mdl.1) # pretty good
AIC(mdl.1) # -697.6286 --> try log transformed as a sanity check

# Log-transformed linear model
mdl.1 <- lm(log(aitch_sim) ~ added_days_step, data = lab)
summary(mdl.1) #  322.8 on 1 and 254 DF,  p-value: < 2.2e-16, Adjusted R-squared:  0.5579 
plot(mdl.1) # also fairly good
AIC(mdl.1) # -272.9251 --> use non-log transformed data

mdl.1 <- lm(aitch_sim ~ added_days_step, data = lab)

# Prediction with non-log-transformed model
summary(lab$added_days_step) 
pdat <- expand.grid(added_days_step = seq(6, 200))

pred <- predict (mdl.1, newdata = pdat, na.rm = T,
                 type= "response", se.fit = TRUE, level=0,
                 interval = "prediction")
predframe <- data.frame (pdat, preds = pred$fit, se = pred$se.fit)
predframe_lab <- predframe

p1 <- ggplot(lab, aes(x=added_days_step, y=aitch_sim)) +
  geom_point(alpha=0.3, colour="#C38D94", size=2) +
  geom_line(aes(y=preds.fit), data=predframe_lab, colour="#C38D94", size=1) +
  geom_ribbon(data = predframe_lab, aes(y = NULL, ymin = preds.fit-(1.96*se),
                                        ymax = preds.fit+(1.96*se)), alpha=0.15, fill="#C38D94") +
  ylab("Jaccard dissimilarity") +
  xlab("Days elapsed") +
  theme_minimal() +
  theme(axis.title=element_text(size=25), axis.text = element_text(size=20),
        title = element_text(size=25)) +
  ggtitle("") + xlim(0,200) + ylim(0,1)  
p <- p1 + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/Lab_turnover_Jaccard.png", width=6.2, height=6.23)

# Lab mouse, now separated by cage density (3 or 5; there is also cage density=2, but these mice have been sampled only in short time intervals and there is a small sample size)
lab <- lab[which(lab$mouse1_cage_density!=2),]
lab$aitch_sim <- lab$BC_distance

# Subset data for two groups
lab_3 <- subset(lab, mouse1_cage_density == 3)
check <- lab_3[,c("mouse1_id", "mouse1")]
check <- check[!duplicated(check),]
table(check$mouse1_id) # 43 samples, 4-6 samples from 9 animals
lab_5 <- subset(lab, mouse1_cage_density == 5)
check <- lab_5[,c("mouse1_id", "mouse1")]
check <- check[!duplicated(check),]
table(check$mouse1_id) # 29 samples, 5-6 samples from 5 animals

# Fit separate linear models
lm_3 <- lm(aitch_sim ~ added_days_step, data = lab_3)
summary(lm_3) # F-statistic: 322.2 on 1 and 125 DF,  p-value: < 2.2e-16, Adjusted R-squared:  0.7183 
plot(lm_3) # looks fine
lm_5 <- lm(aitch_sim ~ added_days_step, data = lab_5)
summary(lm_5) # F-statistic: 198.6 on 1 and 97 DF,  p-value: < 2.2e-16, Adjusted R-squared:  0.6685 
plot(lm_5)

# Create prediction frames for both models
pred_3 <- data.frame(added_days_step = seq(min(lab$added_days_step), max(lab$added_days_step), length.out = 100))
pred_3$preds <- predict(lm_3, newdata = pred_3)

pred_5 <- data.frame(added_days_step = seq(min(lab$added_days_step), max(lab$added_days_step), length.out = 100))
pred_5$preds <- predict(lm_5, newdata = pred_5)

# Add group info for ggplot legend
pred_3$mouse1_cage_density <- 3
pred_5$mouse1_cage_density <- 5

# Combine both prediction datasets
predictions <- rbind(pred_3, pred_5)

# Plot with separate lines and point styles
p1 <- ggplot(lab, aes(x = added_days_step, y = aitch_sim, group = as.factor(mouse1_cage_density))) +
  geom_point(aes(shape = as.factor(mouse1_cage_density), fill = as.factor(mouse1_cage_density)), 
             size = 2, alpha = .8, color = "#C38D94") +  # Maintains original color
  geom_line(data = predictions, aes(y = preds, linetype = as.factor(mouse1_cage_density)), 
            size = 1, color = "#C38D94") +  # Maintains original color
  scale_shape_manual(values = c("3" = 1, "5" = 16)) +  # Empty circle for 3, filled for 5
  scale_linetype_manual(values = c("3" = "dashed", "5" = "solid")) +  # Dashed line for 3, solid for 5
  scale_fill_manual(values = c("3" = "white", "5" = "#C38D94")) +  # White fill for empty circle, original color for filled
  ylab("Jaccard dissimilarity") +
  xlab("Days elapsed") +
  theme_minimal() +
  theme(axis.title = element_text(size = 25), 
        axis.text = element_text(size = 20),
        title = element_text(size = 25),
        legend.position = "none") +  # Removes the legend
  ggtitle("") + xlim(0, 200) + ylim(0, 1)
p <- p1 + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/Lab_turnover_Jaccard_byCageDensity.png", width=6.2, height=6.23)

# For comparison, repeating the above but with across-individual rather than within-individual distances:

# Jaccard
D <- phyloseq::distance(time, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Mouse <- rownames(sd)
sd$newdate <- strptime(as.character(sd$Collection_date), "%d/%m/%Y") # here use current format
sd$Date <- format(sd$newdate, "%Y-%m-%d") 
sd <- sd[order(sd$Animal_ID, sd$Date, decreasing = F),]
sd$Time_point <- as.numeric(unlist(lapply(split(sd, sd$Animal_ID), function(x) { rank(x$Date) })))
sd <- data.frame(sd[,c("Source", "Mouse", "Animal_ID", "Date", "Time_point")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp", "mouse2_source", "mouse2_id", "mouse2_date", "mouse2_tp")
distance <- na.omit(distance)
self <- distance[which(distance$mouse1_id!=distance$mouse2_id),] # exclude self comparisons
self$added_days_step <- difftime(self$mouse2_date, self$mouse1_date, units="days")
self$added_days_step <- as.numeric(abs(self$added_days_step)) # 1079
self <- self[which(self$mouse1_source==self$mouse2_source),]# only keep within-system comparisons 

wild <- self[which(self$mouse1_source=="Wild"),]
lab <- self[which(self$mouse1_source=="Lab"),]

range(wild$added_days_step) # 0-730
range(lab$added_days_step) # 0-197

# Focus on <=200 sampling window

wild <- wild[which(wild$added_days_step<=200),]
lab <- lab[which(lab$added_days_step<=200),]

set.seed(8765)

### WILD
wild$aitch_sim <- wild$BC_distance
plot(wild$added_days_step, wild$aitch_sim) # Linear plateau relationship. Try fit a quadratic plateau model.
wildX <- wild

wildX$aitch_sim <- wildX$aitch_sim
wildY <- wildX[, c("added_days_step", "aitch_sim")]
nlsfit(wildY, model=4) # AIC -3.796202e+05

wildX$aitch_sim_log <- log(wildX$aitch_sim)
wildY <- wildX[, c("added_days_step", "aitch_sim_log")]
nlsfit(wildY, model=4) # AIC -3.337855e+05 -> use non-log transformed

wildX <- wild
wildX$aitch_sim <- wildX$aitch_sim
wildY <- wildX[, c("added_days_step", "aitch_sim")]
nlsfit(wildY, model=4)
# coefficient a                   8.150340e-01
# coefficient b                   1.593050e-03
# coefficient c                  -6.626000e-05
# adjusted r-squared              4.400000e-03
# AIC                            -3.796202e+05 --> -379620.2
# BIC                            -3.795816e+05
# maximum or minimum value for y  8.246092e-01
# critical point in x             1.202116e+01
range(wildY$aitch_sim) # 0.5284091 0.9901639
nlsfit(wildY, model=1) # linear model for comparison. AIC -379145.3 --> quadratic plateau model is a better fit

dta.nls <- nls(aitch_sim ~ (a + b * added_days_step + c * I(added_days_step^2)) * (added_days_step <= -0.5 * b/c) + (a + I(-b^2/(4 * c))) * (added_days_step > -0.5 * b/c), wildY, start=list(a=0.8150340,b=0.001593050,c=as.numeric(-0.00006626000)))
wildY$line <- predict(dta.nls)
plot <- ggplot(wildY, aes(x=added_days_step, y=aitch_sim)) +
  geom_point(size=1, colour = "#096B72",alpha=.04) +
  annotate("segment", x = 12.02116, 
           xend = 12.02116, y = 0, 
           yend = 1,
           colour = "gray50",size=1,linetype = "dashed")+
  xlab("Days elapsed") + ylab("Jaccard dissimilarity") + 
  theme_minimal() +
  theme(axis.title = element_text(size = 25), axis.text = element_text(size = 20),  legend.text = element_text(size=15)) +
  geom_line(aes(y =line),size=1,color='#00474C') + ylim(0,1) + xlim(0,200) + ggtitle("")
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/Wild_beta_Jaccard.png", width=6.2, height=6)

### LAB

# Only include within-cage comparisons
info <- data.frame(sample_data(time))
info <- info[which(info$Source=="Lab"),]
info <- info[,c("Sample_ID", "Cage")]

labX <- merge(lab, info, by.x="mouse1", by.y="Sample_ID", all.x=T, all.y=F)
names(labX)[names(labX) == "Cage"] <- "mouse1_cage"
labX <- merge(labX, info, by.x="mouse2", by.y="Sample_ID", all.x=T, all.y=F)
names(labX)[names(labX) == "Cage"] <- "mouse2_cage"
dim(labX) # 8994
labY <- labX[which(labX$mouse1_cage==labX$mouse2_cage),] # 1602

lab_save <- lab
lab <- labY

# Assign values to the variables and plot initial relationship
lab$aitch_sim <- lab$BC_distance
plot(lab$added_days_step, lab$aitch_sim) # Linear relationship

# Initial linear model
mdl.1 <- lm(aitch_sim ~ added_days_step, data = lab)
summary(mdl.1) #  2515 on 1 and 1600 DF,  p-value: < 2.2e-16, Adjusted R-squared:  0.611 
plot(mdl.1) # really good
AIC(mdl.1) # -4298.259 --> try log transformed as a sanity check

# Log-transformed linear model
mdl.1 <- lm(log(aitch_sim) ~ added_days_step, data = lab)
summary(mdl.1) #  2024 on 1 and 1600 DF,  p-value: < 2.2e-16, Adjusted R-squared:  0.5582 
plot(mdl.1) # worse
AIC(mdl.1) # -1410.615 --> use non-log transformed data

mdl.1 <- lm(aitch_sim ~ added_days_step, data = lab)

# Prediction with log-transformed model
summary(lab$added_days_step) 
pdat <- expand.grid(added_days_step = seq(6, 200))

pred <- predict (mdl.1, newdata = pdat, na.rm = T,
                 type= "response", se.fit = TRUE, level=0,
                 interval = "prediction")
predframe <- data.frame (pdat, preds = pred$fit, se = pred$se.fit)
predframe_lab <- predframe

p1 <- ggplot(lab, aes(x=added_days_step, y=aitch_sim)) +
  geom_point(alpha=0.3, colour="#C38D94", size=2) +
  geom_line(aes(y=preds.fit), data=predframe_lab, colour="#C38D94", size=1) +
  geom_ribbon(data = predframe_lab, aes(y = NULL, ymin = preds.fit-(1.96*se),
                                        ymax = preds.fit+(1.96*se)), alpha=0.15, fill="#C38D94") +
  ylab("Jaccard dissimilarity") +
  xlab("Days elapsed") +
  theme_minimal() +
  theme(axis.title=element_text(size=25), axis.text = element_text(size=20),
        title = element_text(size=25)) +
  ggtitle("") + xlim(0,200) + ylim(0,1)  
p <- p1 + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/Lab_beta_Jaccard.png", width=6.2, height=6.23)

```

```{r asv_retention}

phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds") # Select data for longitudinal analysis. I have longitudinal samples only from Skokholm and King's College mice. I will NOT exclude females showing signs of pregnancy (as for the rest of this study) to maximise sample numbers (will be very low regardless. Can always highlight individual level attributes in figure)
phy <- subset_samples(phy, Buffer!="R" & Buffer!="NB" & Buffer!="E" & Buffer!="PBS" & Buffer!="RNAlater")

phy <- subset_taxa(phy, Genus!="Imtechella")
phy <- subset_taxa(phy, Genus!="Allobacillus")
phy <- microbiome::transform(phy, "compositional")

samples_x <- subset_samples(phy, Project=="Skokholm") # 903 samples
#samples_x <- data.frame(sample_data(samples_x)) # Run this and other rows hashtagged if wanting to exclude based on reprod
#active <- samples_x %>%
 # filter(str_detect(Reprod, "Nipples|Pregnant|PREG|PREG?|NIPP|Nipples|Lact|Lactating|PERF|Perforate|TL|Testis large|Testis large+")) # 138
`%notin%` <- Negate(`%in%`)
#samples_x <- samples_x[which(samples_x$trapdate%notin%active$trapdate),] # 765
#samples_x <- subset_samples(phy, trapdate%in%samples_x$trapdate) # 765
samples_y <- subset_samples(phy, Date=="Kings_college") # 123
phy <- merge_phyloseq(samples_x, samples_y)

sample_data(phy)$datedate <- sample_data(phy)$Collection_date
sample_data(phy)$datedate <- strptime(as.character(sample_data(phy)$datedate), "%d/%m/%Y")
sample_data(phy)$datedate <- format(sample_data(phy)$datedate, "%Y-%m-%d") 
sample_data(phy)$Birthdate <- strptime(as.character(sample_data(phy)$Birthdate), "%d/%m/%Y")
sample_data(phy)$Birthdate <- format(sample_data(phy)$Birthdate, "%Y-%m-%d") 
sample_data(phy)$age <- difftime(sample_data(phy)$datedate, sample_data(phy)$Birthdate, units="days")

mice <- data.frame(sample_data(phy))
mice$Sample_ID <- rownames(mice)
long <- mice %>% group_by(Animal_ID) %>% filter(n()>1)

# Check what kind of sampling frequencies we have
info <- long[,c("Source", "Sample_ID", "Animal_ID", "datedate")]
info <- na.omit(info)
info <- info[which(info$Animal_ID!="SK0"),]
info <- info[which(info$Sample_ID!="Sample252" & info$Sample_ID!="Sample52"),] # these have same animal id and date -> remove
info$date2 <- info$datedate
info$datedate <- strptime(as.character(info$datedate), "%Y-%m-%d")
info$datedate <- format(info$datedate, "%Y/%m/%d") 
info <- info[order(info$Animal_ID, info$datedate, decreasing = F),]
info$Time_point <- as.numeric(unlist(lapply(split(info, info$Animal_ID), function(x) { rank(x$datedate) })))

info2 <- info[which(info$Time_point==1),]
info3 <- merge(info, info2, by="Animal_ID", all=T)
info3$days_apart_starting_tp1 <- as.numeric(abs(difftime(info3$datedate.x, info3$datedate.y, units="days")))

info2 <- info[which(info$Time_point==2),]
info3 <- merge(info3, info2, by="Animal_ID", all=T)
info3$days_apart_starting_tp2 <- as.numeric(abs(difftime(info3$datedate.x, info3$datedate, units="days")))

info2 <- info[which(info$Time_point==3),]
names(info2)[names(info2) == "datedate"] <- "datedate3"
info3 <- merge(info3, info2, by="Animal_ID", all=T)
info3$days_apart_starting_tp3 <- as.numeric(abs(difftime(info3$datedate.x, info3$datedate3, units="days")))

info2 <- info[which(info$Time_point==4),]
names(info2)[names(info2) == "datedate"] <- "datedate4"
info3 <- merge(info3, info2, by="Animal_ID", all=T)
info3$days_apart_starting_tp4 <- as.numeric(abs(difftime(info3$datedate.x, info3$datedate4, units="days")))

info2 <- info[which(info$Time_point==5),]
names(info2)[names(info2) == "datedate"] <- "datedate5"
info3 <- merge(info3, info2, by="Animal_ID", all=T)
info3$days_apart_starting_tp5 <- as.numeric(abs(difftime(info3$datedate.x, info3$datedate5, units="days")))

info2 <- info[which(info$Time_point==6),]
names(info2)[names(info2) == "datedate"] <- "datedate6"
info3 <- merge(info3, info2, by="Animal_ID", all=T)
info3$days_apart_starting_tp6 <- as.numeric(abs(difftime(info3$datedate.x, info3$datedate6, units="days")))

info2 <- info[which(info$Time_point==7),]
names(info2)[names(info2) == "datedate"] <- "datedate7"
info3 <- merge(info3, info2, by="Animal_ID", all=T)
info3$days_apart_starting_tp7 <- as.numeric(abs(difftime(info3$datedate.x, info3$datedate7, units="days")))

info2 <- info[which(info$Time_point==8),]
names(info2)[names(info2) == "datedate"] <- "datedate8"
info3 <- merge(info3, info2, by="Animal_ID", all=T)
info3$days_apart_starting_tp8 <- as.numeric(abs(difftime(info3$datedate.x, info3$datedate8, units="days")))

info3 <- info3[which(info3$Time_point.x>info3$Time_point.y),]

# For lab mice, have the following sampling frequencies:
# A: 49,79,108,142,169,197 days from first sample
# B: 9,15,23,29,45 days from first sample

# For lab mice, go with days 0, 9, 15, 23, and 29 (leave 45 out as no comparable timewindows for wild mice) and for wild mice select samples with following timewindows: 0, 6–8, 12–16, 22–23, and 28–32e.

lab <- info3[which(info3$Source.x=="Lab" & info3$days_apart_starting_tp1<35),]
unique(lab$Animal_ID) # 8

wild_check <- info3[which(info3$Source.x=="Wild"),]
wild4 <- wild_check[which(wild_check$Time_point.x==5),] # get mice with min. 5 samples
wild_check <- wild_check[which(wild_check$Animal_ID%in%wild4$Animal_ID),] 
unique(wild_check$Animal_ID) # 40 mice - inspect available sampling timewindows
wild_check <- wild_check[which(wild_check$Animal_ID!="SK0013" 
                               & wild_check$Animal_ID!="SK0019" 
                               & wild_check$Animal_ID!="SK0001" 
                               & wild_check$Animal_ID!="SK0003" 
                               & wild_check$Animal_ID!="SK0006" 
                               & wild_check$Animal_ID!="SK0010"
                               & wild_check$Animal_ID!="SK0013"
                               & wild_check$Animal_ID!="SK0019"
                               & wild_check$Animal_ID!="SK0022"
                               & wild_check$Animal_ID!="SK0035"
                               & wild_check$Animal_ID!="SK0046"
                               & wild_check$Animal_ID!="SK0179"
                               & wild_check$Animal_ID!="SK0050"
                               & wild_check$Animal_ID!="SK0203"
                               & wild_check$Animal_ID!="SK0205"
                               & wild_check$Animal_ID!="SK0222"
                               & wild_check$Animal_ID!="SK0234"
                               & wild_check$Animal_ID!="SK0237"
                               & wild_check$Animal_ID!="SK0249"
                               & wild_check$Animal_ID!="SK0251"
                               & wild_check$Animal_ID!="SK0260"
                               & wild_check$Animal_ID!="SK0262"
                               & wild_check$Animal_ID!="SK0271"
                               & wild_check$Animal_ID!="SK0282"
                               & wild_check$Animal_ID!="SK0284"
                               & wild_check$Animal_ID!="SK0293"
                               & wild_check$Animal_ID!="SK0319"
                               & wild_check$Animal_ID!="SK0330" 
                               & wild_check$Animal_ID!="SK0022" 
                               &wild_check$Animal_ID!="SK0050" 
                               & wild_check$Animal_ID!="SK0203" 
                               &wild_check$Animal_ID!="SK0205" 
                               &wild_check$Animal_ID!="SK0222" 
                               &wild_check$Animal_ID!="SK0226" 
                               &wild_check$Animal_ID!="SK0271" 
                               &wild_check$Animal_ID!="SK0282" 
                               &wild_check$Animal_ID!="SK0284" 
                               &wild_check$Animal_ID!="SK0319" 
                               &wild_check$Animal_ID!="SK0330" 
                               &wild_check$Animal_ID!="X041A1F2324" 
                               &wild_check$Animal_ID!="X041A1F2404" 
                               &wild_check$Animal_ID!="X041A1F25AE" 
                               &wild_check$Animal_ID!="X041A1F335E" 
                               &wild_check$Animal_ID!="X041A556696" 
                               &wild_check$Animal_ID!="X041A1F2FC6" 
                               &wild_check$Animal_ID!="X041A557019"
                               &wild_check$Animal_ID!="X041A1F2DB5"
                               & wild_check$Animal_ID!="X041A556CB0"),] # Exclude those without any suitable sampling time windows
unique(wild_check$Animal_ID) # 4

other2 <- wild_check[
  order( wild_check[,1], wild_check[,4] ),]

SK0223 <- other2[which(other2$Animal_ID=="SK0223"),]
SK0223$days_apart_starting_tp1 # 4  6  9 11 15 23 31 32 33 
SK0223$days_apart_starting_tp2 # 0  2  5  7 11 19 27 28 29 
SK0223$days_apart_starting_tp3 # 2  0  3  5  9 17 25 26 27 
SK0223$days_apart_starting_tp4 # 5  3  0  2  6 14 22 23 24
SK0223$days_apart_starting_tp5 # 7  5  2  0  4 12 20 21 22
SK0223$days_apart_starting_tp6 # 11  9  6  4  0  8 16 17 18
# --> For SK0223, select 1st, 3rd, 6th, 7th and 9th samples (0, 6, 15, 23, 32)

SK0224 <- other2[which(other2$Animal_ID=="SK0224"),]
SK0224$days_apart_starting_tp1 # 2  6  8 14 22 23 24 30 31 32 
SK0224$days_apart_starting_tp2 # 0  4  6 12 20 21 22 28 29 30 
SK0224$days_apart_starting_tp3 # 4  0  2  8 16 17 18 24 25 26 
SK0224$days_apart_starting_tp4 # 6  2  0  6 14 15 16 22 23 24 
SK0224$days_apart_starting_tp5 # 12  8  6  0  8  9 10 16 17 18 
SK0224$days_apart_starting_tp6 # 20 16 14  8  0  1  2  8  9 10
SK0224$days_apart_starting_tp7 # 21 17 15  9  1  0  1  7  8  9
# --> For SK0224, select 1st, 3rd, 5th, 8th and 10th samples (0, 6, 14, 24, 31)

SK0209 <- other2[which(other2$Animal_ID=="SK0209"),]
SK0209$days_apart_starting_tp1 # 3  8 16 22 32 39 40
SK0209$days_apart_starting_tp2 # 0  5 13 19 29 36 37
SK0209$days_apart_starting_tp3 # 5  0  8 14 24 31 32
SK0209$days_apart_starting_tp4 # 13  8  0  6 16 23 24
SK0209$days_apart_starting_tp5 # 19 14  6  0 10 17 18
SK0209$days_apart_starting_tp6 # 29 24 16 10  0  7  8
SK0209$days_apart_starting_tp7 # 36 31 23 17  7  0  1
# --> For SK0209, select 1st, 3rd, 4th, 5th and 6th samples (0, 8, 16, 22, 32)

SK0235 <- other2[which(other2$Animal_ID=="SK0235"),]
SK0235$days_apart_starting_tp1 # 1  3  5  6 11 12 20 22 28 29 30 
SK0235$days_apart_starting_tp2 # 0  2  4  5 10 11 19 21 27 28 29 
SK0235$days_apart_starting_tp3 # 2  0  2  3  8  9 17 19 25 26 27 
SK0235$days_apart_starting_tp4 # 4  2  0  1  6  7 15 17 23 24 25 
SK0235$days_apart_starting_tp5 # 5  3  1  0  5  6 14 16 22 23 24
SK0235$days_apart_starting_tp6 # 10  8  6  5  0  1  9 11 17 18 19
SK0235$days_apart_starting_tp7 # 11  9  7  6  1  0  8 10 16 17 18
SK0235$days_apart_starting_tp8 # 19 17 15 14  9  8  0  2  8  9 10
# --> For SK0235 select 1st, 5th, 7th, 9th and 12th samples (0, 6, 12, 22, 30)

final <- other2[which(other2$Animal_ID=="SK0223" | other2$Animal_ID=="SK0224" | other2$Animal_ID=="SK0235" | other2$Animal_ID=="SK0209"),]
unique(final$Animal_ID) # 4
final2 <- rbind(final, lab)
unique(final2$Animal_ID) # 12 - correct (4 wild, 8 lab)

info <- long[,c("Source", "Sample_ID", "Animal_ID", "datedate")]
info <- na.omit(info)
info <- info[which(info$Animal_ID!="SK0"),]
info <- info[which(info$Sample_ID!="Sample252" & info$Sample_ID!="Sample52"),] # these have same animal id and date -> remove
info$date2 <- info$datedate
info$datedate <- strptime(as.character(info$datedate), "%Y-%m-%d")
info$datedate <- format(info$datedate, "%Y/%m/%d") 
info <- info[order(info$Animal_ID, info$datedate, decreasing = F),]
info$Time_point <- as.numeric(unlist(lapply(split(info, info$Animal_ID), function(x) { rank(x$datedate) })))

info <- info[which(info$Animal_ID%in%final2$Animal_ID),] # 85
info <- info[which(info$Source=="Wild" | info$Source=="Lab" & info$Time_point<6),] # exclude 6th timepoints from lab mice
SK0223 <- info[which(info$Animal_ID=="SK0223"),]
SK0223 <- SK0223[which(SK0223$Time_point==1 | SK0223$Time_point==3 | SK0223$Time_point==6 | SK0223$Time_point==7 | SK0223$Time_point==9),]
SK0224 <- info[which(info$Animal_ID=="SK0224"),]
SK0224 <- SK0224[which(SK0224$Time_point==1 | SK0224$Time_point==3 | SK0224$Time_point==5 | SK0224$Time_point==8 | SK0224$Time_point==10),]
SK0235 <- info[which(info$Animal_ID=="SK0235"),]
SK0235 <- SK0235[which(SK0235$Time_point==1 | SK0235$Time_point==5 | SK0235$Time_point==7 | SK0235$Time_point==9 | SK0235$Time_point==12),]
SK0209 <- info[which(info$Animal_ID=="SK0209"),]
SK0209 <- SK0209[which(SK0209$Time_point==1 | SK0209$Time_point==3 | SK0209$Time_point==4 | SK0209$Time_point==5 | SK0209$Time_point==6),]
labids <- lab$Sample_ID.x
labids2 <- lab$Sample_ID.y
labids <- c(labids, labids2)
info <- info[which(info$Sample_ID%in%labids),]
info <- info[which(info$Animal_ID!="SK0223" & info$Animal_ID!="SK0224" & info$Animal_ID!="SK0235" & info$Animal_ID!="SK0209"),]
info1 <- rbind(SK0223, SK0224, SK0235, SK0209, info)
table(info1$Animal_ID) # 5 from each except MCUR23.7d, MCUR23.7e, MCUR25.1b which have four -> remove
info1 <- info1[which(info1$Animal_ID!="MCUR23.7d" & info1$Animal_ID!="MCUR23.7e" & info1$Animal_ID!="MCUR25.1b"),]
lll <- info1[which(info1$Source=="Lab"),]
lll <- lll[which(lll$Time_point!=6),] # 5 animals - select 4 randomly to have same sample size as for wild mice
random <- lll[which(lll$Time_point==1),]
random <- random[sample(nrow(random), size=4), ] # 4
lll <- lll[which(lll$Animal_ID%in%random$Animal_ID),] # 20
www <- info1[which(info1$Source=="Wild"),]
info1 <- rbind(lll, www)
table(info1$Animal_ID)

info <- info1
info <- info[order(info$Animal_ID, info$datedate, decreasing = F),]
info$Time_point <- as.numeric(unlist(lapply(split(info, info$Animal_ID), function(x) { rank(x$datedate) })))
info1 <- info[which(info$Time_point==1),]
info3 <- merge(info1, info, by="Animal_ID")
info3$Days_elapsed <- as.numeric(abs(difftime(info3$datedate.y, info3$datedate.x, units="days"))) 
key <- info3[,c("Sample_ID.y", "Time_point.y", "Days_elapsed")]
sampleids <- data.frame(info3$Sample_ID.x)
sampleids2 <- data.frame(info3$Sample_ID.y)
names(sampleids2)[names(sampleids2) == "info3.Sample_ID.y"] <- "info3.Sample_ID.x"
sampleids <- rbind(sampleids, sampleids2)
sampleids <- sampleids[!duplicated(sampleids),]

phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds")

phy <- subset_taxa(phy, Genus!="Imtechella")
phy <- subset_taxa(phy, Genus!="Allobacillus")
phy <- microbiome::transform(phy, "compositional")

sample_data(phy)$Sample_ID <- rownames(sample_data(phy))
phy <- subset_samples(phy, Sample_ID %in% sampleids)
phy 
df <- data.frame(sample_data(phy))
df2 <- data.frame(merge(df, key, by.x="Sample_ID", by.y="Sample_ID.y"))
names(df2)[names(df2) == "Time_point.y"] <- "Time_point"
rownames(df2) <- df2$Sample_ID
sample_data(phy) <- df2

library(phyloseq)
library(dplyr)
library(tidyr)

# Assuming your phyloseq object is named `phy`
# Extract sample data
sample_data_df <- data.frame(sample_data(phy))

# Ensure the sample data has the necessary columns
required_cols <- c("Sample_ID", "Animal_ID", "Time_point", "Days_elapsed")
stopifnot(all(required_cols %in% colnames(sample_data_df)))

# Extract ASV abundance data
asv_abundance_df <- as.data.frame(t(otu_table(phy)))

# Convert to long format for easier manipulation
asv_abundance_long <- asv_abundance_df %>%
  rownames_to_column(var = "ASV") %>%
  pivot_longer(cols = -ASV, names_to = "Sample_ID", values_to = "Abundance")

# Merge with sample data
asv_sample_df <- asv_abundance_long %>%
  inner_join(sample_data_df, by = "Sample_ID")

# Initialize an empty list to store results
results_list <- list()

# Unique Animal_IDs
unique_animals <- unique(sample_data_df$Animal_ID)

# Iterate over each Animal_ID
for (animal in unique_animals) {
  animal_data <- asv_sample_df %>% filter(Animal_ID == animal)
  
  # Unique Time_points for the current animal
  unique_time_points <- sort(unique(animal_data$Time_point))
  
  # Initialize a list to store first detected ASVs at each time point
  asvs_first_detected <- vector("list", length(unique_time_points))
  names(asvs_first_detected) <- unique_time_points
  
  for (i in 1:length(unique_time_points)) {
    time_point <- unique_time_points[i]
    current_data <- animal_data %>% filter(Time_point == time_point)
    
    # ASVs in the current time point
    current_asvs <- current_data %>% filter(Abundance > 0) %>% select(ASV) %>% unique()
    
    # Store detected ASVs
    if (i == 1) {
      asvs_first_detected[[i]] <- current_asvs
    } else {
      previous_asvs <- do.call(rbind, asvs_first_detected[1:(i-1)])
      asvs_first_detected[[i]] <- current_asvs %>% filter(!ASV %in% previous_asvs$ASV)
    }
    
    # Calculate ASV count
    asv_count <- nrow(current_asvs)
    
    # Initialize proportions and relative abundances
    proportions <- rep(0, 5)
    relative_abundances <- rep(0, 5)
    
    # Calculate proportions and relative abundances
    for (j in 1:i) {
      first_detected_asvs <- asvs_first_detected[[j]]$ASV
      common_asvs <- current_asvs %>% filter(ASV %in% first_detected_asvs)
      
      if (asv_count > 0) {
        prop <- nrow(common_asvs) / asv_count
      } else {
        prop <- 0
      }
      
      if (sum(current_data$Abundance) > 0) {
        rel_abundance <- sum(current_data$Abundance[current_data$ASV %in% common_asvs$ASV]) / sum(current_data$Abundance)
      } else {
        rel_abundance <- 0
      }
      
      proportions[j] <- prop
      relative_abundances[j] <- rel_abundance
    }
    
    # Create a result row
    result_row <- data.frame(
      Sample_ID = current_data$Sample_ID[1],
      Animal_ID = animal,
      Time_point = time_point,
      Days_elapsed = current_data$Days_elapsed[1],
      ASV_count = asv_count,
      Proportion_of_ASVs_first_detected_at_TP1 = proportions[1],
      Proportion_of_ASVs_first_detected_at_TP2 = proportions[2],
      Proportion_of_ASVs_first_detected_at_TP3 = proportions[3],
      Proportion_of_ASVs_first_detected_at_TP4 = proportions[4],
      Proportion_of_ASVs_first_detected_at_TP5 = proportions[5],
      Relative_abundance_of_ASVs_first_detected_at_TP1 = relative_abundances[1],
      Relative_abundance_of_ASVs_first_detected_at_TP2 = relative_abundances[2],
      Relative_abundance_of_ASVs_first_detected_at_TP3 = relative_abundances[3],
      Relative_abundance_of_ASVs_first_detected_at_TP4 = relative_abundances[4],
      Relative_abundance_of_ASVs_first_detected_at_TP5 = relative_abundances[5]
    )
    
    results_list[[length(results_list) + 1]] <- result_row
  }
}

# Combine all results into a single dataframe
results_df <- bind_rows(results_list)

# View the resulting dataframe
print(results_df)

# Plot the above
results_df$Source <- "Lab"
unique(results_df$Animal_ID)
results_df$Source[results_df$Animal_ID=="SK0249" | results_df$Animal_ID=="SK0235" | results_df$Animal_ID=="SK0223" | results_df$Animal_ID=="SK0224" | results_df$Animal_ID=="SK0209"] <- "Wild"

# Gather proportions into long format
proportion_long <- results_df %>%
  pivot_longer(cols = starts_with("Proportion_of_ASVs_first_detected_at_TP"), 
               names_to = "Time_detected", 
               names_prefix = "Proportion_of_ASVs_first_detected_at_TP", 
               values_to = "Proportion")

# Gather relative abundances into long format
relative_abundance_long <- results_df %>%
  pivot_longer(cols = starts_with("Relative_abundance_of_ASVs_first_detected_at_TP"), 
               names_to = "Time_detected", 
               names_prefix = "Relative_abundance_of_ASVs_first_detected_at_TP", 
               values_to = "Relative_abundance")

# Convert Time_detected to a numeric value
proportion_long$Time_detected <- as.numeric(proportion_long$Time_detected)
relative_abundance_long$Time_detected <- as.numeric(relative_abundance_long$Time_detected)

# Split to lab and wild
lab_prop <- proportion_long[which(proportion_long$Source=="Lab"),]
wild_prop <- proportion_long[which(proportion_long$Source=="Wild"),]
lab_rel <- relative_abundance_long[which(relative_abundance_long$Source=="Lab"),]
wild_rel <- relative_abundance_long[which(relative_abundance_long$Source=="Wild"),]

# Plot lab mice:

# Calculate mean proportions and relative abundances
lab_mean_proportion <- lab_prop %>%
  group_by(Time_point, Time_detected) %>%
  summarise(mean_proportion = mean(Proportion, na.rm = TRUE)) %>%
  ungroup()

lab_mean_relative_abundance <- lab_rel %>%
  group_by(Time_point, Time_detected) %>%
  summarise(mean_relative_abundance = mean(Relative_abundance, na.rm = TRUE)) %>%
  ungroup()

# Reverse the levels of Time_detected for proper stacking order
lab_mean_proportion$Time_detected <- factor(lab_mean_proportion$Time_detected, levels = rev(sort(unique(lab_mean_proportion$Time_detected))))
lab_mean_relative_abundance$Time_detected <- factor(lab_mean_relative_abundance$Time_detected, levels = rev(sort(unique(lab_mean_relative_abundance$Time_detected))))

# Plot Mean Proportion of ASVs
plot <- ggplot(lab_mean_proportion, aes(x = Time_point, y = mean_proportion, fill = Time_detected)) +
  theme_minimal() +
  geom_area(position = "fill", size = .2, alpha = .5) +
  scale_y_continuous(labels = scales::percent) + 
  labs(y = "Proportion of ASVs (%)", fill = "ASV first detected at") + 
  scale_fill_manual(values = c("#fedee2", "#e4bbc0", "#cb989e", "#b1777e", "#97565f")) +
  theme(
    axis.ticks = element_line(size = 0, color = "black"),
    axis.ticks.length = unit(0.2, "cm"),
    axis.text.x = element_text(size = 20),
    axis.text.y = element_text(size = 20),
    axis.title.y = element_text(size = 22)
  )
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/Lab_ASV_proportions.png", width=8, height=6)

# Plot Relative Abundance of ASVs
plot <- ggplot(lab_mean_relative_abundance, aes(x = Time_point, y = mean_relative_abundance, fill = factor(Time_detected))) +
  theme_minimal() +
  geom_area(position = "fill", size = .2, alpha = .5) +
  scale_y_continuous(labels = scales::percent) + 
  labs(y = "Relative abundance of ASVs (%)", fill = "ASV first detected at") + 
  scale_fill_manual(values = c("#fedee2", "#e4bbc0", "#cb989e", "#b1777e", "#97565f")) +
  theme(
    axis.ticks = element_line(size = 0, color = "black"),
    axis.ticks.length = unit(0.2, "cm"),
    axis.text.x = element_text(size = 20),
    axis.text.y = element_text(size = 20),
    axis.title.y = element_text(size = 22)
  )
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/Lab_ASV_abundances.png", width=8, height=6)

# Plot wild mice:

# Calculate mean proportions and relative abundances
wild_mean_proportion <- wild_prop %>%
  group_by(Time_point, Time_detected) %>%
  summarise(mean_proportion = mean(Proportion, na.rm = TRUE)) %>%
  ungroup()

wild_mean_relative_abundance <- wild_rel %>%
  group_by(Time_point, Time_detected) %>%
  summarise(mean_relative_abundance = mean(Relative_abundance, na.rm = TRUE)) %>%
  ungroup()

# Reverse the levels of Time_detected for proper stacking order
wild_mean_proportion$Time_detected <- factor(wild_mean_proportion$Time_detected, levels = rev(sort(unique(wild_mean_proportion$Time_detected))))
wild_mean_relative_abundance$Time_detected <- factor(wild_mean_relative_abundance$Time_detected, levels = rev(sort(unique(wild_mean_relative_abundance$Time_detected))))

# Plot Mean Proportion of ASVs
plot <- ggplot(wild_mean_proportion, aes(x = Time_point, y = mean_proportion, fill = Time_detected)) +
  theme_minimal() +
  geom_area(position = "fill", size = .2, alpha = .5) +
  scale_y_continuous(lab = scales::percent) + 
  labs(y = "Proportion of ASVs (%)", fill = "ASV first detected at") + 
  scale_fill_manual(values = c("#bfdee0", "#92b6b8","#668f92", "#3a6a6d", "#06474b")) +
  theme(
    axis.ticks = element_line(size = 0, color = "black"),
    axis.ticks.length = unit(0.2, "cm"),
    axis.text.x = element_text(size = 20),
    axis.text.y = element_text(size = 20),
    axis.title.y = element_text(size = 22)
  )
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/Wild_ASV_proportions.png", width=8, height=6)

# Plot Relative Abundance of ASVs
plot <- ggplot(wild_mean_relative_abundance, aes(x = Time_point, y = mean_relative_abundance, fill = factor(Time_detected))) +
  theme_minimal() +
  geom_area(position = "fill", size = .2, alpha = .5) +
  scale_y_continuous(lab = scales::percent) + 
  labs(y = "Relative abundance of ASVs (%)", fill = "ASV first detected at") + 
  scale_fill_manual(values = c("#bfdee0", "#92b6b8","#668f92", "#3a6a6d", "#06474b")) +
  theme(
    axis.ticks = element_line(size = 0, color = "black"),
    axis.ticks.length = unit(0.2, "cm"),
    axis.text.x = element_text(size = 20),
    axis.text.y = element_text(size = 20),
    axis.title.y = element_text(size = 22)
  )
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/Wild_ASV_abundances.png", width=8, height=6)

# How many ASVs are detected in all lab mice in all timepoints?
lab_mice <- subset_samples(phy, Source=="Lab") # 20 samples
lab_mice  <- prune_taxa(taxa_sums(lab_mice) > 0, lab_mice) # 377 taxa
otu_table(lab_mice) <- t(otu_table(lab_mice))
lab_mice <- filter_taxa(lab_mice, function(x) sum(x > 0) > (0.99*length(x)), TRUE) # 35 taxa
taxa <- data.frame(tax_table(lab_mice))
table(taxa$Family)
#  Akkermansiaceae      Bacteroidaceae Erysipelotrichaceae     Lachnospiraceae      Marinifilaceae      Muribaculaceae 
#                  1                   1                   1                   2                   1                  19 
#   Oscillospiraceae               Other       Rikenellaceae      Sutterellaceae      Tannerellaceae 
#                  4                   1                   3                   1                   1 
table(taxa$Genus)
# Akkermansia        Alistipes      Bacteroides Colidextribacter   Faecalibaculum   Intestinimonas      Muribaculum      Odoribacter 
#               1                2                1                2                1                1                1                1 
#           Other  Parabacteroides   Parasutterella        Rikenella 
#              22                1                1                1 
otucheck <- data.frame(otu_table(lab_mice))
summary(colSums(otucheck))
#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.4058  0.5158  0.6228  0.6210  0.7452  0.7849 

# In how many lab samples are these 35 ASVs found (at any abundance)?
taxa <- data.frame(tax_table(lab_mice))
taxa$ASV <- rownames(taxa)
all <- readRDS("IMR_16S_NO_DOUBLETONS.rds")
all <- subset_samples(all, Buffer!="R" & Buffer!="NB" & Buffer!="E" & Buffer!="PBS" & Buffer!="RNAlater")
all <- subset_taxa(all, Genus!="Imtechella")
all <- subset_taxa(all, Genus!="Allobacillus")
all <- microbiome::transform(all, "compositional")
lab <- subset_samples(all, Source=="Lab" & Date=="Kings_college") # 123 samples
lab2 <- prune_taxa(taxa_names(lab) %in% taxa$ASV, lab) # 35 taxa
lab2 <- data.frame(otu_table(lab2))
lab2[lab2>0] <- 1
lab2$sum <- rowSums(lab2)
table(lab2$sum)
# 13 14 15 16 17 20 21 23 24 25 26 27 28 29 30 31 32 33 34 35 
# 3  2  1  3  1  5  1  2 15  9  4 13  5  2  1  2  2 10 10 32 
lab2 <- lab2[which(lab2$sum==35),] # 32 - so in 32/123 (26.0%) lab samples all 35 taxa are detected

# How many ASVs are detected in all wild mice in all timepoints?
wild_mice <- subset_samples(phy, Source=="Wild") # 20 samples
wild_mice  <- prune_taxa(taxa_sums(wild_mice) > 0, wild_mice) # 1536 taxa
otu_table(wild_mice) <- t(otu_table(wild_mice))
wild_mice <- filter_taxa(wild_mice, function(x) sum(x > 0) > (0.99*length(x)), TRUE) # 8 taxa
taxa <- data.frame(tax_table(wild_mice))
table(taxa$Family)
# Lachnospiraceae Lactobacillaceae   Muribaculaceae Oscillospiraceae 
#              1                1                1                5
table(taxa$Genus)
#  Colidextribacter Ligilactobacillus     Oscillibacter             Other 
#              2                 1                 1                 4 
otucheck <- data.frame(otu_table(wild_mice))
summary(colSums(otucheck))
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.02840 0.03822 0.08804 0.09322 0.11959 0.22985

# In how many wild samples are these 35 ASVs found (at any abundance)?
taxa <- data.frame(tax_table(wild_mice))
taxa$ASV <- rownames(taxa)
wild <- subset_samples(all, Project=="Skokholm") # 903 samples
wild2 <- prune_taxa(taxa_names(wild) %in% taxa$ASV, wild) # 8 taxa
wild2 <- data.frame(otu_table(wild2))
wild2[wild2>0] <- 1
wild2$sum <- rowSums(wild2)
table(wild2$sum)
# 0   1   2   3   4   5   6   7   8 
# 1   3   9  10  18  34 125 291 412 
wild2 <- wild2[which(wild2$sum==8),] # 412 - so in 412/903 (45.6%) wild samples all 8 taxa are detected

# Function to create and save plots for each Animal_ID:

create_plots <- function(data, animal_id) {
  # Filter data for the current Animal_ID
  data_filtered <- data %>% filter(Animal_ID == animal_id)
  
  # Determine the source for the current Animal_ID
  source <- unique(data_filtered$Source)
  
  # Gather proportions into long format
  proportion_long <- data_filtered %>%
    pivot_longer(cols = starts_with("Proportion_of_ASVs_first_detected_at_TP"), 
                 names_to = "Time_detected", 
                 names_prefix = "Proportion_of_ASVs_first_detected_at_TP", 
                 values_to = "Proportion")
  
  # Gather relative abundances into long format
  relative_abundance_long <- data_filtered %>%
    pivot_longer(cols = starts_with("Relative_abundance_of_ASVs_first_detected_at_TP"), 
                 names_to = "Time_detected", 
                 names_prefix = "Relative_abundance_of_ASVs_first_detected_at_TP", 
                 values_to = "Relative_abundance")
  
  # Convert Time_detected to a numeric value
  proportion_long$Time_detected <- as.numeric(proportion_long$Time_detected)
  relative_abundance_long$Time_detected <- as.numeric(relative_abundance_long$Time_detected)
  
  # Reverse the levels of Time_detected for proper stacking order
  proportion_long$Time_detected <- factor(proportion_long$Time_detected, levels = rev(sort(unique(proportion_long$Time_detected))))
  relative_abundance_long$Time_detected <- factor(relative_abundance_long$Time_detected, levels = rev(sort(unique(relative_abundance_long$Time_detected))))
  
  # Set color scale based on source
  if (source == "Lab") {
    colors <- c("#fedee2", "#e4bbc0", "#cb989e", "#b1777e", "#97565f")
  } else {
    colors <- c("#bfdee0", "#92b6b8","#668f92", "#3a6a6d", "#06474b")
  }
  
  library(scales)
  
  # Plot Proportion of ASVs
proportion_plot <- ggplot(proportion_long, aes(x = Days_elapsed, y = Proportion, fill = Time_detected)) +
    theme_minimal() +
    geom_area(position = "fill", size = .2, alpha = .5) +
    scale_y_continuous(labels = scales::percent) + 
    labs(title = paste("Animal ID:", animal_id), y = "Proportion of ASVs (%)", fill = "ASV first detected at") + 
    scale_fill_manual(values = colors) +
    scale_x_continuous(breaks = unique(proportion_long$Days_elapsed)) +
    theme(
      axis.ticks = element_line(size = 0, color = "black"),
      axis.ticks.length = unit(0.2, "cm"),
      axis.text.x = element_text(size = 25),
      axis.text.y = element_text(size = 25),
      axis.title.y = element_text(size = 26),
      plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
      legend.position = "none", axis.title.x = element_blank()
    )
  ggsave(filename = paste0("MUSMUS/MUSMUS_figures/Proportion_of_ASVs_Animal_ID_", animal_id, ".png"), plot = proportion_plot, width = 6.5, height = 6)
  print(paste("Saved Proportion plot for Animal_ID:", animal_id))
  
  # Plot Relative Abundance of ASVs
  abundance_plot <- ggplot(relative_abundance_long, aes(x = Days_elapsed, y = Relative_abundance, fill = Time_detected)) +
    theme_minimal() +
    geom_area(position = "fill", size = .2, alpha = .5) +
    scale_y_continuous(labels = scales::percent) + 
    labs(title = paste("Animal ID:", animal_id), y = "Relative abundance of ASVs (%)", fill = "ASV first detected at") + 
    scale_fill_manual(values = colors) +
    scale_x_continuous(breaks = unique(proportion_long$Days_elapsed)) +
    theme(
      axis.ticks = element_line(size = 0, color = "black"),
      axis.ticks.length = unit(0.2, "cm"),
      axis.text.x = element_text(size = 25),
      axis.text.y = element_text(size = 25),
      axis.title.y = element_text(size = 26),
      plot.title = element_text(size = 20, face = "bold", hjust = 0.5),
      legend.position = "none", axis.title.x = element_blank()
    )
  
  ggsave(filename = paste0("MUSMUS/MUSMUS_figures/Relative_abundance_of_ASVs_Animal_ID_", animal_id, ".png"), plot = abundance_plot, width =6.5, height = 6)
  print(paste("Saved Relative abundance plot for Animal_ID:", animal_id))
}

# Get unique Animal_IDs
unique_animal_ids <- unique(results_df$Animal_ID)

# Create and save plots for each Animal_ID
for (animal_id in unique_animal_ids) {
  create_plots(results_df, animal_id)
}

```

***Q9: Do lab and wild mice have different ratios of aerotolerant and anaerobic taxa?***

```{r}

# Extract ligilactobacillus and limosilactobacillus ASVs present in lab/wild cross-sectional and lab/wild longitudinal datasets as the aerotolerance of these is 'mixed' based on genus level taxonomy. Use BLAST to see if get a close match with species level, so can assign aerotolerance.

cross <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds") 
sample_data(phy)$IMR_code <- rownames(sample_data(phy))
# Select animals with >1 sample and those with ID, as done below in longitudinal analysis
sample_data(phy)$datedate <- sample_data(phy)$Collection_date
sample_data(phy)$datedate <- strptime(as.character(sample_data(phy)$datedate), "%d/%m/%Y")
sample_data(phy)$datedate <- format(sample_data(phy)$datedate, "%Y-%m-%d") 
sample_data(phy)$Birthdate <- strptime(as.character(sample_data(phy)$Birthdate), "%d/%m/%Y")
sample_data(phy)$Birthdate <- format(sample_data(phy)$Birthdate, "%Y-%m-%d") 
sample_data(phy)$age <- difftime(sample_data(phy)$datedate, sample_data(phy)$Birthdate, units="days")

mice <- data.frame(sample_data(phy))
mice$Sample_ID <- rownames(mice)
long <- mice %>% group_by(Animal_ID) %>% filter(n()>1) # Select mice sampled a minimum 5 times
unique(long$Animal_ID)
long <- long[which(long$Animal_ID!="NA"),]
long <- long[which(long$Animal_ID!="SK0"),] # Excluded unknown mice
long <- subset_samples(phy, IMR_code%in%long$IMR_code)

ligi1 <- subset_taxa(cross, Genus=="Ligilactobacillus")
ligi1 <- data.frame(colnames(otu_table(ligi1)))
colnames(ligi1) <- c("Ligilactobacillus_ASV")
ligi2 <- subset_taxa(long, Genus=="Ligilactobacillus")
ligi2 <- data.frame(colnames(otu_table(ligi2)))
colnames(ligi2) <- c("Ligilactobacillus_ASV")
ligi <- rbind(ligi1, ligi2)
ligi <- data.frame(ligi[!duplicated(ligi),]) # 127 ASVs

limo1 <- subset_taxa(cross, Genus=="Limosilactobacillus")
limo1 <- data.frame(colnames(otu_table(limo1)))
colnames(limo1) <- c("Limosilactobacillus_ASV")
limo2 <- subset_taxa(long, Genus=="Limosilactobacillus")
limo2 <- data.frame(colnames(otu_table(limo2)))
colnames(limo2) <- c("Limosilactobacillus_ASV")
limo <- rbind(limo1, limo2)
limo <- data.frame(limo[!duplicated(limo),]) # 98 ASVs

ligi$Genus <- "Ligilactobacillus"
limo$Genus <- "Limosilactobacillus"
colnames(ligi) <- c("ASV", "Genus")
colnames(limo) <- c("ASV", "Genus")

lw_sequences <- data.frame(t(ligi[,c("ASV")]))
first_row <- lw_sequences[1, ]
lw_sequences <- as.list(lw_sequences)
library(seqinr)
write.fasta(sequences = lw_sequences, names = lw_sequences, file.out = "MUSMUS/Ligilactobacillus_sequences_MUSMUS.fna") # FASTA header is same as sequence, i.e. using sequences as otu id's.

lw_sequences <- data.frame(t(limo[,c("ASV")]))
first_row <- lw_sequences[1, ]
lw_sequences <- as.list(lw_sequences)
library(seqinr)
write.fasta(sequences = lw_sequences, names = lw_sequences, file.out = "MUSMUS/Limosilactobacillus_sequences_MUSMUS.fna") # FASTA header is same as sequence, i.e. using sequences as otu id's.

```

```{r}

aero <- read.csv("MUSMUS/MUSMUS_aerotolerance_data.csv") 

aero$Aerotolerance_cat[aero$Aerotolerance_cat=="nonaeotolerant"] <- "nonaerotolerant"
aero <- aero[,c("Family", "Genus", "Aerotolerance_cat")]
aero$Family_Genus <- paste(aero$Family, aero$Genus, sep="-") # 698
aero <- aero[!duplicated(aero),] # 698

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)
taxa <- data.frame(tax_table(phy)) 
taxa <- taxa[,c("Family", "Genus")]
taxa <- taxa[!duplicated(taxa),] # 807
taxa$Family_Genus <- paste(taxa$Family, taxa$Genus, sep="-")

taxa <- merge(taxa, aero, by.all="Family_Genus", all.x=TRUE, all.y=FALSE) # 807
table(taxa$Aerotolerance_cat) # 360 aerotolerant, 136 anaerobes
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Unknown"] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat==""] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Mixed"] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="mixed"] <- "unknown"
taxa$unknown <- "unknown"
taxa$Aerotolerance_cat <- ifelse(is.na(taxa$Aerotolerance_cat), taxa$unknown, taxa$Aerotolerance_cat)
table(taxa$Aerotolerance_cat) # 311 unknown
un <- taxa[which(taxa$Aerotolerance_cat=="unknown"),]
length(unique(un$Family)) # 169

phy2 <- phy %>%
  #aggregate_taxa(level = "Genus") %>%  
  microbiome::transform(transform = "compositional")
phy_melt <- psmelt(phy2)
taxa_min <- taxa[,c("Family", "Genus", "Aerotolerance_cat")]
taxa_min$fam_gen <- paste(taxa_min$Family, taxa_min$Genus, sep="-")
phy_melt$fam_gen <- paste(phy_melt$Family, phy_melt$Genus, sep="-") # 498192
phy_phe <- merge(phy_melt, taxa_min, by="fam_gen", all.x=TRUE, all.y=FALSE) # 498192

aero_key <- phy_phe[,c("OTU", "Aerotolerance_cat")]
aero_key <- aero_key[!duplicated(aero_key),]

arr <- phy_phe[,c("Sample", "Aerotolerance_cat", "Abundance")]
arr$ID <- paste(arr$Sample, arr$Aerotolerance_cat, sep="-")
arr <- arr[,c("ID", "Abundance")]

arr2 <- arr %>%
  group_by(ID) %>%
  summarise(total_count = sum(Abundance))

arr2$Sample_ID <- sub("\\-.*", "", arr2$ID)
arr2$Aerotolerance <- sub("[^-]+", '', x = arr2$ID)
arr2$Aerotolerance <- str_sub(arr2$Aerotolerance, start = 2)

source <- phy_phe[,c("Sample", "Source", "Project_ID")]
arr3 <- merge(arr2, source, by.x="Sample_ID", by.y="Sample", all.x=T, all.y=F) # 1494576
arr3 <- arr3[!duplicated(arr3),] # 1746

# Quickly check relative abundance of aerotolerant and anaerobic taxa in lab vs wild
checkAero <- arr3[which(arr3$Aerotolerance=="aerotolerant"),]
checkAeroLab <- checkAero[which(checkAero$Source=="Lab"),]
summary(checkAeroLab$total_count)
#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.02455 0.05804 0.09736 0.11154 0.14233 0.35806
sd(checkAeroLab$total_count) # 0.07108256
checkAeroWild <- checkAero[which(checkAero$Source=="Wild"),]
summary(checkAeroWild$total_count)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.07145 0.23042 0.32259 0.37684 0.45740 0.99927 
sd(checkAeroWild$total_count) # 0.2099403
checkAnaero <- arr3[which(arr3$Aerotolerance=="nonaerotolerant"),]
checkAnaeroLab <- checkAnaero[which(checkAnaero$Source=="Lab"),]
summary(checkAnaeroLab$total_count)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.6183  0.8511  0.8982  0.8788  0.9374  0.9734 
sd(checkAnaeroLab$total_count) # 0.07474462
checkAnaeroWild <- checkAnaero[which(checkAnaero$Source=="Wild"),]
summary(checkAnaeroWild$total_count)
#  Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
# 0.000656 0.495033 0.610519 0.571712 0.696629 0.875493
sd(checkAnaeroWild$total_count) # 0.2003766

# Continue

arr4 <- arr3[,c("Project_ID", "Aerotolerance", "total_count")]
arr4$id <- paste(arr4$Project_ID, arr4$Aerotolerance, sep="*")
arr4 <- arr4[,c("id", "total_count")]
arr5 <- arr4 %>%
  group_by(id) %>%
  summarise(total_count_mean = mean(total_count))
arr5$Project_ID <- sub("\\*.*", "", arr5$id)
arr5$Aerotolerance <- sub("[^*]+", '', x = arr5$id)
arr5$Aerotolerance <- str_sub(arr5$Aerotolerance, start = 2)

arr5$order <- NA
arr5$order[arr5$Project_ID=="BMS-04/12/2020-PdgfraCreERT2"] <- 13
  arr5$order[arr5$Project_ID=="C57BL_6-Kings_college-C57BL/6"] <- 12
  arr5$order[arr5$Project_ID=="BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg"] <- 11
  arr5$order[arr5$Project_ID=="Lab-Kennedy-C57BL/6"] <- 10
  arr5$order[arr5$Project_ID=="Cologne"] <- 7
  arr5$order[arr5$Project_ID=="C57BL_6-BMS_18102021-C57BL/6"] <- 9
  arr5$order[arr5$Project_ID=="Lab-Kennedy-SKG"] <- 8
  arr5$order[arr5$Project_ID=="Wytham"] <- 6
  arr5$order[arr5$Project_ID=="Skokholm"] <- 5
  arr5$order[arr5$Project_ID=="Faroe"] <- 4
  arr5$order[arr5$Project_ID=="Midway_Atoll"] <- 3
  arr5$order[arr5$Project_ID=="Isle_of_May"] <- 2
  arr5$order[arr5$Project_ID=="Espelette"] <- 1

arr5$Aerotolerance <- factor(arr5$Aerotolerance, levels = c("aerotolerant", "nonaerotolerant", "unknown"))

plot <- ggplot(arr5, aes(fill = Aerotolerance, y = total_count_mean, x = reorder(order, order))) + 
  geom_bar(position = "fill", stat = "identity") +
  scale_fill_manual(values = c("aerotolerant" = "#3F6584", "nonaerotolerant" = "#B14D3E", "unknown" = "grey62")) +
  theme_classic() + theme(axis.text.x = element_text(size=20),
        axis.text.y = element_text(size=30),
        axis.title.y = element_text(size=35),
        axis.ticks.x=element_blank(),panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.spacing = unit(2, "lines")) + ylab("Relative abundance")
p <- plot + theme(rect = element_rect(fill = "transparent"), axis.title.x = element_blank(), legend.title = element_blank(), legend.text = element_blank())
ggsave("MUSMUS/MUSMUS_figures/LabWild_aerotolerance.png", width=8.5, height=6)

# Does the ratio between anaerobes and aerobes vary significantly between lab and wild mice?
library(stringr)
arr <- arr3[,c("Sample_ID", "Source", "Aerotolerance", "total_count")]
arr <- spread(arr, Aerotolerance, total_count)
arr$ratio <- arr$aerotolerant/arr$nonaerotolerant

# Add read counts
phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
df <- data.frame(sample_data(phy))
df$Animal_ID <- ifelse(is.na(df$Animal_ID), df$trapdate, df$Animal_ID)
df <- df[,c("IMR_code", "read_count", "Animal_ID")]
arr <- merge(arr, df, by.x="Sample_ID", by.y="IMR_code", all.x=T, all.y=F)

write.csv(arr, "MUSMUS/MUSMUS_aerotolerance_ratio_brm/labwild_ratio/LabWild_aerotolerance_ratio.csv")

hist(arr$ratio)

# Run the below model on super computer

# numerical values to 0-1
scalecols <- c("ratio", "unknown", "read_count")
range.use <- function(x, min.use, max.use) {
  (x - min(x, na.rm = TRUE)) / (max(x, na.rm = TRUE) - min(x, na.rm = TRUE)) * (max.use - min.use) + min.use
}

for (col in scalecols) {
  df[, col] <- range.use(df[, col], 0, 1)
}

samplesize <- nrow(df)
df$ratio <- (df$ratio * (samplesize - 1) + 0.5) / samplesize

#model1 <- brm(ratio ~ 1 + Source + (1 | Animal_ID) + (1 | read_count) + (1 | Unknown), data = df, family = "Beta",control = list(adapt_delta = 0.9, max_treedepth = 12), warmup = 2000, iter = 4000, cores = 1, chains = 1,save_pars = save_pars(group = FALSE), init = 0)

```

```{r}

# Could the higher proportion of aerotolerant taxa in wild mice be explained by (possibly) higher relative abundance of gammaproteobacteria? Repeat above analysis but exclude gammaproteobacteria.

aero <- read.csv("MUSMUS/MUSMUS_aerotolerance_data.csv") 

aero$Aerotolerance_cat[aero$Aerotolerance_cat=="nonaeotolerant"] <- "nonaerotolerant"
aero <- aero[,c("Family", "Genus", "Aerotolerance_cat")]
aero$Family_Genus <- paste(aero$Family, aero$Genus, sep="-") # 698
aero <- aero[!duplicated(aero),] # 698

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS") # 7934 taxa

# Remove Gammaproteobacteria from the phyloseq 
phy <- subset_taxa(phy, Class != "Gammaproteobacteria") # 7641 taxa
phy <- microbiome::transform(phy, "compositional")
  
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)
taxa <- data.frame(tax_table(phy)) 
taxa <- taxa[,c("Family", "Genus")]
taxa <- taxa[!duplicated(taxa),] 
taxa$Family_Genus <- paste(taxa$Family, taxa$Genus, sep="-")

taxa <- merge(taxa, aero, by.all="Family_Genus", all.x=TRUE, all.y=FALSE) 
table(taxa$Aerotolerance_cat)
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Unknown"] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat==""] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Mixed"] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="mixed"] <- "unknown"
taxa$unknown <- "unknown"
taxa$Aerotolerance_cat <- ifelse(is.na(taxa$Aerotolerance_cat), taxa$unknown, taxa$Aerotolerance_cat)
table(taxa$Aerotolerance_cat) 
un <- taxa[which(taxa$Aerotolerance_cat=="unknown"),]
length(unique(un$Family)) 

phy2 <- phy %>%
  #aggregate_taxa(level = "Genus") %>%  
  microbiome::transform(transform = "compositional")
phy_melt <- psmelt(phy2)
taxa_min <- taxa[,c("Family", "Genus", "Aerotolerance_cat")]
taxa_min$fam_gen <- paste(taxa_min$Family, taxa_min$Genus, sep="-")
phy_melt$fam_gen <- paste(phy_melt$Family, phy_melt$Genus, sep="-") # 498192
phy_phe <- merge(phy_melt, taxa_min, by="fam_gen", all.x=TRUE, all.y=FALSE) # 498192

aero_key <- phy_phe[,c("OTU", "Aerotolerance_cat")]
aero_key <- aero_key[!duplicated(aero_key),]

arr <- phy_phe[,c("Sample", "Aerotolerance_cat", "Abundance")]
arr$ID <- paste(arr$Sample, arr$Aerotolerance_cat, sep="-")
arr <- arr[,c("ID", "Abundance")]

arr2 <- arr %>%
  group_by(ID) %>%
  summarise(total_count = sum(Abundance))

arr2$Sample_ID <- sub("\\-.*", "", arr2$ID)
arr2$Aerotolerance <- sub("[^-]+", '', x = arr2$ID)
arr2$Aerotolerance <- str_sub(arr2$Aerotolerance, start = 2)

source <- phy_phe[,c("Sample", "Source", "Project_ID")]
arr3 <- merge(arr2, source, by.x="Sample_ID", by.y="Sample", all.x=T, all.y=F) # 1494576
arr3 <- arr3[!duplicated(arr3),] # 1746

# Quickly check relative abundance of aerotolerant and anaerobic taxa in lab vs wild
checkAero <- arr3[which(arr3$Aerotolerance=="aerotolerant"),]
checkAeroLab <- checkAero[which(checkAero$Source=="Lab"),]
summary(checkAeroLab$total_count)
#   Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.02466 0.05835 0.09783 0.11198 0.14330 0.35828
sd(checkAeroLab$total_count) # 0.07133448
checkAeroWild <- checkAero[which(checkAero$Source=="Wild"),]
summary(checkAeroWild$total_count)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
# 0.06175 0.22301 0.30670 0.36496 0.42799 0.99927 
sd(checkAeroWild$total_count) # 0.2049814
checkAnaero <- arr3[which(arr3$Aerotolerance=="nonaerotolerant"),]
checkAnaeroLab <- checkAnaero[which(checkAnaero$Source=="Lab"),]
summary(checkAnaeroLab$total_count)
# Min. 1st Qu.  Median    Mean 3rd Qu.    Max. 
#  0.6180  0.8499  0.8977  0.8783  0.9372  0.9733  
sd(checkAnaeroLab$total_count) # 0.07496234
checkAnaeroWild <- checkAnaero[which(checkAnaero$Source=="Wild"),]
summary(checkAnaeroWild$total_count)
#  Min.  1st Qu.   Median     Mean  3rd Qu.     Max. 
# 0.0006598 0.5144387 0.6219397 0.5845853 0.7107845 0.8864183
sd(checkAnaeroWild$total_count) # 0.1961931

# Continue

arr4 <- arr3[,c("Project_ID", "Aerotolerance", "total_count")]
arr4$id <- paste(arr4$Project_ID, arr4$Aerotolerance, sep="*")
arr4 <- arr4[,c("id", "total_count")]
arr5 <- arr4 %>%
  group_by(id) %>%
  summarise(total_count_mean = mean(total_count))
arr5$Project_ID <- sub("\\*.*", "", arr5$id)
arr5$Aerotolerance <- sub("[^*]+", '', x = arr5$id)
arr5$Aerotolerance <- str_sub(arr5$Aerotolerance, start = 2)

arr5$order <- NA
arr5$order[arr5$Project_ID=="BMS-04/12/2020-PdgfraCreERT2"] <- 13
  arr5$order[arr5$Project_ID=="C57BL_6-Kings_college-C57BL/6"] <- 12
  arr5$order[arr5$Project_ID=="BMS-04/12/2020-CCSP_rtTAtg/(tetO)75FTPBtg"] <- 11
  arr5$order[arr5$Project_ID=="Lab-Kennedy-C57BL/6"] <- 10
  arr5$order[arr5$Project_ID=="Cologne"] <- 7
  arr5$order[arr5$Project_ID=="C57BL_6-BMS_18102021-C57BL/6"] <- 9
  arr5$order[arr5$Project_ID=="Lab-Kennedy-SKG"] <- 8
  arr5$order[arr5$Project_ID=="Wytham"] <- 6
  arr5$order[arr5$Project_ID=="Skokholm"] <- 5
  arr5$order[arr5$Project_ID=="Faroe"] <- 4
  arr5$order[arr5$Project_ID=="Midway_Atoll"] <- 3
  arr5$order[arr5$Project_ID=="Isle_of_May"] <- 2
  arr5$order[arr5$Project_ID=="Espelette"] <- 1

arr5$Aerotolerance <- factor(arr5$Aerotolerance, levels = c("aerotolerant", "nonaerotolerant", "unknown"))

plot <- ggplot(arr5, aes(fill = Aerotolerance, y = total_count_mean, x = reorder(order, order))) + 
  geom_bar(position = "fill", stat = "identity") +
  scale_fill_manual(values = c("aerotolerant" = "#3F6584", "nonaerotolerant" = "#B14D3E", "unknown" = "grey62")) +
  theme_classic() + theme(axis.text.x = element_text(size=20),
        axis.text.y = element_text(size=30),
        axis.title.y = element_text(size=35),
        axis.ticks.x=element_blank(),panel.grid.major = element_blank(), panel.grid.minor = element_blank(), panel.spacing = unit(2, "lines")) + ylab("Relative abundance")
p <- plot + theme(rect = element_rect(fill = "transparent"), axis.title.x = element_blank(), legend.title = element_blank(), legend.text = element_blank())
ggsave("MUSMUS/MUSMUS_figures/LabWild_aerotolerance_wOut_Gammaproteobacteria.png", width=8.5, height=6)

```


***Q10: Are anaerobic and aerobic communities equally variable across individuals?***

```{r}

aero <- read.csv("MUSMUS/MUSMUS_aerotolerance_data.csv")
aero$Aerotolerance_cat[aero$Aerotolerance_cat=="nonaeotolerant"] <- "nonaerotolerant"
aero <- aero[,c("Family", "Genus", "Aerotolerance_cat")]
aero$Family_Genus <- paste(aero$Family, aero$Genus, sep="-") # 698
aero <- aero[!duplicated(aero),] # 698

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")

wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)

taxa <- data.frame(tax_table(phy)) 
taxa <- taxa[,c("Family", "Genus")]
taxa <- taxa[!duplicated(taxa),] # 807
taxa$Family_Genus <- paste(taxa$Family, taxa$Genus, sep="-")

taxa <- merge(taxa, aero, by.all="Family_Genus", all.x=TRUE, all.y=FALSE) # 807
table(taxa$Aerotolerance_cat) # 360 aerotolerant, 136 anaerobes
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Unknown"] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat==""] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Mixed"] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="mixed"] <- "unknown"
taxa$unknown <- "unknown"
taxa$Aerotolerance_cat <- ifelse(is.na(taxa$Aerotolerance_cat), taxa$unknown, taxa$Aerotolerance_cat)
table(taxa$Aerotolerance_cat) # 311 unknown
un <- taxa[which(taxa$Aerotolerance_cat=="unknown"),]
length(unique(un$Family)) # 169

phy2 <- phy %>%
  #aggregate_taxa(level = "Genus") %>%  
  microbiome::transform(transform = "compositional")
phy_melt <- psmelt(phy2)
taxa_min <- taxa[,c("Family", "Genus", "Aerotolerance_cat")]
taxa_min$fam_gen <- paste(taxa_min$Family, taxa_min$Genus, sep="-")
phy_melt$fam_gen <- paste(phy_melt$Family, phy_melt$Genus, sep="-") # 498192
phy_phe <- merge(phy_melt, taxa_min, by="fam_gen", all.x=TRUE, all.y=FALSE) # 498192

# Count unique ASVs per aerotolerance category
asv <- phy_phe[,c("Aerotolerance_cat", "OTU", "Sample", "Project_ID", "Source", "Abundance")]
asv <- asv[which(asv$Abundance!=0),]
asv <- asv %>%
  group_by(Aerotolerance_cat, Sample) %>%
  mutate(ASV_count = n()) %>%
  ungroup()
asv <- asv[,c("Sample", "Project_ID", "Source", "Aerotolerance_cat", "ASV_count")] # 131418
asv <- asv[!duplicated(asv),] # 1746 (3 rows per sample since 3 levels of aerotolerance)
asv <- asv[which(asv$Aerotolerance_cat!="unknown"),] # 1164

# Save aerotolerant and anaerotolerant richness measures, to be tested in a later chunk that looks at influence of exposure time
save_aero <- asv[which(asv$Aerotolerance_cat=="aerotolerant"),]
df <- data.frame(sample_data(readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")))
df$Animal_ID <- ifelse(is.na(df$Animal_ID), df$trapdate, df$Animal_ID)
df <- df[,c("IMR_code", "read_count", "Animal_ID")]
save_aero <- merge(save_aero, df, by.x="Sample", by.y="IMR_code", all.x=T, all.y=F)
write.csv(save_aero, "MUSMUS/MUSMUS_aerotolerance_ratio_brm/exposure_time/exposure_time_richness/aerotolerant_richness.csv")

save_aero <- asv[which(asv$Aerotolerance_cat=="nonaerotolerant"),]
df <- data.frame(sample_data(readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")))
df$Animal_ID <- ifelse(is.na(df$Animal_ID), df$trapdate, df$Animal_ID)
df <- df[,c("IMR_code", "read_count", "Animal_ID")]
save_aero <- merge(save_aero, df, by.x="Sample", by.y="IMR_code", all.x=T, all.y=F)
write.csv(save_aero, "MUSMUS/MUSMUS_aerotolerance_ratio_brm/exposure_time/exposure_time_richness/anaerobic_richness.csv")

# Wilcoxon rank sum tests to see if levels are significantly different

# Test for difference
lab <- asv[which(asv$Source=="Lab"),]
wild <- asv[which(asv$Source=="Wild"),]

u.obs<-wilcox.test(ASV_count ~ Aerotolerance_cat, data=lab)
u.obs # W = 2, p-value < 2.2e-16
u.obs<-wilcox.test(ASV_count ~ Aerotolerance_cat, data=wild)
u.obs # W = 2021.5, p-value < 2.2e-16

aero <- asv[which(asv$Aerotolerance_cat=="aerotolerant"),]
anaero <- asv[which(asv$Aerotolerance_cat=="nonaerotolerant"),]

u.obs<-wilcox.test(ASV_count ~ Source, data=aero)
u.obs # W = 1053.5, p-value < 2.2e-16
u.obs<-wilcox.test(ASV_count ~ Source, data=anaero)
u.obs # W = 6806, p-value = 7.179e-14

# Plot
plot <- ggplot(asv, aes(x = Source, y = ASV_count, fill = Aerotolerance_cat)) + 
  ggdist::stat_halfeye(
    aes(color = Aerotolerance_cat),  # Map color to id for the points
    adjust = .5, 
    width = .6, 
    alpha = .8,
    .width = 0, 
    justification = -.3, 
    point_colour = NA
  ) +
  geom_boxplot(
    aes(fill = Aerotolerance_cat),  # Map fill to id for the boxplot
    width = .25, 
    alpha = .8,
    outlier.shape = NA, outlier.alpha = 0
  ) +
  scale_fill_manual(values = c("aerotolerant" = "steelblue", "nonaerotolerant" = "indianred")) +
  scale_color_manual(values = c("aerotolerant" = "steelblue", "nonaerotolerant" = "indianred")) + theme_classic() +
  theme(axis.text.y = element_text(size=30), axis.title.y = element_text(size=35), axis.text.x = element_text(size=25)) + scale_y_continuous(limits = c(0, 100), breaks = c(0,20,40,60,80,100))  + ylab("ASV richness")
p <- plot + theme(rect = element_rect(fill = "transparent"), axis.title.x = element_blank(), legend.title = element_blank(), legend.text = element_blank())
ggsave("MUSMUS/MUSMUS_figures/LabWild_aerotolerance_ASVcount.png", width=5.5, height=6)

# Continue with plotting beta diversity

aero_key <- phy_phe[,c("OTU", "Aerotolerance_cat")]
aero_key <- aero_key[!duplicated(aero_key),]

library(microbiome)
library(ggplot2)
library(dplyr)
library(vegan)

time <- phy

time # 12138 taxa 
aerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="aerotolerant"),]
nonaerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="nonaerotolerant"),]
otu_table(time) <- t(otu_table(time))
aerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% aerotolerant_key$OTU)
aerotolerant <- merge_phyloseq(aerotolerant, tax_table(time), sample_data(time))
aerotolerant
nonaerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% nonaerotolerant_key$OTU)
nonaerotolerant <- merge_phyloseq(nonaerotolerant, tax_table(time), sample_data(time))
nonaerotolerant 

# aerotolerant:

D <- phyloseq::distance(aerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Sample_ID <- rownames(sd)
sd <- sd[,c("Project_ID", "Sample_ID")]
get <- merge(D_melt, sd, by.x="Var1", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var1", "Var2", "Distance", "Source1")
get <- merge(get, sd, by.x="Var2", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var2", "Var1", "Distance", "Source1", "Source2")
get <- get[which(get$Var2!=get$Var1),] # remove self comps

get$aerotolerance <- "Aerotolerant"
save <- get

# nonaerotolerant

D <- phyloseq::distance(time, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Sample_ID <- rownames(sd)
sd <- sd[,c("Project_ID", "Sample_ID")]
get <- merge(D_melt, sd, by.x="Var1", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var1", "Var2", "Distance", "Source1")
get <- merge(get, sd, by.x="Var2", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var2", "Var1", "Distance", "Source1", "Source2")
get <- get[which(get$Var2!=get$Var1),] # remove self comps

get$aerotolerance <- "Anaerobic"
get <- rbind(get, save)

get2 <- get[which(get$Source1==get$Source2),]
get2$mouse <- "mouse"
unique(get2$Source1)
get2$source <- "Lab"
get2$source[get2$Source1=="Skokholm" | get2$Source1=="Espelette" | get2$Source1=="Faroe" | get2$Source1=="Midway_Atoll" | get2$Source1=="Wytham" | get2$Source1=="Cologne" | get2$Source1=="Isle_of_May"] <- "Wild"
get2$id <- paste(get2$source, get2$aerotolerance, sep="-")
get2$id2 <- paste(get2$Source1, get2$aerotolerance, sep="-")

aerotolerance_across <- get2
aerotolerance_across$pair <- paste(aerotolerance_across$Var2, aerotolerance_across$Var1, sep="-")

aerotolerance_across$Source_num[aerotolerance_across$source=="Lab"] <- 0
aerotolerance_across$Source_num[aerotolerance_across$source=="Wild"] <- 1

#write.csv(aerotolerance_across, "/Users/user/Documents/Science/DPhil/Data/IMR/IMR_phyloseq/IMR_analysis/GM_drivers/GM_drivers_aerotolerance/aerotolerance_across.csv")

# plot beta diversity

get3 <- get2

# Test for difference
lab <- get3[which(get3$source=="Lab"),]
wild <- get3[which(get3$source=="Wild"),]

# lab
u.obs<-wilcox.test(Distance ~ aerotolerance, data=lab)
u.obs # W = 82571810, p-value = 0.004291
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"Distance"
for (i in 1:Nperm) {
  d<-lab[,!names(lab) %in% lose]
  d$dist.rand<-sample(lab$Distance, length(lab$Distance), replace=F)
  x<-wilcox.test(dist.rand ~ aerotolerance, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat>u.obs$statistic, 1, 0)
p<-mean(perm1$result)
p*2 # 0.006

# wild
u.obs<-wilcox.test(Distance ~ aerotolerance, data=wild)
u.obs # W = 47271230, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"Distance"
for (i in 1:Nperm) {
  d<-wild[,!names(wild) %in% lose]
  d$dist.rand<-sample(wild$Distance, length(wild$Distance), replace=F)
  x<-wilcox.test(dist.rand ~ aerotolerance, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat<u.obs$statistic, 1, 0)
p<-mean(perm1$result)
p*2 # 0

# lab vs wild
aero <- get3[which(get3$aerotolerance=="Aerotolerant"),]
anaero <- get3[which(get3$aerotolerance=="Anaerobic"),]

# aerotolerant
u.obs<-wilcox.test(Distance ~ source, data=aero)
u.obs # W = 15620684, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"Distance"
for (i in 1:Nperm) {
  d<-aero[,!names(aero) %in% lose]
  d$dist.rand<-sample(aero$Distance, length(aero$Distance), replace=F)
  x<-wilcox.test(dist.rand ~ source, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat<u.obs$statistic, 1, 0)
p<-mean(perm1$result)
p*2 # 0

# nonaerotolerant
u.obs<-wilcox.test(Distance ~ source, data=anaero)
u.obs # W = 5063894, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"Distance"
for (i in 1:Nperm) {
  d<-anaero[,!names(anaero) %in% lose]
  d$dist.rand<-sample(anaero$Distance, length(anaero$Distance), replace=F)
  x<-wilcox.test(dist.rand ~ source, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat<u.obs$statistic, 1, 0)
p<-mean(perm1$result)
p*2 # 0

# Plotting

plot <- ggplot(get3, aes(x = source, y = Distance, fill = id)) + 
  ggdist::stat_halfeye(
    aes(color = id),  # Map color to id for the points
    adjust = .5, 
    width = .6, 
    alpha = .8,
    .width = 0, 
    justification = -.3, 
    point_colour = NA
  ) +
  geom_boxplot(
    aes(fill = id),  # Map fill to id for the boxplot
    width = .25, 
    alpha = .8,
    outlier.shape = NA, outlier.alpha = 0
  ) +
  scale_fill_manual(values = c("Lab-Aerotolerant" = "steelblue", "Lab-Anaerobic" = "indianred", "Wild-Aerotolerant" = "steelblue", "Wild-Anaerobic" = "indianred")) +
  scale_color_manual(values = c("Lab-Aerotolerant" = "steelblue", "Lab-anaerobic" = "indianred", "Wild-Aerotolerant" = "steelblue", "Wild-Anaerobic" = "indianred")) + theme_classic() +
  theme(axis.text.y = element_text(size=30), axis.title.y = element_text(size=25), axis.text.x = element_text(size=25)) + ylab("Jaccard distance across individuals")
p <- plot + theme(rect = element_rect(fill = "transparent"), axis.title.x = element_blank(), legend.title = element_blank(), legend.text = element_blank())
ggsave("MUSMUS/MUSMUS_figures/LabWild_aerotolerance_acrossIndividualsJaccard.png", width=5.5, height=6)

```

***Q11: Are anaerobic and aerobic communities equally variable within individuals?***

```{r}

aero <- read.csv("MUSMUS/MUSMUS_aerotolerance_data.csv")
aero$Aerotolerance_cat[aero$Aerotolerance_cat=="nonaeotolerant"] <- "nonaerotolerant"
aero <- aero[,c("Family", "Genus", "Aerotolerance_cat")]
aero$Family_Genus <- paste(aero$Family, aero$Genus, sep="-") # 698
aero <- aero[!duplicated(aero),] # 698

phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds") # select all longitudinally sampled lab and wild mice

# Select animals with >1 sample 
sample_data(phy)$datedate <- sample_data(phy)$Collection_date
sample_data(phy)$datedate <- strptime(as.character(sample_data(phy)$datedate), "%d/%m/%Y")
sample_data(phy)$datedate <- format(sample_data(phy)$datedate, "%Y-%m-%d") 
sample_data(phy)$Birthdate <- strptime(as.character(sample_data(phy)$Birthdate), "%d/%m/%Y")
sample_data(phy)$Birthdate <- format(sample_data(phy)$Birthdate, "%Y-%m-%d") 
sample_data(phy)$age <- difftime(sample_data(phy)$datedate, sample_data(phy)$Birthdate, units="days")

mice <- data.frame(sample_data(phy))
mice$Sample_ID <- rownames(mice)
long <- mice %>% group_by(Animal_ID) %>% filter(n()>1) # Select mice sampled a minimum 5 times
unique(long$Animal_ID)
long <- long[which(long$Animal_ID!="NA"),]
long <- long[which(long$Animal_ID!="SK0"),] # Excluded unknown mice
table(long$Source) # 98 lab (King's college, C57BL/6), 762 wild (Skokholm)

time <- phy
sample_data(time)$Sample_ID <- rownames(sample_data(time))
sample_data(time)$include[sample_data(time)$Sample_ID%in%long$Sample_ID] <- "yes"
time <- subset_samples(time, include=="yes") # 861 samples
phy <- time

taxa <- data.frame(tax_table(phy)) 
taxa <- taxa[,c("Family", "Genus")]
taxa <- taxa[!duplicated(taxa),] # 805
taxa$Family_Genus <- paste(taxa$Family, taxa$Genus, sep="-")

taxa <- merge(taxa, aero, by.all="Family_Genus", all.x=TRUE, all.y=FALSE) # 
table(taxa$Aerotolerance_cat) #
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Unknown"] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat==""] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Mixed"] <- "unknown"
taxa$unknown <- "unknown"
taxa$Aerotolerance_cat <- ifelse(is.na(taxa$Aerotolerance_cat), taxa$unknown, taxa$Aerotolerance_cat)
table(taxa$Aerotolerance_cat) # 310 unknown
un <- taxa[which(taxa$Aerotolerance_cat=="unknown"),]
length(unique(un$Family)) # 169

phy2 <- phy %>%
  #aggregate_taxa(level = "Genus") %>%  
  microbiome::transform(transform = "compositional")
phy_melt <- psmelt(phy2)
taxa_min <- taxa[,c("Family", "Genus", "Aerotolerance_cat")]
taxa_min$fam_gen <- paste(taxa_min$Family, taxa_min$Genus, sep="-")
phy_melt$fam_gen <- paste(phy_melt$Family, phy_melt$Genus, sep="-") # 
phy_phe <- merge(phy_melt, taxa_min, by="fam_gen", all.x=TRUE, all.y=FALSE) # 

aero_key <- phy_phe[,c("OTU", "Aerotolerance_cat")]
aero_key <- aero_key[!duplicated(aero_key),]

library(microbiome)
library(ggplot2)
library(dplyr)
library(vegan)

time <- phy

time # 
aerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="aerotolerant"),]
nonaerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="nonaerotolerant"),]
otu_table(time) <- t(otu_table(time))
aerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% aerotolerant_key$OTU)
aerotolerant <- merge_phyloseq(aerotolerant, tax_table(time), sample_data(time))
aerotolerant # 3123 taxa and 340 samples
nonaerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% nonaerotolerant_key$OTU)
nonaerotolerant <- merge_phyloseq(nonaerotolerant, tax_table(time), sample_data(time))
nonaerotolerant # 6186 taxa and 340 samples 

# aerotolerant:

D <- phyloseq::distance(aerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Sample_ID <- rownames(sd)
sd <- sd[,c("Project", "Sample_ID", "Animal_ID", "Collection_date")]
sd$id_date <- paste(sd$Animal_ID, sd$Collection_date, sep="-")
sd <- sd[,c("Project", "Sample_ID", "Animal_ID", "id_date")]
get <- merge(D_melt, sd, by.x="Var1", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var1", "Var2", "Distance", "Source1", "id1", "id_date1")
get <- merge(get, sd, by.x="Var2", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var2", "Var1", "Distance", "Source1", "id1", "id_date1", "Source2", "id2", "id_date2")
get <- get[which(get$Var2!=get$Var1),] # remove same sample comps
get <- get[which(get$id1==get$id2),] # only keep self comparisons

get$aerotolerance <- "Aerotolerant"
save <- get

# nonaerotolerant

D <- phyloseq::distance(nonaerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Sample_ID <- rownames(sd)
sd <- sd[,c("Project", "Sample_ID", "Animal_ID", "Collection_date")]
sd$id_date <- paste(sd$Animal_ID, sd$Collection_date, sep="-")
sd <- sd[,c("Project", "Sample_ID", "Animal_ID", "id_date")]
get <- merge(D_melt, sd, by.x="Var1", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var1", "Var2", "Distance", "Source1", "id1", "id_date1")
get <- merge(get, sd, by.x="Var2", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var2", "Var1", "Distance", "Source1", "id1", "id_date1", "Source2", "id2", "id_date2")
get <- get[which(get$Var2!=get$Var1),] # remove same sample comps
get <- get[which(get$id1==get$id2),] # only keep self comparisons

get$aerotolerance <- "Anaerobic"
get <- rbind(get, save)

get2 <- get[which(get$Source1==get$Source2),]
get2$mouse <- "mouse"
unique(get2$Source1)
get2$source <- "Lab"
get2$source[get2$Source1=="Skokholm" | get2$Source1=="Espelette" | get2$Source1=="Faroe" | get2$Source1=="Midway_Atoll" | get2$Source1=="Wytham" | get2$Source1=="Cologne" | get2$Source1=="Isle_of_May"] <- "Wild"
get2$id <- paste(get2$source, get2$aerotolerance, sep="-")
get2$id2 <- paste(get2$Source1, get2$aerotolerance, sep="-")

# plot

table(get2$Source1)
# C57BL_6 Skokholm 
#   1044     5720 

get3 <- get2

# Test for difference
lab <- get3[which(get3$source=="Lab"),]
wild <- get3[which(get3$source=="Wild"),]

# lab
u.obs<-wilcox.test(Distance ~ aerotolerance, data=lab)
u.obs # W = 151334, p-value = 0.001943
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"Distance"
for (i in 1:Nperm) {
  d<-lab[,!names(lab) %in% lose]
  d$dist.rand<-sample(lab$Distance, length(lab$Distance), replace=F)
  x<-wilcox.test(dist.rand ~ aerotolerance, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat>u.obs$statistic, 1, 0)
p<-mean(perm1$result)
p*2 # 0.002

# wild
u.obs<-wilcox.test(Distance ~ aerotolerance, data=wild)
u.obs # W = 3474726, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"Distance"
for (i in 1:Nperm) {
  d<-wild[,!names(wild) %in% lose]
  d$dist.rand<-sample(wild$Distance, length(wild$Distance), replace=F)
  x<-wilcox.test(dist.rand ~ aerotolerance, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat<u.obs$statistic, 1, 0)
p<-mean(perm1$result)
p*2 # 0

# lab vs wild
aero <- get3[which(get3$aerotolerance=="Aerotolerant"),]
anaero <- get3[which(get3$aerotolerance=="Anaerobic"),]

# aerotolerant
u.obs<-wilcox.test(Distance ~ source, data=aero)
u.obs # W = 270474, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"Distance"
for (i in 1:Nperm) {
  d<-aero[,!names(aero) %in% lose]
  d$dist.rand<-sample(aero$Distance, length(aero$Distance), replace=F)
  x<-wilcox.test(dist.rand ~ source, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat<u.obs$statistic, 1, 0)
p<-mean(perm1$result)
p*2 # 0

# nonaerotolerant
u.obs<-wilcox.test(Distance ~ source, data=anaero)
u.obs # W = 172420, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"Distance"
for (i in 1:Nperm) {
  d<-anaero[,!names(anaero) %in% lose]
  d$dist.rand<-sample(anaero$Distance, length(anaero$Distance), replace=F)
  x<-wilcox.test(dist.rand ~ source, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat<u.obs$statistic, 1, 0)
p<-mean(perm1$result)
p*2 # 0

library(ggdist)
plot <- ggplot(get3, aes(x = source, y = Distance, fill = id)) + 
  ggdist::stat_halfeye(
    aes(color = id),  # Map color to id for the points
    adjust = .5, 
    width = .6, 
    alpha = .8,
    .width = 0, 
    justification = -.3, 
    point_colour = NA
  ) +
  geom_boxplot(
    aes(fill = id),  # Map fill to id for the boxplot
    width = .25, 
    alpha = .8,
    outlier.shape = NA, outlier.alpha = 0
  ) +
  scale_fill_manual(values = c("Lab-Aerotolerant" = "steelblue", "Lab-Anaerobic" = "indianred", "Wild-Aerotolerant" = "steelblue", "Wild-Anaerobic" = "indianred")) +
  scale_color_manual(values = c("Lab-Aerotolerant" = "steelblue", "Lab-anaerobic" = "indianred", "Wild-Aerotolerant" = "steelblue", "Wild-Anaerobic" = "indianred")) + theme_classic() +
  theme(axis.text.y = element_text(size=30), axis.title.y = element_text(size=25), axis.text.x = element_text(size=25)) + ylab("Jaccard distance within individuals")
p <- plot + theme(rect = element_rect(fill = "transparent"), axis.title.x = element_blank(), legend.title = element_blank(), legend.text = element_blank())
ggsave("MUSMUS/MUSMUS_figures/LabWild_aerotolerance_withinIndividualsJaccard.png", width=5.5, height=6)

```

***Q12: Do the anaerobic and aerobic communities change at different rates?***

```{r}

aero <- read.csv("MUSMUS/MUSMUS_aerotolerance_data.csv")
aero$Aerotolerance_cat[aero$Aerotolerance_cat=="nonaeotolerant"] <- "nonaerotolerant"
aero <- aero[,c("Family", "Genus", "Aerotolerance_cat")]
aero$Family_Genus <- paste(aero$Family, aero$Genus, sep="-") # 698
aero <- aero[!duplicated(aero),] # 698

# Is microbial turnover rate depended on bacterial aerotolerance?
phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds") # select all longitudinally sampled lab and wild mice
phy <- subset_samples(phy, Buffer!="R" & Buffer!="NB" & Buffer!="E" & Buffer!="PBS" & Buffer!="RNAlater")

phy <- subset_taxa(phy, Genus!="Imtechella")
phy <- subset_taxa(phy, Genus!="Allobacillus")
phy <- microbiome::transform(phy, "compositional")

library(microbiome)
library(ggplot2)
library(dplyr)
library(vegan)

sample_data(phy)$datedate <- sample_data(phy)$Collection_date
sample_data(phy)$datedate <- strptime(as.character(sample_data(phy)$datedate), "%d/%m/%Y")
sample_data(phy)$datedate <- format(sample_data(phy)$datedate, "%Y-%m-%d") 
sample_data(phy)$Birthdate <- strptime(as.character(sample_data(phy)$Birthdate), "%d/%m/%Y")
sample_data(phy)$Birthdate <- format(sample_data(phy)$Birthdate, "%Y-%m-%d") 
sample_data(phy)$age <- difftime(sample_data(phy)$datedate, sample_data(phy)$Birthdate, units="days")

mice <- data.frame(sample_data(phy))
mice$Sample_ID <- rownames(mice)
long <- mice %>% group_by(Animal_ID) %>% filter(n()>1) # Select mice sampled a minimum 2 times
unique(long$Animal_ID)
long <- long[which(long$Animal_ID!="NA"),]
long <- long[which(long$Animal_ID!="SK0"),] # Excluded unknown mice
table(long$Source) # 99 lab (King's college, C57BL/6), 717 wild (Skokholm)

check_l <- long[which(long$Source=="Lab"),]
table(check_l$Animal_ID) # 5-7 samples each
length(unique(check_l$Animal_ID)) # 16

check_l <- long[which(long$Source=="Wild"),]
table(check_l$Animal_ID) # 2-12 samples each
length(unique(check_l$Animal_ID)) # 217

time <- phy
sample_data(time)$Sample_ID <- rownames(sample_data(time))
sample_data(time)$include[sample_data(time)$Sample_ID%in%long$Sample_ID] <- "yes"
time <- subset_samples(time, include=="yes")

aero_key <- phy_phe[,c("OTU", "Aerotolerance_cat")] # need to run above 2 chunks
aero_key <- aero_key[!duplicated(aero_key),]

time # 12138 taxa 
time  <- prune_taxa(taxa_sums(time) > 0, time) # 4361 taxa
aerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="aerotolerant"),]
nonaerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="nonaerotolerant"),]
otu_table(time) <- t(otu_table(time))
aerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% aerotolerant_key$OTU)
aerotolerant <- merge_phyloseq(aerotolerant, tax_table(time), sample_data(time))
aerotolerant # 898 taxa
nonaerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% nonaerotolerant_key$OTU)
nonaerotolerant <- merge_phyloseq(nonaerotolerant, tax_table(time), sample_data(time))
nonaerotolerant # 2537 taxa

# aerotolerant:

D <- phyloseq::distance(aerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Mouse <- rownames(sd)
sd$newdate <- strptime(as.character(sd$Collection_date), "%d/%m/%Y") # here use current format
sd$Date <- format(sd$newdate, "%Y-%m-%d") 
sd <- sd[order(sd$Animal_ID, sd$Date, decreasing = F),]
sd$Time_point <- as.numeric(unlist(lapply(split(sd, sd$Animal_ID), function(x) { rank(x$Date) })))
sd <- data.frame(sd[,c("Source", "Mouse", "Animal_ID", "Date", "Time_point")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp", "mouse2_source", "mouse2_id", "mouse2_date", "mouse2_tp")
distance <- na.omit(distance)
self <- distance[which(distance$mouse1_id==distance$mouse2_id),] # only self comparisons
first_tp <- sd[which(sd$Time_point==1),]
first_tp <- first_tp[,c("Animal_ID", "Date")]
self <- merge(self, first_tp, by.x="mouse1_id", by.y="Animal_ID", all.x=FALSE, all.y=FALSE)
self$added_days <- difftime(self$mouse2_date, self$Date, units="days") # counting how many days added to first sample
self <- self[which(self$added_days>0),]
self$added_days_step <- difftime(self$mouse2_date, self$mouse1_date, units="days")
self <- self[which(self$added_days_step>0),]
#self <- self[which(self$added_days_step>=6),]
self$tp <- paste(self$mouse1_tp, self$mouse2_tp, sep="-")
#self <- self[which(self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="2-3" | self$tp=="3-4" | self$tp=="4-5" | self$tp=="5-6" | self$tp=="6-7" | self$tp=="7-8" | self$tp=="8-9" | self$tp=="9-10"),]
self <- self[which(self$mouse1_tp!=self$mouse2_tp),]
self$added_days <- as.numeric(self$added_days)
self$added_days_step <- as.numeric(self$added_days_step)

wild <- self[which(self$mouse1_source=="Wild"),]
lab <- self[which(self$mouse1_source=="Lab"),]

library(easynls)
library(nlshelper)

# Wild:
wildX <- wild
wildX$aitch_sim <- log(wildX$BC_distance)
wildY <- wildX[, c("added_days_step", "aitch_sim")]
nlsplot (wildY, model=4,xlab="",ylab="") 
nlsfit(wildY, model=4) # adj r2=0.30160000, AIC -622.80774806
wildY <- wildX[, c("added_days_step", "BC_distance")]
nlsplot (wildY, model=4,xlab="",ylab="") 
nlsfit(wildY, model=4) # adj r2=3.110000e-01, AIC -1.993288e+03 --> non-log transformed gives a better fit
nlsfit(wildY, model=1) # AIC -1.844592e+03 --> delta AIC 148.696
dta.nls <- nls(BC_distance ~ (a + b * added_days_step + c * I(added_days_step^2)) * (added_days_step <= -0.5 * b/c) + (a + I(-b^2/(4 * c))) * (added_days_step > -0.5 * b/c), wildY, start=list(a=0.4962564,b=0.01089391,c=as.numeric(-0.0001488900)))
wildY$line <- predict(dta.nls)
plot <- ggplot(wildY, aes(x=added_days_step, y=BC_distance)) +
  annotate("segment", x = 36.58329, 
           xend = 36.58329, y = 0, 
           yend = 1,
           colour = "steelblue4",size=1,linetype = "dashed")+
  geom_point(size=2, colour = "steelblue",alpha=.1) +
  xlab("Days elapsed") + ylab("Jaccard distance") + 
  theme_minimal() +
  theme(axis.title = element_text(size = 20), axis.text = element_text(size = 20),  legend.text = element_text(size=15),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + scale_x_continuous(n.breaks = 5) +
  geom_line(aes(y =line),size=1,color='steelblue') + scale_x_continuous(breaks=c(0, 50, 100, 150, 200, 250, 300, 350, 400)) 
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/wild_turnover_Jaccard_aero.png", width=7, height=5)

# Lab:
lab$aitch_sim <- lab$BC_distance
mdl.1 <- lm((aitch_sim) ~ added_days_step, data = lab)
summary(mdl.1) # F-statistic: 29.78, 254 DF, p-value = 1.15e-07, Adjusted R-squared: 0.1014 
plot(mdl.1)
AIC(mdl.1) # -409.7189
mdl.1 <- lm(log(aitch_sim) ~ added_days_step, data = lab)
summary(mdl.1) # F-statistic: 25.51, 254 DF, p-value = 8.412e-07, Adjusted R-squared: 0.08769 
plot(mdl.1)
AIC(mdl.1) # 85.96133 --> use non-log transformed

mdl.1 <- lm((aitch_sim) ~ added_days_step, data = lab)

#use it for prediction
summary(lab$added_days_step) # predict between 6-197 days
pdat <- expand.grid(added_days_step=seq(6,200))

pred <- predict (mdl.1, newdata = pdat, na.rm = T,
                 type= "response", se.fit = TRUE, level=0,
                 interval = "prediction")
predframe <- data.frame (pdat, preds = pred$fit, se = pred$se.fit)
predframe_lab <- predframe

p1 <- ggplot(lab, aes(x=added_days_step, y=BC_distance)) +
  geom_point(alpha=0.3, colour="steelblue", size=2) +
  geom_line(aes(y=preds.fit), data=predframe_lab, colour="steelblue", size=1) +
  geom_ribbon(data = predframe_lab, aes(y = NULL, ymin = preds.fit-(1.96*se),
                                        ymax = preds.fit+(1.96*se)), alpha=0.15, fill="steelblue") +
  ylab("Jaccard distance") +
  xlab("Days elapsed") +
  theme_minimal() +
  theme(axis.title=element_text(size=20), axis.text = element_text(size=20),
        title = element_text(size=20), panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("") + ylim(0,1)# +
  stat_regline_equation(aes(label =  paste(..rr.label.., sep = "~~~~"))) # r2=0.1
p <- p1 + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/lab_turnover_Jaccard_aero.png", width=5, height=5)

# nonaerotolerant

D <- phyloseq::distance(nonaerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Mouse <- rownames(sd)
sd$newdate <- strptime(as.character(sd$Collection_date), "%d/%m/%Y") # here use current format
sd$Date <- format(sd$newdate, "%Y-%m-%d") 
sd <- sd[order(sd$Animal_ID, sd$Date, decreasing = F),]
sd$Time_point <- as.numeric(unlist(lapply(split(sd, sd$Animal_ID), function(x) { rank(x$Date) })))
sd <- data.frame(sd[,c("Source", "Mouse", "Animal_ID", "Date", "Time_point")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp", "mouse2_source", "mouse2_id", "mouse2_date", "mouse2_tp")
distance <- na.omit(distance)
self <- distance[which(distance$mouse1_id==distance$mouse2_id),] # only self comparisons
first_tp <- sd[which(sd$Time_point==1),]
first_tp <- first_tp[,c("Animal_ID", "Date")]
self <- merge(self, first_tp, by.x="mouse1_id", by.y="Animal_ID", all.x=FALSE, all.y=FALSE)
self$added_days <- difftime(self$mouse2_date, self$Date, units="days") # counting how many days added to first sample
self <- self[which(self$added_days>0),]
self$added_days_step <- difftime(self$mouse2_date, self$mouse1_date, units="days")
self <- self[which(self$added_days_step>0),]
#self <- self[which(self$added_days_step>=6),]
self$tp <- paste(self$mouse1_tp, self$mouse2_tp, sep="-")
#self <- self[which(self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="2-3" | self$tp=="3-4" | self$tp=="4-5" | self$tp=="5-6" | self$tp=="6-7" | self$tp=="7-8" | self$tp=="8-9" | self$tp=="9-10"),]
self <- self[which(self$mouse1_tp!=self$mouse2_tp),]
self$added_days <- as.numeric(self$added_days)
self$added_days_step <- as.numeric(self$added_days_step)

wild <- self[which(self$mouse1_source=="Wild"),]
lab <- self[which(self$mouse1_source=="Lab"),]

library(easynls)
library(nlshelper)

# Wild:
wildX <- wild
wildX$aitch_sim <- log(wildX$BC_distance)
wildY <- wildX[, c("added_days_step", "aitch_sim")]
nlsfit(wildY, model=4) # adj R2=4.988000e-01, AIC = -1.379099e+03
wildY <- wildX[, c("added_days_step", "BC_distance")]
nlsfit(wildY, model=4) # adj R2=4.930000e-01, AIC = -2.493665e+03 --> better fit
nlsfit(wildY, model=1) # AIC -1.994662e+03 --> delta AIC
dta.nls <- nls(BC_distance ~ (a + b * added_days_step + c * I(added_days_step^2)) * (added_days_step <= -0.5 * b/c) + (a + I(-b^2/(4 * c))) * (added_days_step > -0.5 * b/c), wildY, start=list(a=0.4986297,b=0.02313252,c=as.numeric(-0.0005745700)))
wildY$line <- predict(dta.nls)
plot <- ggplot(wildY, aes(x=added_days_step, y=BC_distance)) +
  annotate("segment", x = 20.13029, 
           xend = 20.13029, y = 0, 
           yend = 1,
           colour = "indianred4",size=1,linetype = "dashed")+
  geom_point(size=2, colour = "indianred",alpha=.1) +
  xlab("Days elapsed") + ylab("Jaccard distance") + 
  theme_minimal() +
  theme(axis.title = element_text(size = 20), axis.text = element_text(size = 20),  legend.text = element_text(size=20),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + scale_x_continuous(n.breaks = 5) +
  geom_line(aes(y =line),size=1,color='indianred')  + scale_x_continuous(breaks=c(0, 50, 100, 150, 200, 250, 300, 350, 400)) #+ scale_y_continuous(n.breaks = 5) + ylim(2.5,5.0) 
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/wild_turnover_Jaccard_anaero.png", width=7, height=5)

# Lab:
lab$aitch_sim <- lab$BC_distance
mdl.1 <- lm(log(aitch_sim) ~ added_days_step, data = lab)
summary(mdl.1) # F-statistic: 277.2, 254 DF, p-value: < 2.2e-16, Adjusted R-squared:  0.5199 
plot(mdl.1)
AIC(mdl.1) # -211.285
mdl.1 <- lm((aitch_sim) ~ added_days_step, data = lab)
plot(mdl.1)
AIC(mdl.1) # -636.5217 -> use non-log transformed

#use it for prediction
summary(lab$added_days_step) # predict between 9-197 days
pdat <- expand.grid(added_days_step=seq(6,200))

pred <- predict (mdl.1, newdata = pdat, na.rm = T,
                 type= "response", se.fit = TRUE, level=0,
                 interval = "prediction")
predframe <- data.frame (pdat, preds = pred$fit, se = pred$se.fit)
predframe_lab <- predframe

p1 <- ggplot(lab, aes(x=added_days_step, y=BC_distance)) +
  geom_point(alpha=0.3, colour="indianred", size=2) +
  geom_line(aes(y=preds.fit), data=predframe_lab, colour="indianred", size=1) +
  geom_ribbon(data = predframe_lab, aes(y = NULL, ymin = preds.fit-(1.96*se),
                                        ymax = preds.fit+(1.96*se)), alpha=0.15, fill="indianred") +
  ylab("Jaccard distance") +
  xlab("Days elapsed") +
  theme_minimal() +
  theme(axis.title=element_text(size=20), axis.text = element_text(size=20),
        title = element_text(size=20), panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("") + ylim(0,1) #+
  stat_regline_equation(aes(label =  paste(..rr.label.., sep = "~~~~"))) # r2=0.57
p <- p1 + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/lab_turnover_Jaccard_anaero.png", width=5, height=5)

```

```{r equal_sized_subsets}

# Going to repeat the above, but with equal-sized subsets of anaerobes and aerobes, because more anaerobes in gut microbiota.
aero <- read.csv("MUSMUS/MUSMUS_aerotolerance_data.csv")
aero$Aerotolerance_cat[aero$Aerotolerance_cat=="nonaeotolerant"] <- "nonaerotolerant"
aero <- aero[,c("Family", "Genus", "Aerotolerance_cat")]
aero$Family_Genus <- paste(aero$Family, aero$Genus, sep="-") # 698
aero <- aero[!duplicated(aero),] # 698

# WILD MICE:

# Is microbial turnover rate depended on bacterial aerotolerance?
phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds") # select all longitudinally sampled lab and wild mice
phy <- subset_samples(phy, Buffer!="R" & Buffer!="NB" & Buffer!="E" & Buffer!="PBS" & Buffer!="RNAlater")

phy <- subset_taxa(phy, Genus!="Imtechella")
phy <- subset_taxa(phy, Genus!="Allobacillus")
phy <- microbiome::transform(phy, "compositional")

phy <- subset_samples(phy, Source=="Wild")
phy <- filter_taxa(phy, function(x) max(x) > 0, TRUE)

library(microbiome)
library(ggplot2)
library(dplyr)
library(vegan)

sample_data(phy)$datedate <- sample_data(phy)$Collection_date
sample_data(phy)$datedate <- strptime(as.character(sample_data(phy)$datedate), "%d/%m/%Y")
sample_data(phy)$datedate <- format(sample_data(phy)$datedate, "%Y-%m-%d") 
sample_data(phy)$Birthdate <- strptime(as.character(sample_data(phy)$Birthdate), "%d/%m/%Y")
sample_data(phy)$Birthdate <- format(sample_data(phy)$Birthdate, "%Y-%m-%d") 
sample_data(phy)$age <- difftime(sample_data(phy)$datedate, sample_data(phy)$Birthdate, units="days")

mice <- data.frame(sample_data(phy))
mice$Sample_ID <- rownames(mice)
long <- mice %>% group_by(Animal_ID) %>% filter(n()>1) # Select mice sampled a minimum 2 times
unique(long$Animal_ID)
long <- long[which(long$Animal_ID!="NA"),]
long <- long[which(long$Animal_ID!="SK0"),] # Excluded unknown mice
table(long$Source) # 99 lab (King's college, C57BL/6), 717 wild (Skokholm)

check_l <- long[which(long$Source=="Lab"),]
table(check_l$Animal_ID) # 5-7 samples each
length(unique(check_l$Animal_ID)) # 16

check_l <- long[which(long$Source=="Wild"),]
table(check_l$Animal_ID) # 2-12 samples each
length(unique(check_l$Animal_ID)) # 217

time <- phy
sample_data(time)$Sample_ID <- rownames(sample_data(time))
sample_data(time)$include[sample_data(time)$Sample_ID%in%long$Sample_ID] <- "yes"
time <- subset_samples(time, include=="yes")

aero_key <- phy_phe[,c("OTU", "Aerotolerance_cat")] # need to run above 2 chunks
aero_key <- aero_key[!duplicated(aero_key),]

time 
time  <- prune_taxa(taxa_sums(time) > 0, time) 
aerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="aerotolerant"),]
nonaerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="nonaerotolerant"),]
otu_table(time) <- t(otu_table(time))
aerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% aerotolerant_key$OTU)
aerotolerant <- merge_phyloseq(aerotolerant, tax_table(time), sample_data(time))
aerotolerant # 3582  taxa
nonaerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% nonaerotolerant_key$OTU)
nonaerotolerant <- merge_phyloseq(nonaerotolerant, tax_table(time), sample_data(time))
nonaerotolerant # 6150 taxa

# --> take a random subset of 3000 from each 
set.seed(7865)
all_taxa <- taxa_names(aerotolerant)
selected_taxa <- sample(all_taxa, 3000)
aerotolerant <- prune_taxa(selected_taxa, aerotolerant) # 3000 taxa
all_taxa <- taxa_names(nonaerotolerant)
selected_taxa <- sample(all_taxa, 3000)
nonaerotolerant <- prune_taxa(selected_taxa, nonaerotolerant) # 3000 taxa

# aerotolerant:

D <- phyloseq::distance(aerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Mouse <- rownames(sd)
sd$newdate <- strptime(as.character(sd$Collection_date), "%d/%m/%Y") # here use current format
sd$Date <- format(sd$newdate, "%Y-%m-%d") 
sd <- sd[order(sd$Animal_ID, sd$Date, decreasing = F),]
sd$Time_point <- as.numeric(unlist(lapply(split(sd, sd$Animal_ID), function(x) { rank(x$Date) })))
sd <- data.frame(sd[,c("Source", "Mouse", "Animal_ID", "Date", "Time_point")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp", "mouse2_source", "mouse2_id", "mouse2_date", "mouse2_tp")
distance <- na.omit(distance)
self <- distance[which(distance$mouse1_id==distance$mouse2_id),] # only self comparisons
first_tp <- sd[which(sd$Time_point==1),]
first_tp <- first_tp[,c("Animal_ID", "Date")]
self <- merge(self, first_tp, by.x="mouse1_id", by.y="Animal_ID", all.x=FALSE, all.y=FALSE)
self$added_days <- difftime(self$mouse2_date, self$Date, units="days") # counting how many days added to first sample
self <- self[which(self$added_days>0),]
self$added_days_step <- difftime(self$mouse2_date, self$mouse1_date, units="days")
self <- self[which(self$added_days_step>0),]
#self <- self[which(self$added_days_step>=6),]
self$tp <- paste(self$mouse1_tp, self$mouse2_tp, sep="-")
#self <- self[which(self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="2-3" | self$tp=="3-4" | self$tp=="4-5" | self$tp=="5-6" | self$tp=="6-7" | self$tp=="7-8" | self$tp=="8-9" | self$tp=="9-10"),]
self <- self[which(self$mouse1_tp!=self$mouse2_tp),]
self$added_days <- as.numeric(self$added_days)
self$added_days_step <- as.numeric(self$added_days_step)

wild <- self[which(self$mouse1_source=="Wild"),]
lab <- self[which(self$mouse1_source=="Lab"),]

library(easynls)
library(nlshelper)

# Wild:
wildX <- wild
wildX$aitch_sim <- log(wildX$BC_distance)
wildY <- wildX[, c("added_days_step", "aitch_sim")]
nlsplot (wildY, model=4,xlab="",ylab="") 
nlsfit(wildY, model=4) # adj r2=0.25110000, AIC -810.24967064
wildY <- wildX[, c("added_days_step", "BC_distance")]
nlsplot (wildY, model=4,xlab="",ylab="") 
nlsfit(wildY, model=4) # adj r2=2.620000e-01, AIC -1.988368e+03 --> non-log transformed gives a better fit
dta.nls <- nls(BC_distance ~ (a + b * added_days_step + c * I(added_days_step^2)) * (added_days_step <= -0.5 * b/c) + (a + I(-b^2/(4 * c))) * (added_days_step > -0.5 * b/c), wildY, start=list(a=0.5679285,b=0.009196730,c=as.numeric(-0.0001189000)))
wildY$line <- predict(dta.nls)
plot <- ggplot(wildY, aes(x=added_days_step, y=BC_distance)) +
  annotate("segment", x = 38.67312, 
           xend = 38.67312, y = 0, 
           yend = 1,
           colour = "steelblue4",size=1,linetype = "dashed")+
  geom_point(size=2, colour = "steelblue",alpha=.1) +
  xlab("Days elapsed") + ylab("Jaccard distance") + 
  theme_minimal() +
  theme(axis.title = element_text(size = 20), axis.text = element_text(size = 20),  legend.text = element_text(size=15),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + scale_x_continuous(n.breaks = 5) +
  geom_line(aes(y =line),size=1,color='steelblue') + scale_x_continuous(breaks=c(0, 50, 100, 150, 200, 250, 300, 350, 400)) 
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/wild_turnover_Jaccard_aero_SUBSET3000.png", width=7, height=5)

# nonaerotolerant

D <- phyloseq::distance(nonaerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Mouse <- rownames(sd)
sd$newdate <- strptime(as.character(sd$Collection_date), "%d/%m/%Y") # here use current format
sd$Date <- format(sd$newdate, "%Y-%m-%d") 
sd <- sd[order(sd$Animal_ID, sd$Date, decreasing = F),]
sd$Time_point <- as.numeric(unlist(lapply(split(sd, sd$Animal_ID), function(x) { rank(x$Date) })))
sd <- data.frame(sd[,c("Source", "Mouse", "Animal_ID", "Date", "Time_point")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp", "mouse2_source", "mouse2_id", "mouse2_date", "mouse2_tp")
distance <- na.omit(distance)
self <- distance[which(distance$mouse1_id==distance$mouse2_id),] # only self comparisons
first_tp <- sd[which(sd$Time_point==1),]
first_tp <- first_tp[,c("Animal_ID", "Date")]
self <- merge(self, first_tp, by.x="mouse1_id", by.y="Animal_ID", all.x=FALSE, all.y=FALSE)
self$added_days <- difftime(self$mouse2_date, self$Date, units="days") # counting how many days added to first sample
self <- self[which(self$added_days>0),]
self$added_days_step <- difftime(self$mouse2_date, self$mouse1_date, units="days")
self <- self[which(self$added_days_step>0),]
#self <- self[which(self$added_days_step>=6),]
self$tp <- paste(self$mouse1_tp, self$mouse2_tp, sep="-")
#self <- self[which(self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="2-3" | self$tp=="3-4" | self$tp=="4-5" | self$tp=="5-6" | self$tp=="6-7" | self$tp=="7-8" | self$tp=="8-9" | self$tp=="9-10"),]
self <- self[which(self$mouse1_tp!=self$mouse2_tp),]
self$added_days <- as.numeric(self$added_days)
self$added_days_step <- as.numeric(self$added_days_step)

wild <- self[which(self$mouse1_source=="Wild"),]
lab <- self[which(self$mouse1_source=="Lab"),]

library(easynls)
library(nlshelper)

# Wild:
wildX <- wild
wildX$aitch_sim <- log(wildX$BC_distance)
wildY <- wildX[, c("added_days_step", "aitch_sim")]
nlsfit(wildY, model=4) # adj R2= 4.609000e-01, AIC = -1.517089e+03
wildY <- wildX[, c("added_days_step", "BC_distance")]
nlsfit(wildY, model=4) # adj R2=4.624000e-01, AIC = -2.512718e+03 --> better fit
dta.nls <- nls(BC_distance ~ (a + b * added_days_step + c * I(added_days_step^2)) * (added_days_step <= -0.5 * b/c) + (a + I(-b^2/(4 * c))) * (added_days_step > -0.5 * b/c), wildY, start=list(a=0.5518267,b=0.02128972,c=as.numeric(-0.0005220500)))
wildY$line <- predict(dta.nls)
plot <- ggplot(wildY, aes(x=added_days_step, y=BC_distance)) +
  annotate("segment", x = 20.39040, 
           xend = 20.39040, y = 0, 
           yend = 1,
           colour = "indianred4",size=1,linetype = "dashed")+
  geom_point(size=2, colour = "indianred",alpha=.1) +
  xlab("Days elapsed") + ylab("Jaccard distance") + 
  theme_minimal() +
  theme(axis.title = element_text(size = 20), axis.text = element_text(size = 20),  legend.text = element_text(size=20),
        panel.grid.major = element_blank(), panel.grid.minor = element_blank()) + scale_x_continuous(n.breaks = 5) +
  geom_line(aes(y =line),size=1,color='indianred')  + scale_x_continuous(breaks=c(0, 50, 100, 150, 200, 250, 300, 350, 400)) #+ scale_y_continuous(n.breaks = 5) + ylim(2.5,5.0) 
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/wild_turnover_Jaccard_anaero_SUBSET3000.png", width=7, height=5)

# LAB MICE:

# Is microbial turnover rate depended on bacterial aerotolerance?
phy <- readRDS("IMR_16S_NO_DOUBLETONS.rds") # select all longitudinally sampled lab and wild mice
phy <- subset_samples(phy, Buffer!="R" & Buffer!="NB" & Buffer!="E" & Buffer!="PBS" & Buffer!="RNAlater")

phy <- subset_taxa(phy, Genus!="Imtechella")
phy <- subset_taxa(phy, Genus!="Allobacillus")
phy <- microbiome::transform(phy, "compositional")

phy <- subset_samples(phy, Source=="Lab")
phy <- filter_taxa(phy, function(x) max(x) > 0, TRUE)

library(microbiome)
library(ggplot2)
library(dplyr)
library(vegan)

sample_data(phy)$datedate <- sample_data(phy)$Collection_date
sample_data(phy)$datedate <- strptime(as.character(sample_data(phy)$datedate), "%d/%m/%Y")
sample_data(phy)$datedate <- format(sample_data(phy)$datedate, "%Y-%m-%d") 
sample_data(phy)$Birthdate <- strptime(as.character(sample_data(phy)$Birthdate), "%d/%m/%Y")
sample_data(phy)$Birthdate <- format(sample_data(phy)$Birthdate, "%Y-%m-%d") 
sample_data(phy)$age <- difftime(sample_data(phy)$datedate, sample_data(phy)$Birthdate, units="days")

mice <- data.frame(sample_data(phy))
mice$Sample_ID <- rownames(mice)
long <- mice %>% group_by(Animal_ID) %>% filter(n()>1) # Select mice sampled a minimum 2 times
unique(long$Animal_ID)
long <- long[which(long$Animal_ID!="NA"),]
long <- long[which(long$Animal_ID!="SK0"),] # Excluded unknown mice
table(long$Source) # 99 lab (King's college, C57BL/6), 717 wild (Skokholm)

check_l <- long[which(long$Source=="Lab"),]
table(check_l$Animal_ID) # 5-7 samples each
length(unique(check_l$Animal_ID)) # 16

check_l <- long[which(long$Source=="Wild"),]
table(check_l$Animal_ID) # 2-12 samples each
length(unique(check_l$Animal_ID)) # 217

time <- phy
sample_data(time)$Sample_ID <- rownames(sample_data(time))
sample_data(time)$include[sample_data(time)$Sample_ID%in%long$Sample_ID] <- "yes"
time <- subset_samples(time, include=="yes")

aero_key <- phy_phe[,c("OTU", "Aerotolerance_cat")] # need to run above 2 chunks
aero_key <- aero_key[!duplicated(aero_key),]

time # 
time  <- prune_taxa(taxa_sums(time) > 0, time) # 
aerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="aerotolerant"),]
nonaerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="nonaerotolerant"),]
otu_table(time) <- t(otu_table(time))
aerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% aerotolerant_key$OTU)
aerotolerant <- merge_phyloseq(aerotolerant, tax_table(time), sample_data(time))
aerotolerant # 117 taxa
nonaerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% nonaerotolerant_key$OTU)
nonaerotolerant <- merge_phyloseq(nonaerotolerant, tax_table(time), sample_data(time))
nonaerotolerant # 619 taxa

# --> take a random subset of 100 from each 
set.seed(7865)
all_taxa <- taxa_names(aerotolerant)
selected_taxa <- sample(all_taxa, 100)
aerotolerant <- prune_taxa(selected_taxa, aerotolerant) # 100 taxa
all_taxa <- taxa_names(nonaerotolerant)
selected_taxa <- sample(all_taxa, 100)
nonaerotolerant <- prune_taxa(selected_taxa, nonaerotolerant) # 100 taxa

# aerotolerant:

D <- phyloseq::distance(aerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Mouse <- rownames(sd)
sd$newdate <- strptime(as.character(sd$Collection_date), "%d/%m/%Y") # here use current format
sd$Date <- format(sd$newdate, "%Y-%m-%d") 
sd <- sd[order(sd$Animal_ID, sd$Date, decreasing = F),]
sd$Time_point <- as.numeric(unlist(lapply(split(sd, sd$Animal_ID), function(x) { rank(x$Date) })))
sd <- data.frame(sd[,c("Source", "Mouse", "Animal_ID", "Date", "Time_point")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp", "mouse2_source", "mouse2_id", "mouse2_date", "mouse2_tp")
distance <- na.omit(distance)
self <- distance[which(distance$mouse1_id==distance$mouse2_id),] # only self comparisons
first_tp <- sd[which(sd$Time_point==1),]
first_tp <- first_tp[,c("Animal_ID", "Date")]
self <- merge(self, first_tp, by.x="mouse1_id", by.y="Animal_ID", all.x=FALSE, all.y=FALSE)
self$added_days <- difftime(self$mouse2_date, self$Date, units="days") # counting how many days added to first sample
self <- self[which(self$added_days>0),]
self$added_days_step <- difftime(self$mouse2_date, self$mouse1_date, units="days")
self <- self[which(self$added_days_step>0),]
#self <- self[which(self$added_days_step>=6),]
self$tp <- paste(self$mouse1_tp, self$mouse2_tp, sep="-")
#self <- self[which(self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="2-3" | self$tp=="3-4" | self$tp=="4-5" | self$tp=="5-6" | self$tp=="6-7" | self$tp=="7-8" | self$tp=="8-9" | self$tp=="9-10"),]
self <- self[which(self$mouse1_tp!=self$mouse2_tp),]
self$added_days <- as.numeric(self$added_days)
self$added_days_step <- as.numeric(self$added_days_step)

wild <- self[which(self$mouse1_source=="Wild"),]
lab <- self[which(self$mouse1_source=="Lab"),]

library(easynls)
library(nlshelper)

# Lab:
lab$aitch_sim <- lab$BC_distance
mdl.1 <- lm((aitch_sim) ~ added_days_step, data = lab)
summary(mdl.1) # F-statistic: 25.76, 254 DF, p-value = 7.465e-07, Adjusted R-squared: 0.08852 
plot(mdl.1) # a little skewed
AIC(mdl.1) # -404.5773
mdl.1 <- lm(log(aitch_sim) ~ added_days_step, data = lab)
summary(mdl.1) # F-statistic: 22.95, 254 DF, p-value = 2.828e-06, Adjusted R-squared: 0.07926 
plot(mdl.1) # looks worse
AIC(mdl.1) # 141.3249 --> use non-log transformed

mdl.1 <- lm((aitch_sim) ~ added_days_step, data = lab)

#use it for prediction
summary(lab$added_days_step) # predict between 6-197 days
pdat <- expand.grid(added_days_step=seq(6,200))

pred <- predict (mdl.1, newdata = pdat, na.rm = T,
                 type= "response", se.fit = TRUE, level=0,
                 interval = "prediction")
predframe <- data.frame (pdat, preds = pred$fit, se = pred$se.fit)
predframe_lab <- predframe

p1 <- ggplot(lab, aes(x=added_days_step, y=BC_distance)) +
  geom_point(alpha=0.3, colour="steelblue", size=2) +
  geom_line(aes(y=preds.fit), data=predframe_lab, colour="steelblue", size=1) +
  geom_ribbon(data = predframe_lab, aes(y = NULL, ymin = preds.fit-(1.96*se),
                                        ymax = preds.fit+(1.96*se)), alpha=0.15, fill="steelblue") +
  ylab("Jaccard distance") +
  xlab("Days elapsed") +
  theme_minimal() +
  theme(axis.title=element_text(size=20), axis.text = element_text(size=20),
        title = element_text(size=20), panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("") + ylim(0,1)# +
  stat_regline_equation(aes(label =  paste(..rr.label.., sep = "~~~~"))) # r2=0.1
p <- p1 + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/lab_turnover_Jaccard_aero_SUBSET100.png", width=5, height=5)

# nonaerotolerant

D <- phyloseq::distance(nonaerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Mouse <- rownames(sd)
sd$newdate <- strptime(as.character(sd$Collection_date), "%d/%m/%Y") # here use current format
sd$Date <- format(sd$newdate, "%Y-%m-%d") 
sd <- sd[order(sd$Animal_ID, sd$Date, decreasing = F),]
sd$Time_point <- as.numeric(unlist(lapply(split(sd, sd$Animal_ID), function(x) { rank(x$Date) })))
sd <- data.frame(sd[,c("Source", "Mouse", "Animal_ID", "Date", "Time_point")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_id", "mouse1_date", "mouse1_tp", "mouse2_source", "mouse2_id", "mouse2_date", "mouse2_tp")
distance <- na.omit(distance)
self <- distance[which(distance$mouse1_id==distance$mouse2_id),] # only self comparisons
first_tp <- sd[which(sd$Time_point==1),]
first_tp <- first_tp[,c("Animal_ID", "Date")]
self <- merge(self, first_tp, by.x="mouse1_id", by.y="Animal_ID", all.x=FALSE, all.y=FALSE)
self$added_days <- difftime(self$mouse2_date, self$Date, units="days") # counting how many days added to first sample
self <- self[which(self$added_days>0),]
self$added_days_step <- difftime(self$mouse2_date, self$mouse1_date, units="days")
self <- self[which(self$added_days_step>0),]
#self <- self[which(self$added_days_step>=6),]
self$tp <- paste(self$mouse1_tp, self$mouse2_tp, sep="-")
#self <- self[which(self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="1-2" | self$tp=="2-3" | self$tp=="3-4" | self$tp=="4-5" | self$tp=="5-6" | self$tp=="6-7" | self$tp=="7-8" | self$tp=="8-9" | self$tp=="9-10"),]
self <- self[which(self$mouse1_tp!=self$mouse2_tp),]
self$added_days <- as.numeric(self$added_days)
self$added_days_step <- as.numeric(self$added_days_step)

wild <- self[which(self$mouse1_source=="Wild"),]
lab <- self[which(self$mouse1_source=="Lab"),]

library(easynls)
library(nlshelper)

# Lab:
lab$aitch_sim <- lab$BC_distance
mdl.1 <- lm(log(aitch_sim) ~ added_days_step, data = lab)
summary(mdl.1) # F-statistic: 144.2, 254 DF, p-value: < 2.2e-16, Adjusted R-squared:  0.3595 
plot(mdl.1) # not too good fit
AIC(mdl.1) # 124.2522
mdl.1 <- lm((aitch_sim) ~ added_days_step, data = lab)
plot(mdl.1) # better
AIC(mdl.1) # -444.2949 -> use log transformed

mdl.1 <- lm(log(aitch_sim) ~ added_days_step, data = lab)

#use it for prediction
summary(lab$added_days_step) # predict between 9-197 days
pdat <- expand.grid(added_days_step=seq(6,200))

pred <- predict (mdl.1, newdata = pdat, na.rm = T,
                 type= "response", se.fit = TRUE, level=0,
                 interval = "prediction")
predframe <- data.frame (pdat, preds = pred$fit, se = pred$se.fit)
predframe_lab <- predframe

p1 <- ggplot(lab, aes(x=added_days_step, y=log(BC_distance))) +
  geom_point(alpha=0.3, colour="indianred", size=2) +
  geom_line(aes(y=preds.fit), data=predframe_lab, colour="indianred", size=1) +
  geom_ribbon(data = predframe_lab, aes(y = NULL, ymin = preds.fit-(1.96*se),
                                        ymax = preds.fit+(1.96*se)), alpha=0.15, fill="indianred") +
  ylab("log(Jaccard distance)") +
  xlab("Days elapsed") +
  theme_minimal() +
  theme(axis.title=element_text(size=20), axis.text = element_text(size=20),
        title = element_text(size=20), panel.grid.major = element_blank(), panel.grid.minor = element_blank()) +
  ggtitle("") #+ ylim(0,1) #+
  stat_regline_equation(aes(label =  paste(..rr.label.., sep = "~~~~"))) # r2=0.57
p <- p1 + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/lab_turnover_Jaccard_anaero_SUBSET100.png", width=5, height=5)

```

***Q13: Does collection time of faecal sample predict ratio between anaerobes and aerobes?***

```{r}

aero <- read.csv("MUSMUS/MUSMUS_aerotolerance_data.csv")
aero$Aerotolerance_cat[aero$Aerotolerance_cat=="nonaeotolerant"] <- "nonaerotolerant"
aero <- aero[,c("Family", "Genus", "Aerotolerance_cat")]
aero$Family_Genus <- paste(aero$Family, aero$Genus, sep="-") # 698
aero <- aero[!duplicated(aero),] # 698

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
phy <- subset_samples(phy, Project=="Skokholm")
taxa <- data.frame(tax_table(phy)) 
taxa <- taxa[,c("Family", "Genus")]
taxa <- taxa[!duplicated(taxa),] 
taxa$Family_Genus <- paste(taxa$Family, taxa$Genus, sep="-")

taxa <- merge(taxa, aero, by.all="Family_Genus", all.x=TRUE, all.y=FALSE) 
table(taxa$Aerotolerance_cat)
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Unknown"] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat==""] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Mixed"] <- "unknown"
taxa$unknown <- "unknown"
taxa$Aerotolerance_cat <- ifelse(is.na(taxa$Aerotolerance_cat), taxa$unknown, taxa$Aerotolerance_cat)
table(taxa$Aerotolerance_cat)
un <- taxa[which(taxa$Aerotolerance_cat=="unknown"),]
length(unique(un$Family)) 

phy2 <- phy %>%
  #aggregate_taxa(level = "Genus") %>%  
  microbiome::transform(transform = "compositional")
phy_melt <- psmelt(phy2)
taxa_min <- taxa[,c("Family", "Genus", "Aerotolerance_cat")]
taxa_min$fam_gen <- paste(taxa_min$Family, taxa_min$Genus, sep="-")
phy_melt$fam_gen <- paste(phy_melt$Family, phy_melt$Genus, sep="-")
phy_phe <- merge(phy_melt, taxa_min, by="fam_gen", all.x=TRUE, all.y=FALSE)

aero_key <- phy_phe[,c("OTU", "Aerotolerance_cat")]
aero_key <- aero_key[!duplicated(aero_key),]

arr <- phy_phe[,c("Sample", "Aerotolerance_cat", "Abundance")]
arr$ID <- paste(arr$Sample, arr$Aerotolerance_cat, sep="-")
arr <- arr[,c("ID", "Abundance")]

arr2 <- arr %>%
  group_by(ID) %>%
  summarise(total_count = sum(Abundance))

arr2$Sample_ID <- sub("\\-.*", "", arr2$ID)
arr2$Aerotolerance <- sub("[^-]+", '', x = arr2$ID)
arr2$Aerotolerance <- str_sub(arr2$Aerotolerance, start = 2)

source <- phy_phe[,c("Sample", "Source", "Project", "Collection_date", "Body_mass_g", "read_count", "Faeces_trap_colle", "Peanut_left_g", "Trap_cold", "Animal_ID")]
arr3 <- merge(arr2, source, by.x="Sample_ID", by.y="Sample", all.x=T, all.y=F)
arr3 <- arr3[!duplicated(arr3),] 

colnames(arr3)
arr3 <- arr3[,c(1,3:13)]

arr3 <- arr3 %>%
  mutate(Aerobes = ifelse(Aerotolerance == "aerotolerant", total_count, NA),
         Anaerobes = ifelse(Aerotolerance == "nonaerotolerant", total_count, NA),
         Unknown = ifelse(Aerotolerance == "unknown", total_count, NA)) %>%
  select(Sample_ID, Aerobes, Anaerobes, Unknown, Source, Project, 
         Collection_date, Body_mass_g, read_count, Faeces_trap_colle, 
         Peanut_left_g, Trap_cold, Animal_ID)

arr3 <- arr3 %>%
  group_by(Sample_ID, Source, Project, Collection_date, Body_mass_g, read_count, Faeces_trap_colle, Peanut_left_g, Trap_cold, Animal_ID) %>%
  summarize(
    Aerobes = max(Aerobes, na.rm = TRUE),
    Anaerobes = max(Anaerobes, na.rm = TRUE),
    Unknown = max(Unknown, na.rm = TRUE),
    .groups = 'drop'
  )

arr3$ratio <- arr3$Aerobes/arr3$Anaerobes

# Add session info (year 2020)
session2020 <- read.csv("2020_Sessions.csv")
colnames(session2020)
session2020 <- session2020[,c(14, 10, 11, 16, 17)]
colnames(session2020) <- c("Collection_date", "Setup_start", "Setup_end", "Collection_start", "Collection_end")

df <- merge(arr3, session2020, by="Collection_date", all.x=T, all.y=F)
df$Setup_earliest <- "17:00"
df$Collection_date <- as.POSIXct(df$Collection_date, format = "%d/%m/%Y")
df$Trap_set_date <- df$Collection_date - 1
df$Trap_set_date <- str_sub(df$Trap_set_date, start = 1, end = 10)

df$Setup_start_complete <- paste(df$Trap_set_date, df$Setup_earliest)
df$Setup_start_complete <- as.POSIXct(df$Setup_start_complete, format = "%Y-%m-%d %H:%M")
df$Sample_complete <- paste(df$Collection_date, df$Faeces_trap_colle)
df$Sample_complete <- as.POSIXct(df$Sample_complete, format = "%Y-%m-%d %H:%M:%S")

df$max_time_until_collection <- as.numeric(difftime(df$Sample_complete, df$Setup_start_complete, units = "hours"))
df <- df[complete.cases(df$max_time_until_collection),]
summary(df$Unknown)
# Min.   1st Qu.    Median      Mean   3rd Qu.      Max. 
# 0.0008584 0.0326320 0.0479399 0.0547266 0.0665659 0.2055542   
df <- df[which(df$Unknown<0.05),] # exclude samples in which proportion of taxa with unknown aerotolerance >5%
write.csv(df, "MUSMUS/MUSMUS_aerotolerance_ratio_brm/exposure_time/exposure_time_ratio/aerotolerance_ratio.csv")

hist(df$ratio)

# Run the below model on super computer

# numerical values to 0-1
scalecols <- c("ratio", "Unknown", "read_count", "max_time_until_collection")
range.use <- function(x, min.use, max.use) {
  (x - min(x, na.rm = TRUE)) / (max(x, na.rm = TRUE) - min(x, na.rm = TRUE)) * (max.use - min.use) + min.use
}

for (col in scalecols) {
  df[, col] <- range.use(df[, col], 0, 1)
}

samplesize <- nrow(df)
df$ratio <- (df$ratio * (samplesize - 1) + 0.5) / samplesize

df$Trap_cold[is.na(df$Trap_cold)] <- "Trap not in cold"
df$Trap_in_cold <- "Yes"
df$Trap_in_cold[df$Trap_cold=="Trap not in cold"] <- "No"

#model1 <- brm(ratio ~ 1 + max_time_until_collection + Trap_in_cold + (1 | Animal_ID) + (1 | read_count) + (1 | Unknown), data = df, family = "Beta",control = list(adapt_delta = 0.9, max_treedepth = 12), warmup = 2000, iter = 4000, cores = 1, chains = 1,save_pars = save_pars(group = FALSE), init = 0)

# Does exposure time predict asv richness of aerotolerant or anaerobic community?
aero <- read.csv("MUSMUS/MUSMUS_aerotolerance_ratio_brm/exposure_time/exposure_time_richness/aerotolerant_richness.csv")
add <- read.csv("MUSMUS/MUSMUS_aerotolerance_ratio_brm/exposure_time/exposure_time_ratio/aerotolerance_ratio.csv")
add <- add[,c("Sample_ID", "Trap_cold", "max_time_until_collection")]
aero <- merge(aero, add, by.x="Sample", by.y="Sample_ID", all.x=T, all.y=T)
aero <- aero[which(aero$Project_ID=="Skokholm"),]
write.csv(aero, "MUSMUS/MUSMUS_aerotolerance_ratio_brm/exposure_time/exposure_time_richness/aerotolerant_richness.csv")

aero <- read.csv("MUSMUS/MUSMUS_aerotolerance_ratio_brm/exposure_time/exposure_time_richness/anaerobic_richness.csv")
add <- read.csv("MUSMUS/MUSMUS_aerotolerance_ratio_brm/exposure_time/exposure_time_ratio/aerotolerance_ratio.csv")
add <- add[,c("Sample_ID", "Trap_cold", "max_time_until_collection")]
aero <- merge(aero, add, by.x="Sample", by.y="Sample_ID", all.x=T, all.y=T)
aero <- aero[which(aero$Project_ID=="Skokholm"),]
write.csv(aero, "MUSMUS/MUSMUS_aerotolerance_ratio_brm/exposure_time/exposure_time_richness/anaerobic_richness.csv")

# Use super computer to run brm models

```

# PART 2

**PART 2: GUT MICROBIOTA VARIATION IN MAINLAND VS ISLAND HOUSE MICE**

***Step 1: Select data***

```{r}

phy <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
phy <- subset_samples(phy, Source=="Wild")
phy # 180 samples
table(sample_data(phy)$Project)
# Cologne    Espelette        Faroe  Isle_of_May Midway_Atoll     Skokholm       Wytham 
#    11            8           38            3           15           98            7 
saveRDS(phy, "MUSMUS/musmus_mainlandVSisland_final_PS.RDS")

```

***Q1: How much of all gut microbial variation does (1) habitat (island vs mainland) and (2) population explain?***

```{r PERMANOVA}

set.seed(8765)
phy <- readRDS("MUSMUS/musmus_mainlandVSisland_final_PS.RDS")
sample_data(phy)$Group <- "Island"
sample_data(phy)$Group[sample_data(phy)$Project=="Cologne" | sample_data(phy)$Project=="Espelette" | sample_data(phy)$Project=="Wytham"] <- "Mainland"

# Habitat

# Jaccard

M4 <- data.frame(sample_data(phy))
D <- phyloseq::distance(phy, method = "jaccard", type = "samples", binary=T)

adonis2(D ~ M4$read_count + M4$MiSeq_batch +
          M4$Group, na.rm=TRUE,
        by="margin")
# read count R2=0.00902, F=1.6865, p=0.001
# Miseq batch R2=0.01934, F=3.6156, p=0.001
# Group R2=0.02592, F=4.8455, p=0.001

beta <- betadisper(D, M4$Group)
permutest(beta) # F=2.3209, p=0.146

# Aitchison

M4 <- data.frame(sample_data(phy))
phy_clr <- microbiome::transform(phy, "clr")
D <- phyloseq::distance(phy_clr, method = "euclidean", type = "samples")

adonis2(D ~ M4$read_count + M4$MiSeq_batch +
          M4$Group, na.rm=TRUE,
        by="margin")
# read count R2=0.00676, F=1.2666, p=0.059
# Miseq batch R2=0.02067, F=3.8717, p=0.001
# Group R2=0.02711, F=5.0776, p=0.001

beta <- betadisper(D, M4$Group)
permutest(beta) # F=16.154, p=0.001

# UniFrac

M4 <- data.frame(sample_data(phy))
D <- phyloseq::distance(phy, method = "unifrac", type = "samples")

adonis2(D ~ M4$read_count + M4$MiSeq_batch +
          M4$Group, na.rm=TRUE,
        by="margin")
# read count R2=0.01056, F=2.0160, p=0.006
# Miseq batch R2=0.03301, F=6.3025, p=0.001
# Group R2=0.03041, F=5.8055, p=0.001

beta <- betadisper(D, M4$Group)
permutest(beta) # F=0.0525, p=0.824

# Weighted UniFrac

M4 <- data.frame(sample_data(phy))
D <- phyloseq::distance(phy, method = "wunifrac", type = "samples")

adonis2(D ~ M4$read_count + M4$MiSeq_batch +
          M4$Group, na.rm=TRUE,
        by="margin")
# read count R2=0.01938, F=3.6760, p=0.007
# Miseq batch R2=0.02822, F=5.3536, p=0.001
# Group R2=0.02145, F=4.0698, p=0.002

beta <- betadisper(D, M4$Group)
permutest(beta) # F=0.414, p=0.531

# Population

# Jaccard

M4 <- data.frame(sample_data(phy))
D <- phyloseq::distance(phy, method = "jaccard", type = "samples", binary=T)

adonis2(D ~ M4$read_count + M4$MiSeq_batch +
          M4$Project, na.rm=TRUE,
        by="margin")
# read count R2=0.00748, F=1.5087, p=0.008
# Miseq batch R2=0.00741, F=1.4938, p=0.003
# Project R2=0.11904, F=3.9998, p=0.001

beta <- betadisper(D, M4$Project)
permutest(beta) # F=38.093, p=0.001

# Aitchison

M4 <- data.frame(sample_data(phy))
phy_clr <- microbiome::transform(phy, "clr")
D <- phyloseq::distance(phy_clr, method = "euclidean", type = "samples")

adonis2(D ~ M4$read_count + M4$MiSeq_batch +
          M4$Project, na.rm=TRUE,
        by="margin")
# read count R2=0.00607, F=1.2373, p=0.068
# Miseq batch R2=0.00666, F=1.3562, p=0.028
# Group R2=0.12745, F=4.3277, p=0.001

beta <- betadisper(D, M4$Project)
permutest(beta) # F=11.575, p=0.001

```

***Q2: Do mainland and island samples cluster separately?***

```{r}

phy <- readRDS("MUSMUS/musmus_mainlandVSisland_final_PS.RDS")
table(sample_data(phy)$Project)
#     Cologne    Espelette        Faroe  Isle_of_May Midway_Atoll     Skokholm       Wytham 
#         11            8           38            3           15           98            7 
# That is 26 mainland samples and 154 island samples

# Jaccard
ord <- ordinate(phy, "PCoA", "jaccard", binary=T)
plot <- plot_ordination(phy, ord, color="Project") +
                geom_point(size = 6, alpha=.9, stroke=0) +
  scale_color_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb")) +
  theme_classic() + 
  labs(title = "Jaccard",
       x = "PCo1 (6.2%)",
       y = "PCo2 (4.1%)",
       color = "") +
  theme(axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        title = element_text(size=20),
        legend.text = element_text(size = 5),
        legend.position = "none")  #+
  stat_ellipse(type = "norm", linetype = 2)
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_PCoA_jaccard.png", width=6, height=6)

# Aitchison
phy_clr <- microbiome::transform(phy, "clr")
ord <- ordinate(phy_clr, "PCoA", "euclidean")
plot <- plot_ordination(phy, ord, color="Project") +
                geom_point(size = 6, alpha=.9, stroke=0) +
  scale_color_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb")) +
  theme_classic() + 
  labs(title = "Aitchison",
       x = "PCo1 (8.1%)",
       y = "PCo2 (4.7%)",
       color = "") +
  theme(axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        title = element_text(size=20),
        legend.text = element_text(size = 5),
        legend.position = "none") 
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_PCoA_aitchison.png", width=6, height=6)
  
# UniFrac
ord <- ordinate(phy, "PCoA", "unifrac")
plot <- plot_ordination(phy, ord, color="Project") +
                geom_point(size = 6, alpha=.9, stroke=0) +
  scale_color_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb")) +
  theme_classic() + 
  labs(title = "Unweighted UniFrac",
       x = "PCo1 (11.7%)",
       y = "PCo2 (7.4%)",
       color = "") +
  theme(axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        title = element_text(size=20),
        legend.text = element_text(size = 5),
        legend.position = "none") 
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_PCoA_unifrac.png", width=6, height=6)
 
# Weighted UniFrac
ord <- ordinate(phy, "PCoA", "wunifrac")
plot <- plot_ordination(phy, ord, color="Project") +
                geom_point(size = 6, alpha=.9, stroke=0) +
  scale_color_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway_Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle_of_May" = "#3498c3", "Skokholm" = "#b7e7fb")) +
  theme_classic() + 
  labs(title = "Weighted UniFrac",
       x = "PCo1 (33.6%)",
       y = "PCo2 (13.0%)",
       color = "") +
  theme(axis.text = element_text(size=20),
        axis.title = element_text(size=20),
        title = element_text(size=20),
        legend.text = element_text(size = 5),
        legend.position = "none")  
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_PCoA_wunifrac.png", width=6, height=6)

```

***Q3: Which taxa are indicative of habitat?***

```{r Random Forest}

phy <- readRDS("MUSMUS/musmus_mainlandVSisland_final_PS.RDS")
sample_data(phy)$Group <- "Island"
sample_data(phy)$Group[sample_data(phy)$Project=="Cologne" | sample_data(phy)$Project=="Espelette" | sample_data(phy)$Project=="Wytham"] <- "Mainland"

library(randomForest)

predictors <- otu_table(phy)
dim(predictors) 
source <- as.factor(sample_data(phy)$Group) 
data <- data.frame(source, predictors) 

library(randomForest)
set.seed(321)
mouse_source <- randomForest(source ~ ., data = data, ntree=10000, importance=T)
print(mouse_source)

## look at SV importance (mean decrease gini)
mouse_source_RF <- importance(mouse_source)
mouse_source_RF_dat <- data.frame(predictors = rownames(mouse_source_RF), mouse_source_RF)
mouse_source_RF_sort <- arrange(mouse_source_RF_dat, desc(MeanDecreaseGini)) # mean decrease gini = IncNodePurity --> measures how much the model error increases when a particular variable is randomly permuted or shuffled
mouse_source_RF_sort$predictors <- factor(mouse_source_RF_sort$predictors, levels = mouse_source_RF_sort$predictors)
mouse_source_RF_sub <- mouse_source_RF_sort[1:30, ]

## coloured by family
tax_tab <- tax_table(phy)
colnames(mouse_source_RF_sub) <- c('ASV_name', "Island", "Mainland", 'MeanDecreaseAccuracy', 'MeanDecreaseGini')
tax_SV_fam <- tax_tab[,5]
tax_SV_fam <- as.data.frame(tax_SV_fam)
tax_SV_fam$ASV_name <- rownames(tax_SV_fam)
phy_plusfam <- merge(mouse_source_RF_sub, tax_SV_fam, by='ASV_name', all.x = T)
phy_plusfam_sort <- arrange(phy_plusfam, desc(MeanDecreaseGini))

# Plot
top6 <- as.list(as.character(phy_plusfam_sort$ASV_name[c(1:10)]))
phy_top6 <- subset_taxa(phy, taxa_names(phy)%in%top6)

phy_top6.melt <- psmelt(phy_top6)

#calculate mean and se of abundances per source per ASV
library(plyr)
phy_top6_sum <-  ddply(phy_top6.melt, c("OTU", "Group"), summarise,
                       N   = length(Abundance),
                       mean = mean(Abundance) ,
                       sd  = sd(Abundance),
                       se  = sd / sqrt(N))
phy_top6_sum
colnames(phy_top6_sum)[colnames(phy_top6_sum)=="mean"] <- "Abundance"
#add in taxa
phy_top6_sum$Phylum <- phy_top6.melt$Phylum[match(phy_top6_sum$OTU, phy_top6.melt$OTU)]
phy_top6_sum$Class <- phy_top6.melt$Class[match(phy_top6_sum$OTU, phy_top6.melt$OTU)]
phy_top6_sum$Order <- phy_top6.melt$Order[match(phy_top6_sum$OTU, phy_top6.melt$OTU)]
phy_top6_sum$Family <- phy_top6.melt$Family[match(phy_top6_sum$OTU, phy_top6.melt$OTU)]
phy_top6_sum$Genus <- phy_top6.melt$Genus[match(phy_top6_sum$OTU, phy_top6.melt$OTU)]
phy_top6_sum$Species <- phy_top6.melt$Species[match(phy_top6_sum$OTU, phy_top6.melt$OTU)]
phy_top6_sum$Fam_gen <- paste(phy_top6_sum$Family, phy_top6_sum$Genus, sep=" ")

phy_top6_sum$Taxa <- paste(phy_top6_sum$Phylum, phy_top6_sum$Class, phy_top6_sum$Order, phy_top6_sum$Family, sep=", ")
phy_top6_sum$Genus[phy_top6_sum$Genus=="Other"] <- "genus unknown"
phy_top6_sum$Species[phy_top6_sum$Species=="Other"] <- ", species unknown"
phy_top6_sum$Species[phy_top6_sum$Species=="intestinale"] <- " intestinale"
phy_top6_sum$Taxa2 <- paste(phy_top6_sum$Genus, phy_top6_sum$Species, sep="")
phy_top6_sum$Taxa2[phy_top6_sum$Genus=="genus unknown"] <- "genus unknown"
phy_top6_sum$taxa_full <- paste(phy_top6_sum$Taxa, phy_top6_sum$Taxa2, sep=", ")
unique(phy_top6_sum$taxa_full)
# [1] "Bacteroidota, Bacteroidia, Bacteroidales, Tannerellaceae, Parabacteroides, species unknown"  
# [2] "Bacteroidota, Bacteroidia, Bacteroidales, Muribaculaceae, Muribaculum intestinale"           
# [3] "Firmicutes, Clostridia, Lachnospirales, Lachnospiraceae, genus unknown"                      
# [4] "Firmicutes, Bacilli, Lactobacillales, Lactobacillaceae, Limosilactobacillus, species unknown"
# [5] "Firmicutes, Clostridia, Oscillospirales, Oscillospiraceae, Oscillibacter, species unknown"   
# [6] "Firmicutes, Clostridia, Oscillospirales, Oscillospiraceae, genus unknown"                    
# [7] "Firmicutes, Clostridia, Lachnospirales, Lachnospiraceae, A2, species unknown"                
# [8] "Firmicutes, Bacilli, Lactobacillales, Lactobacillaceae, Leuconostoc, species unknown"

unique(phy_top6_sum$Fam_gen)
# [1] "Tannerellaceae Parabacteroides"       "Muribaculaceae Muribaculum"           "Lachnospiraceae Other"               
# [4] "Lactobacillaceae Limosilactobacillus" "Oscillospiraceae Oscillibacter"       "Oscillospiraceae Other"              
# [7] "Lachnospiraceae A2"                   "Lactobacillaceae Leuconostoc"

phy_top6_sum$Fam_gen[phy_top6_sum$Fam_gen==""]

ordered_levels <- unique(phy_top6_sum$Fam_gen)

# Reorder factor levels of 'Fam_gen' based on the extracted order
phy_top6_sum$Fam_gen <- factor(phy_top6_sum$Fam_gen, levels = ordered_levels)

# Define the colors for the levels
color_values <- c("Tannerellaceae Parabacteroides" = "#FFADAD",
                  "Lachnospiraceae A2" = "#FFD6A5",
                  "Lachnospiraceae Other" = "#FDFFB6",
                  "Oscillospiraceae Oscillibacter" = "#CAFFBF",
                  "Lactobacillaceae Limosilactobacillus" = "#9BF6FF",
                  "Muribaculaceae Muribaculum" = "#A0C4FF",
                  "Oscillospiraceae Other" = "#BDB2FF",
                  "Lactobacillaceae Leuconostoc" = "#FFC6FF")

# Subset the color_values to only include the colors present in the ordered_levels
color_values <- color_values[ordered_levels]

p2 <- ggplot(phy_top6_sum, aes(x=Group, y=Abundance)) +
  geom_bar(stat = "identity", aes(fill=Fam_gen)) +
  geom_errorbar(data=phy_top6_sum,
                aes(ymax=Abundance + se, ymin=Abundance),
                width=0.5) +
  facet_wrap(~OTU, scales = "free_y", nrow = 1) +
  theme_minimal() +
  theme(axis.text.x = element_text(size=18, angle = 45, hjust=1),
        axis.text.y = element_text(size=20), 
        axis.title = element_text(size=30), 
        panel.spacing = unit(2, "lines"),
        legend.position = "bottom",  # Move legend to the bottom
        legend.title = element_blank(),  # Optionally remove legend title
        legend.text = element_text(size=15),  # Adjust legend text size if needed
        legend.key.size = unit(1, "cm"),  # Height of legend keys
        legend.key.width = unit(2, "cm")) +  # Width of legend keys
  ggtitle("") +
  scale_fill_manual(values = color_values) +
  guides(fill = guide_legend(ncol = 1))  
p <- p2 + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_RF.png", width=21, height=12)

# What is the relative abundance of Muribaculum intestinale?
phy <- readRDS("MUSMUS/musmus_mainlandVSisland_final_PS.RDS")
sample_data(phy)$Group <- "Island"
sample_data(phy)$Group[sample_data(phy)$Project=="Cologne" | sample_data(phy)$Project=="Espelette" | sample_data(phy)$Project=="Wytham"] <- "Mainland"
phy <- tax_glom(phy, taxrank = "Species")
phy <- subset_taxa(phy, Species=="intestinale")
View(tax_table(phy)) # only taxon is Muribaculum intestinale
View(otu_table(phy))
summary(otu_table(phy))
#  Min.   :0.0000000, Mean   :0.0004694, Median :0.0000000, Max.   :0.0110378 

```

***Q4: Does beta diversity within populations vary by habitat?***

```{r}

phy <- readRDS("MUSMUS/musmus_mainlandVSisland_final_PS.RDS")

sample_data(phy)$Source <- "Island"
sample_data(phy)$Source[sample_data(phy)$Project=="Cologne" | sample_data(phy)$Project=="Espelette" | sample_data(phy)$Project=="Wytham"] <- "Mainland"

sample_data(phy)$Project_ID <- sample_data(phy)$Project
unique(sample_data(phy)$Project_ID)
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Isle_of_May"] <- "Isle of May"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Midway_Atoll"] <- "Midway Atoll"

# Aitchison
phy_clr <- microbiome::transform(phy, "clr")
D <- phyloseq::distance(phy_clr, method = "euclidean", type = "samples")
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(phy))
sd$Mouse <- rownames(sd)
sd$Sample_ID <- rownames(sd)
sd$Animal_ID <- ifelse(is.na(sd$Animal_ID), sd$Sample_ID, sd$Animal_ID)
sd <- data.frame(sd[,c("Source", "Mouse", "Project_ID", "Sample_ID")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID", "mouse2_source", "mouse2_project", "mouse2_ID")
distance <- na.omit(distance)
distance <- distance[distance$mouse1!=distance$mouse2, ] 
distance <- distance[distance$mouse1_ID!=distance$mouse2_ID, ]
distance$Source <- paste(distance$mouse1_source, distance$mouse2_source, sep="-")
distance <- distance[which(distance$Source!="Wild-Lab"),]
distance <- distance[which(distance$Source!="Lab-Wild"),]
distance$Project <- paste(distance$mouse1_project, distance$mouse2_project, sep="-")
same <- distance[which(distance$mouse1_project==distance$mouse2_project),]

# Wilcoxon test to check if differences between bins (within source) are significant
#Get observed u-statistic
set.seed(8976)
u.obs<-wilcox.test(BC_distance ~ mouse2_source, data=same)
u.obs # W = 1944808, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"BC_distance"
for (i in 1:Nperm) {
  d<-same[,!names(same) %in% lose]
  d$dist.rand<-sample(same$BC_distance, length(same$BC_distance), replace=F)
  x<-wilcox.test(dist.rand ~ mouse2_source, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat>u.obs$statistic, 1, 0)
perm1$result 
p<-mean(perm1$result)
p*2 # 0

# Plotting
same$Project_nr <- NA
same2 <- same
same2$mouse2_project[same2$mouse2_source=="Island"] <- "Island"
same2$Project_nr[same2$mouse2_source=="Island"] <- 9
same2$mouse2_project[same2$mouse2_source=="Mainland"] <- "Mainland"
same2$Project_nr[same2$mouse2_source=="Mainland"] <- 4
same <- rbind(same, same2)

same$Project_nr[same$mouse2_project=="Skokholm"] <- "8"
same$Project_nr[same$mouse2_project=="Isle of May"] <- "7"
same$Project_nr[same$mouse2_project=="Espelette"] <- "3"
same$Project_nr[same$mouse2_project=="Faroe"] <- "6"
same$Project_nr[same$mouse2_project=="Midway Atoll"] <- "5"
same$Project_nr[same$mouse2_project=="Cologne"] <- "1"
same$Project_nr[same$mouse2_project=="Wytham"] <- "2"
same$Project_nr <- as.numeric(as.character(same$Project_nr))

same$Color <- ifelse(same$mouse2_project == "Mainland", "#87A878", 
                     ifelse(same$mouse2_project == "Island", "#5BC0EB", "black"))

same <- same %>%
  mutate(Color = case_when(
    mouse2_project == "Mainland" ~ "#87A878",
    mouse2_project == "Island" ~ "#5BC0EB",
    TRUE ~ "black"
  ),
  Label = case_when(
    Project_nr == 4 ~ "M",
    Project_nr == 9 ~ "I",
    TRUE ~ as.character(Project_nr)
  ))

plot <- ggplot(data = same, aes(x = reorder(Label, Project_nr), y = BC_distance, fill = mouse2_project, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle of May" = "#3498c3", "Skokholm" = "#b7e7fb", "Island" = "white", "Mainland" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("Aitchison dissimilarity within populations") +
  rremove("legend.title") +
    theme(axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        axis.text.x = element_text(face=c("plain", "plain", "plain", "bold", "plain", "plain", "plain", "plain", "bold"), size=20), 
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() + scale_y_continuous(n.breaks=6) 
p <- plot + geom_signif(comparisons = list(c("I", "M")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_Aitchison.png", width=4, height=6)

# Jaccard
D <- phyloseq::distance(phy, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(phy))
sd$Mouse <- rownames(sd)
sd$Sample_ID <- rownames(sd)
sd$Animal_ID <- ifelse(is.na(sd$Animal_ID), sd$Sample_ID, sd$Animal_ID)
sd <- data.frame(sd[,c("Source", "Mouse", "Project_ID", "Sample_ID")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID", "mouse2_source", "mouse2_project", "mouse2_ID")
distance <- na.omit(distance)
distance <- distance[distance$mouse1!=distance$mouse2, ] 
distance <- distance[distance$mouse1_ID!=distance$mouse2_ID, ]
distance$Source <- paste(distance$mouse1_source, distance$mouse2_source, sep="-")
distance <- distance[which(distance$Source!="Wild-Lab"),]
distance <- distance[which(distance$Source!="Lab-Wild"),]
distance$Project <- paste(distance$mouse1_project, distance$mouse2_project, sep="-")
same <- distance[which(distance$mouse1_project==distance$mouse2_project),]

# Wilcoxon test to check if differences between bins (within source) are significant
#Get observed u-statistic
set.seed(8976)
u.obs<-wilcox.test(BC_distance ~ mouse2_source, data=same)
u.obs # W = 1667392, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"BC_distance"
for (i in 1:Nperm) {
  d<-same[,!names(same) %in% lose]
  d$dist.rand<-sample(same$BC_distance, length(same$BC_distance), replace=F)
  x<-wilcox.test(dist.rand ~ mouse2_source, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat>u.obs$statistic, 1, 0)
perm1$result 
p<-mean(perm1$result)
p*2 # 0

# Plotting
same$Project_nr <- NA
same2 <- same
same2$mouse2_project[same2$mouse2_source=="Island"] <- "Island"
same2$Project_nr[same2$mouse2_source=="Island"] <- 9
same2$mouse2_project[same2$mouse2_source=="Mainland"] <- "Mainland"
same2$Project_nr[same2$mouse2_source=="Mainland"] <- 4
same <- rbind(same, same2)

same$Project_nr[same$mouse2_project=="Skokholm"] <- "8"
same$Project_nr[same$mouse2_project=="Isle of May"] <- "7"
same$Project_nr[same$mouse2_project=="Espelette"] <- "3"
same$Project_nr[same$mouse2_project=="Faroe"] <- "6"
same$Project_nr[same$mouse2_project=="Midway Atoll"] <- "5"
same$Project_nr[same$mouse2_project=="Cologne"] <- "1"
same$Project_nr[same$mouse2_project=="Wytham"] <- "2"
same$Project_nr <- as.numeric(as.character(same$Project_nr))

same$Color <- ifelse(same$mouse2_project == "Mainland", "#87A878", 
                     ifelse(same$mouse2_project == "Island", "#5BC0EB", "black"))

same <- same %>%
  mutate(Color = case_when(
    mouse2_project == "Mainland" ~ "#87A878",
    mouse2_project == "Island" ~ "#5BC0EB",
    TRUE ~ "black"
  ),
  Label = case_when(
    Project_nr == 4 ~ "M",
    Project_nr == 9 ~ "I",
    TRUE ~ as.character(Project_nr)
  ))

plot <- ggplot(data = same, aes(x = reorder(Label, Project_nr), y = BC_distance, fill = mouse2_project, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle of May" = "#3498c3", "Skokholm" = "#b7e7fb", "Island" = "white", "Mainland" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("Jaccard dissimilarity within populations") +
  rremove("legend.title") +
    theme(axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        axis.text.x = element_text(face=c("plain", "plain", "plain", "bold", "plain", "plain", "plain", "plain", "bold"), size=20), 
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() + scale_y_continuous(n.breaks=6) 
p <- plot + geom_signif(comparisons = list(c("I", "M")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_Jaccard.png", width=4, height=6)

```

***Q5: Does beta diversity across populations vary by habitat?***

```{r}

phy <- readRDS("MUSMUS/musmus_mainlandVSisland_final_PS.RDS")

sample_data(phy)$Source <- "Island"
sample_data(phy)$Source[sample_data(phy)$Project=="Cologne" | sample_data(phy)$Project=="Espelette" | sample_data(phy)$Project=="Wytham"] <- "Mainland"

sample_data(phy)$Project_ID <- sample_data(phy)$Project
unique(sample_data(phy)$Project_ID)
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Isle_of_May"] <- "Isle of May"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Midway_Atoll"] <- "Midway Atoll"

# Aitchison
phy_clr <- microbiome::transform(phy, "clr")
D <- phyloseq::distance(phy_clr, method = "euclidean", type = "samples")
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(phy))
sd$Mouse <- rownames(sd)
sd$Sample_ID <- rownames(sd)
sd$Animal_ID <- ifelse(is.na(sd$Animal_ID), sd$Sample_ID, sd$Animal_ID)
sd <- data.frame(sd[,c("Source", "Mouse", "Project_ID", "Sample_ID")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID", "mouse2_source", "mouse2_project", "mouse2_ID")
distance <- na.omit(distance)
distance <- distance[distance$mouse1!=distance$mouse2, ] 
distance <- distance[distance$mouse1_ID!=distance$mouse2_ID, ]
distance$Source <- paste(distance$mouse1_source, distance$mouse2_source, sep="-")
distance <- distance[which(distance$Source!="Wild-Lab"),]
distance <- distance[which(distance$Source!="Lab-Wild"),]
distance$Project <- paste(distance$mouse1_project, distance$mouse2_project, sep="-")
same <- distance[which(distance$mouse1_project!=distance$mouse2_project),] # only across population comparisons
same <- same[which(same$mouse1_source==same$mouse2_source),] # only within habitat comparisons

# Wilcoxon test to check if differences between bins (within source) are significant
#Get observed u-statistic
set.seed(8976)
u.obs<-wilcox.test(BC_distance ~ mouse2_source, data=same)
u.obs # W = 4362948, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"BC_distance"
for (i in 1:Nperm) {
  d<-same[,!names(same) %in% lose]
  d$dist.rand<-sample(same$BC_distance, length(same$BC_distance), replace=F)
  x<-wilcox.test(dist.rand ~ mouse2_source, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat>u.obs$statistic, 1, 0)
perm1$result 
p<-mean(perm1$result)
p*2 # 0

# Plotting
same$Project_nr <- NA
same2 <- same
same2$mouse2_project[same2$mouse2_source=="Island"] <- "Island"
same2$Project_nr[same2$mouse2_source=="Island"] <- 9
same2$mouse2_project[same2$mouse2_source=="Mainland"] <- "Mainland"
same2$Project_nr[same2$mouse2_source=="Mainland"] <- 4
same <- rbind(same, same2)

same$Project_nr[same$mouse2_project=="Skokholm"] <- "8"
same$Project_nr[same$mouse2_project=="Isle of May"] <- "7"
same$Project_nr[same$mouse2_project=="Espelette"] <- "3"
same$Project_nr[same$mouse2_project=="Faroe"] <- "6"
same$Project_nr[same$mouse2_project=="Midway Atoll"] <- "5"
same$Project_nr[same$mouse2_project=="Cologne"] <- "1"
same$Project_nr[same$mouse2_project=="Wytham"] <- "2"
same$Project_nr <- as.numeric(as.character(same$Project_nr))

same$Color <- ifelse(same$mouse2_project == "Mainland", "#87A878", 
                     ifelse(same$mouse2_project == "Island", "#5BC0EB", "black"))

same <- same %>%
  mutate(Color = case_when(
    mouse2_project == "Mainland" ~ "#87A878",
    mouse2_project == "Island" ~ "#5BC0EB",
    TRUE ~ "black"
  ),
  Label = case_when(
    Project_nr == 4 ~ "M",
    Project_nr == 9 ~ "I",
    TRUE ~ as.character(Project_nr)
  ))

plot <- ggplot(data = same, aes(x = reorder(Label, Project_nr), y = BC_distance, fill = mouse2_project, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle of May" = "#3498c3", "Skokholm" = "#b7e7fb", "Island" = "white", "Mainland" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("Aitchison dissimilarity across populations") +
  rremove("legend.title") +
    theme(axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        axis.text.x = element_text(face=c("plain", "plain", "plain", "bold", "plain", "plain", "plain", "plain", "bold"), size=20), 
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() + scale_y_continuous(n.breaks=6) 
p <- plot + geom_signif(comparisons = list(c("I", "M")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_across_Aitchison.png", width=4.18, height=6)

# Jaccard
D <- phyloseq::distance(phy, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(phy))
sd$Mouse <- rownames(sd)
sd$Sample_ID <- rownames(sd)
sd$Animal_ID <- ifelse(is.na(sd$Animal_ID), sd$Sample_ID, sd$Animal_ID)
sd <- data.frame(sd[,c("Source", "Mouse", "Project_ID", "Sample_ID")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID", "mouse2_source", "mouse2_project", "mouse2_ID")
distance <- na.omit(distance)
distance <- distance[distance$mouse1!=distance$mouse2, ] 
distance <- distance[distance$mouse1_ID!=distance$mouse2_ID, ]
distance$Source <- paste(distance$mouse1_source, distance$mouse2_source, sep="-")
distance <- distance[which(distance$Source!="Wild-Lab"),]
distance <- distance[which(distance$Source!="Lab-Wild"),]
distance$Project <- paste(distance$mouse1_project, distance$mouse2_project, sep="-")
same <- distance[which(distance$mouse1_project!=distance$mouse2_project),] # only across population comparisons
same <- same[which(same$mouse1_source==same$mouse2_source),] # only within habitat comparisons

# Wilcoxon test to check if differences between bins (within source) are significant
#Get observed u-statistic
set.seed(8976)
u.obs<-wilcox.test(BC_distance ~ mouse2_source, data=same)
u.obs # W = 13467384, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"BC_distance"
for (i in 1:Nperm) {
  d<-same[,!names(same) %in% lose]
  d$dist.rand<-sample(same$BC_distance, length(same$BC_distance), replace=F)
  x<-wilcox.test(dist.rand ~ mouse2_source, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat>u.obs$statistic, 1, 0)
perm1$result 
p<-mean(perm1$result)
p*2 # 0

# Plotting
same$Project_nr <- NA
same2 <- same
same2$mouse2_project[same2$mouse2_source=="Island"] <- "Island"
same2$Project_nr[same2$mouse2_source=="Island"] <- 9
same2$mouse2_project[same2$mouse2_source=="Mainland"] <- "Mainland"
same2$Project_nr[same2$mouse2_source=="Mainland"] <- 4
same <- rbind(same, same2)

same$Project_nr[same$mouse2_project=="Skokholm"] <- "8"
same$Project_nr[same$mouse2_project=="Isle of May"] <- "7"
same$Project_nr[same$mouse2_project=="Espelette"] <- "3"
same$Project_nr[same$mouse2_project=="Faroe"] <- "6"
same$Project_nr[same$mouse2_project=="Midway Atoll"] <- "5"
same$Project_nr[same$mouse2_project=="Cologne"] <- "1"
same$Project_nr[same$mouse2_project=="Wytham"] <- "2"
same$Project_nr <- as.numeric(as.character(same$Project_nr))

same$Color <- ifelse(same$mouse2_project == "Mainland", "#87A878", 
                     ifelse(same$mouse2_project == "Island", "#5BC0EB", "black"))

same <- same %>%
  mutate(Color = case_when(
    mouse2_project == "Mainland" ~ "#87A878",
    mouse2_project == "Island" ~ "#5BC0EB",
    TRUE ~ "black"
  ),
  Label = case_when(
    Project_nr == 4 ~ "M",
    Project_nr == 9 ~ "I",
    TRUE ~ as.character(Project_nr)
  ))

plot <- ggplot(data = same, aes(x = reorder(Label, Project_nr), y = BC_distance, fill = mouse2_project, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle of May" = "#3498c3", "Skokholm" = "#b7e7fb", "Island" = "white", "Mainland" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("Jaccard dissimilarity across populations") +
  rremove("legend.title") +
    theme(axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        axis.text.x = element_text(face=c("plain", "plain", "plain", "bold", "plain", "plain", "plain", "plain", "bold"), size=20), 
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() + scale_y_continuous(n.breaks=6) 
p <- plot + geom_signif(comparisons = list(c("I", "M")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_across_Jaccard.png", width=4.18, height=6)

```

***Q6: Does similarity to lab mice vary by habitat?***

```{r}

wild <- readRDS("MUSMUS/musmus_mainlandVSisland_final_PS.RDS")
lab <- readRDS("MUSMUS/musmus_labVSwild_final_PS.RDS")
lab <- subset_samples(lab, Source=="Lab")
phy <- merge_phyloseq(wild, lab)
table(sample_data(phy)$Source)
# Lab Wild 
# 146  180 

sample_data(phy)$Sample_ID <- rownames(sample_data(phy))
wild <- subset_samples(phy, Source=="Wild")
sample_data(wild)$Project_ID <- sample_data(wild)$Project
lab <- subset_samples(phy, Source=="Lab")
sample_data(lab)$Project_ID <- paste(sample_data(lab)$Project, sample_data(lab)$Date, sample_data(lab)$Strain, sep="-")
phy <- merge_phyloseq(wild, lab)

sample_data(phy)$Project_ID <- sample_data(phy)$Project
unique(sample_data(phy)$Project_ID)
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Isle_of_May"] <- "Isle of May"
sample_data(phy)$Project_ID[sample_data(phy)$Project_ID=="Midway_Atoll"] <- "Midway Atoll"

# Aitchison
phy_clr <- microbiome::transform(phy, "clr")
D <- phyloseq::distance(phy_clr, method = "euclidean", type = "samples")
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(phy))
sd$Mouse <- rownames(sd)
sd$Sample_ID <- rownames(sd)
sd$Animal_ID <- ifelse(is.na(sd$Animal_ID), sd$Sample_ID, sd$Animal_ID)
sd <- data.frame(sd[,c("Source", "Mouse", "Project_ID", "Sample_ID")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID", "mouse2_source", "mouse2_project", "mouse2_ID")
distance <- na.omit(distance)
distance <- distance[distance$mouse1!=distance$mouse2, ] 
distance <- distance[distance$mouse1_ID!=distance$mouse2_ID, ]
distance$Source <- paste(distance$mouse1_source, distance$mouse2_source, sep="-")
distance$Project <- paste(distance$mouse1_project, distance$mouse2_project, sep="-")
same <- distance[which(distance$mouse1_project!=distance$mouse2_project),] # only across population comparisons
same <- same[which(same$mouse1_source!=same$mouse2_source),] # only across lab and wild comparisons
same <- same[which(same$mouse1_source=="Wild"),]
same$Habitat <- "Island"
same$Habitat[same$mouse1_project=="Cologne" | same$mouse1_project=="Espelette" | same$mouse1_project=="Wytham"] <- "Mainland"

# Wilcoxon test to check if differences between bins (within source) are significant
#Get observed u-statistic
set.seed(8976)
u.obs<-wilcox.test(BC_distance ~ Habitat, data=same)
u.obs # W = 61718930, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"BC_distance"
for (i in 1:Nperm) {
  d<-same[,!names(same) %in% lose]
  d$dist.rand<-sample(same$BC_distance, length(same$BC_distance), replace=F)
  x<-wilcox.test(dist.rand ~ Habitat, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat>u.obs$statistic, 1, 0)
perm1$result 
p<-mean(perm1$result)
p*2 # 0

# Plotting
same$Project_nr <- NA
same2 <- same
same2$mouse1_project[same2$Habitat=="Island"] <- "Island"
same2$Project_nr[same2$Habitat=="Island"] <- 9
same2$mouse1_project[same2$Habitat=="Mainland"] <- "Mainland"
same2$Project_nr[same2$Habitat=="Mainland"] <- 4
same <- rbind(same, same2)

same$Project_nr[same$mouse1_project=="Skokholm"] <- "8"
same$Project_nr[same$mouse1_project=="Isle of May"] <- "7"
same$Project_nr[same$mouse1_project=="Espelette"] <- "3"
same$Project_nr[same$mouse1_project=="Faroe"] <- "6"
same$Project_nr[same$mouse1_project=="Midway Atoll"] <- "5"
same$Project_nr[same$mouse1_project=="Cologne"] <- "1"
same$Project_nr[same$mouse1_project=="Wytham"] <- "2"
same$Project_nr <- as.numeric(as.character(same$Project_nr))

same$Color <- ifelse(same$mouse1_project == "Mainland", "#87A878", 
                     ifelse(same$mouse1_project == "Island", "#5BC0EB", "black"))

same <- same %>%
  mutate(Color = case_when(
    mouse1_project == "Mainland" ~ "#87A878",
    mouse1_project == "Island" ~ "#5BC0EB",
    TRUE ~ "black"
  ),
  Label = case_when(
    Project_nr == 4 ~ "M",
    Project_nr == 9 ~ "I",
    TRUE ~ as.character(Project_nr)
  ))

plot <- ggplot(data = same, aes(x = reorder(Label, Project_nr), y = BC_distance, fill = mouse1_project, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway Atoll" = "#106f71", "Faroe" = "#0fb7b9", "Isle of May" = "#3498c3", "Skokholm" = "#b7e7fb", "Island" = "white", "Mainland" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("Aitchison dissimilarity to lab mice") +
  rremove("legend.title") +
    theme(axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        axis.text.x = element_text(face=c("plain", "plain", "plain", "bold", "plain", "plain", "plain", "plain", "bold"), size=20), 
        axis.ticks.x=element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() + scale_y_continuous(n.breaks=6) 
p <- plot + geom_signif(comparisons = list(c("I", "M")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_Lab_Aitchison.png", width=4.18, height=6)

# Jaccard
D <- phyloseq::distance(phy, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(phy))
sd$Mouse <- rownames(sd)
sd$Sample_ID <- rownames(sd)
sd$Animal_ID <- ifelse(is.na(sd$Animal_ID), sd$Sample_ID, sd$Animal_ID)
sd <- data.frame(sd[,c("Source", "Mouse", "Project_ID", "Sample_ID")])
colnames(D_melt) <- c("mouse1", "mouse2", "BC_distance")
distance <- merge(D_melt, sd, by.x="mouse1", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse1", "mouse2", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID")
distance <- merge(distance, sd, by.x="mouse2", by.y="Mouse", all.x = TRUE, all.y=FALSE)
colnames(distance) <- c("mouse2", "mouse1", "BC_distance", "mouse1_source", "mouse1_project", "mouse1_ID", "mouse2_source", "mouse2_project", "mouse2_ID")
distance <- na.omit(distance)
distance <- distance[distance$mouse1!=distance$mouse2, ] 
distance <- distance[distance$mouse1_ID!=distance$mouse2_ID, ]
distance$Source <- paste(distance$mouse1_source, distance$mouse2_source, sep="-")
distance$Project <- paste(distance$mouse1_project, distance$mouse2_project, sep="-")
same <- distance[which(distance$mouse1_project!=distance$mouse2_project),] # only across population comparisons
same <- same[which(same$mouse1_source!=same$mouse2_source),] # only across lab and wild comparisons
same <- same[which(same$mouse1_source=="Wild"),]
same$Habitat <- "Island"
same$Habitat[same$mouse1_project=="Cologne" | same$mouse1_project=="Espelette" | same$mouse1_project=="Wytham"] <- "Mainland"

# Wilcoxon test to check if differences between bins (within source) are significant
#Get observed u-statistic
set.seed(8976)
u.obs<-wilcox.test(BC_distance ~ Habitat, data=same)
u.obs # W = 61718930, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"BC_distance"
for (i in 1:Nperm) {
  d<-same[,!names(same) %in% lose]
  d$dist.rand<-sample(same$BC_distance, length(same$BC_distance), replace=F)
  x<-wilcox.test(dist.rand ~ Habitat, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat>u.obs$statistic, 1, 0)
perm1$result 
p<-mean(perm1$result)
p*2 # 0

# Plotting
same$Project_nr <- NA
same2 <- same
same2$mouse1_project[same2$Habitat=="Island"] <- "Island"
same2$Project_nr[same2$Habitat=="Island"] <- 9
same2$mouse1_project[same2$Habitat=="Mainland"] <- "Mainland"
same2$Project_nr[same2$Habitat=="Mainland"] <- 4
same <- rbind(same, same2)

same$Project_nr[same$mouse1_project=="Skokholm"] <- "8"
same$Project_nr[same$mouse1_project=="Isle of May"] <- "7"
same$Project_nr[same$mouse1_project=="Espelette"] <- "3"
same$Project_nr[same$mouse1_project=="Faroe"] <- "6"
same$Project_nr[same$mouse1_project=="Midway Atoll"] <- "5"
same$Project_nr[same$mouse1_project=="Cologne"] <- "1"
same$Project_nr[same$mouse1_project=="Wytham"] <- "2"
same$Project_nr <- as.numeric(as.character(same$Project_nr))

same$Color <- ifelse(same$mouse1_project == "Mainland", "#87A878", 
                     ifelse(same$mouse1_project == "Island", "#5BC0EB", "black"))

same <- same %>%
  mutate(Color = case_when(
    mouse1_project == "Mainland" ~ "#87A878",
    mouse1_project == "Island" ~ "#5BC0EB",
    TRUE ~ "black"
  ),
  Label = case_when(
    Project_nr == 4 ~ "M",
    Project_nr == 9 ~ "I",
    TRUE ~ as.character(Project_nr)
  ))

plot <- ggplot(data = same, aes(x = reorder(Label, Project_nr), y = BC_distance, fill = mouse1_project, color = Color)) +
  scale_fill_manual(values = c("Cologne" = "#96d594", "Wytham" = "#46a343", "Espelette" = "#226c20", "Midway Atoll" = "#106f71", 
                               "Faroe" = "#0fb7b9", "Isle of May" = "#3498c3", "Skokholm" = "#b7e7fb", "Island" = "white", "Mainland" = "white")) +
  geom_violin(alpha = 0.4, position = position_dodge(width = 0.75), size = 1) +
  geom_boxplot(notch = TRUE, outlier.size = -1, lwd = 1.2, alpha = 0.6, position = position_dodge(width = 0.75)) +
  theme_minimal() +
  ylab("Jaccard dissimilarity to lab mice") +
  theme(legend.title = element_blank(), # Remove legend title
        axis.ticks = element_line(size = 2, color = "black"),
        axis.ticks.length = unit(0.2, "cm"),
        axis.text.y = element_text(size = 20),
        axis.title.y = element_text(size = 22),
        legend.position = "none",
        axis.text.x = element_text(face = c("plain", "plain", "plain", "bold", "plain", "plain", "plain", "plain", "bold"), size = 20), 
        axis.ticks.x = element_blank(),
        axis.title.x = element_blank()) +
  scale_color_identity() +
  scale_y_continuous(limits = c(0.4, 1), breaks = seq(0.4, 1, by = 0.1))
p <- plot + geom_signif(comparisons = list(c("I", "M")), map_signif_level = TRUE, textsize = 12, vjust = 0.4, color = "black", size = 1)
p <- p + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_Lab_Jaccard.png", width=4, height=6)

```

***Q7: Does aerotolerant community vary more across populations than does the anaerobic one?***

```{r}

aero <- read.csv("MUSMUS/MUSMUS_aerotolerance_data.csv") # Eveliina and Aura's mouse aerotolerance database
aero$Aerotolerance_cat[aero$Aerotolerance_cat=="nonaeotolerant"] <- "nonaerotolerant"
aero <- aero[,c("Family", "Genus", "Aerotolerance_cat")]
aero$Family_Genus <- paste(aero$Family, aero$Genus, sep="-") # 698
aero <- aero[!duplicated(aero),] # 698

phy <- readRDS("MUSMUS/musmus_mainlandVSisland_final_PS.RDS")

taxa <- data.frame(tax_table(phy)) 
taxa <- taxa[,c("Family", "Genus")]
taxa <- taxa[!duplicated(taxa),] # 807
taxa$Family_Genus <- paste(taxa$Family, taxa$Genus, sep="-")

taxa <- merge(taxa, aero, by.all="Family_Genus", all.x=TRUE, all.y=FALSE) # 807
table(taxa$Aerotolerance_cat) # 360 aerotolerant, 136 anaerobes
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Unknown"] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat==""] <- "unknown"
taxa$Aerotolerance_cat[taxa$Aerotolerance_cat=="Mixed"] <- "unknown"
taxa$unknown <- "unknown"
taxa$Aerotolerance_cat <- ifelse(is.na(taxa$Aerotolerance_cat), taxa$unknown, taxa$Aerotolerance_cat)
table(taxa$Aerotolerance_cat) # 311 unknown
un <- taxa[which(taxa$Aerotolerance_cat=="unknown"),]
length(unique(un$Family)) # 169

phy_melt <- psmelt(phy)
taxa_min <- taxa[,c("Family", "Genus", "Aerotolerance_cat")]
taxa_min$fam_gen <- paste(taxa_min$Family, taxa_min$Genus, sep="-")
phy_melt$fam_gen <- paste(phy_melt$Family, phy_melt$Genus, sep="-") # 498192
phy_phe <- merge(phy_melt, taxa_min, by="fam_gen", all.x=TRUE, all.y=FALSE) # 498192

# Count unique ASVs per aerotolerance category
asv <- phy_phe[,c("Aerotolerance_cat", "OTU", "Sample", "Project", "Source", "Abundance")]
asv <- asv[which(asv$Abundance!=0),]
asv <- asv %>%
  group_by(Aerotolerance_cat, Sample) %>%
  mutate(ASV_count = n()) %>%
  ungroup()
asv <- asv[,c("Sample", "Project", "Source", "Aerotolerance_cat", "ASV_count")] # 131418
asv <- asv[!duplicated(asv),] # 1746 (3 rows per sample since 3 levels of aerotolerance)
asv <- asv[which(asv$Aerotolerance_cat!="unknown"),] # 1164

# Continue with plotting beta diversity

aero_key <- phy_phe[,c("OTU", "Aerotolerance_cat")]
aero_key <- aero_key[!duplicated(aero_key),]

time <- phy

time # 12138 taxa 
aerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="aerotolerant"),]
nonaerotolerant_key <- aero_key[which(aero_key$Aerotolerance_cat=="nonaerotolerant"),]
otu_table(time) <- t(otu_table(time))
aerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% aerotolerant_key$OTU)
aerotolerant <- merge_phyloseq(aerotolerant, tax_table(time), sample_data(time))
aerotolerant
nonaerotolerant <- subset(otu_table(time), rownames(otu_table(time)) %in% nonaerotolerant_key$OTU)
nonaerotolerant <- merge_phyloseq(nonaerotolerant, tax_table(time), sample_data(time))
nonaerotolerant 

# aerotolerant:

#time_clr <- microbiome::transform(aerotolerant, "clr") 
D <- phyloseq::distance(aerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Sample_ID <- rownames(sd)
sd <- sd[,c("Project", "Sample_ID")]
get <- merge(D_melt, sd, by.x="Var1", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var1", "Var2", "Distance", "Project1")
get <- merge(get, sd, by.x="Var2", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var2", "Var1", "Distance", "Project1", "Project2")
get <- get[which(get$Var2!=get$Var1),] # remove self comps
get$Source1 <- "Island"
get$Source2 <- "Island"
get$Source1[get$Project1=="Cologne" | get$Project1=="Espelette" | get$Project1=="Wytham"] <- "Mainland"
get$Source2[get$Project2=="Cologne" | get$Project2=="Espelette" | get$Project2=="Wytham"] <- "Mainland"
get <- get[which(get$Source1==get$Source2),] # only within habitat comparisons
get <- get[which(get$Project1!=get$Project2),] # only different pop comparisons

get$aerotolerance <- "Aerotolerant"
save <- get

# nonaerotolerant

#time_clr <- microbiome::transform(nonaerotolerant, "clr") 
D <- phyloseq::distance(nonaerotolerant, method = "jaccard", type = "samples", binary=T)
D <- as.matrix(D)
D_melt <- melt(D)
sd <- data.frame(sample_data(time))
sd$Sample_ID <- rownames(sd)
sd <- sd[,c("Project", "Sample_ID")]
get <- merge(D_melt, sd, by.x="Var1", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var1", "Var2", "Distance", "Project1")
get <- merge(get, sd, by.x="Var2", by.y="Sample_ID", all.x=T, all.y=F)
colnames(get) <- c("Var2", "Var1", "Distance", "Project1", "Project2")
get <- get[which(get$Var2!=get$Var1),] # remove self comps
get$Source1 <- "Island"
get$Source2 <- "Island"
get$Source1[get$Project1=="Cologne" | get$Project1=="Espelette" | get$Project1=="Wytham"] <- "Mainland"
get$Source2[get$Project2=="Cologne" | get$Project2=="Espelette" | get$Project2=="Wytham"] <- "Mainland"
get <- get[which(get$Source1==get$Source2),] # only within habitat comparisons
get <- get[which(get$Project1!=get$Project2),] # only different pop comparisons

get$aerotolerance <- "Anaerobic"
get <- rbind(get, save)

ww2 <- get
table(ww2$Source1)
#  Island Mainland 
#  24868      884 
ww2$id <- paste(ww2$Project1, ww2$aerotolerance, sep="-")

ww3 <- ww2
ww3$Project1 <- ww3$Source1
ww3$Project2 <- ww3$Source2
ww3$id <- paste(ww3$Project1, ww3$aerotolerance, sep="-")
ww2 <- rbind(ww2, ww3)
same <- ww2

# Plotting
same$Project_nr[same$Project1=="Skokholm"] <- 8
same$Project_nr[same$Project1=="Isle_of_May"] <- 7
same$Project_nr[same$Project1=="Espelette"] <- 3
same$Project_nr[same$Project1=="Faroe"] <- 6
same$Project_nr[same$Project1=="Midway_Atoll"] <- 5
same$Project_nr[same$Project1=="Cologne"] <- 1
same$Project_nr[same$Project1=="Wytham"] <- 2
same$Project_nr[same$Project1=="Mainland"] <- 4
same$Project_nr[same$Project1=="Island"] <- 9
same$Project_nr <- as.numeric(as.character(same$Project_nr))

library(dplyr)
library(ggplot2)
library(ggdist)

same$aerotolerance2 <- same$aerotolerance
same$aerotolerance[same$Project1=="Island"] <- NA
same$aerotolerance[same$Project1=="Mainland"] <- NA

# Update the data frame with necessary columns
same <- same %>%
  mutate(
    Color = case_when(
      id == "Mainland-Aerotolerant" ~ "steelblue",
      id == "Island-Aerotolerant" ~ "steelblue",
      id == "Mainland-Anaerobic" ~ "indianred",
      id == "Island-Anaerobic" ~ "indianred",
      TRUE ~ "black"
    ),
    Label = case_when(
      Project_nr == 4 ~ "M",
      Project_nr == 9 ~ "I",
      TRUE ~ as.character(Project_nr)
    ),
    HalfEyeFillColor = case_when(
      id %in% c("Mainland-Aerotolerant", "Island-Aerotolerant") ~ "steelblue",
      id %in% c("Mainland-Anaerobic", "Island-Anaerobic") ~ "indianred",
      aerotolerance %in% c("Aerotolerant") ~ "steelblue",
      aerotolerance %in% c("Anaerobic") ~ "indianred",
      TRUE ~ "white"
    ),
    BoxPlotFillColor = case_when(
      id %in% c("Mainland-Aerotolerant", "Island-Aerotolerant") ~ "white",
      id %in% c("Mainland-Anaerobic", "Island-Anaerobic") ~ "white",
      aerotolerance %in% c("Aerotolerant") ~ "steelblue",
      aerotolerance %in% c("Anaerobic") ~ "indianred",
      TRUE ~ "white"
    )
  )

# Define color scale for ids
id_colors <- c(
  "steelblue" = "steelblue",
  "indianred" = "indianred",
  "black" = "black"
)

# Define fill scale for boxplots and half-eyes
halfeye_fill_colors <- c(
  "steelblue" = "steelblue",
  "indianred" = "indianred",
  "white" = "white"
)

boxplot_fill_colors <- c(
  "steelblue" = "steelblue",
  "indianred" = "indianred",
  "white" = "white"
)

# Bold the labels M and I
custom_labels <- function(labels) {
  sapply(labels, function(label) {
    if (label %in% c("M", "I")) {
      bquote(bold(.(label)))
    } else {
      label
    }
  })
}

plot <- ggplot(same, aes(x = reorder(Label, Project_nr), y = Distance, color = Color)) + 
  ggdist::stat_halfeye(
    aes(fill = HalfEyeFillColor),
    alpha = 0.6,
    width = .6, 
    .width = 0, 
    justification = -.3, 
    point_colour = NA
  ) +
  geom_boxplot(
    aes(fill = BoxPlotFillColor),
    alpha = 0.6,
    width = .25, 
    outlier.shape = 1, 
    outlier.alpha = 0,
    lwd = 1
  ) + 
  scale_fill_manual(name = "HalfEyeFill", values = halfeye_fill_colors, guide = "none") +
  scale_color_manual(values = id_colors) + 
  theme_classic() +
  theme(
    axis.ticks = element_line(size = 2, color = "black"),
    axis.ticks.length = unit(0.2, "cm"),
    axis.text.y = element_text(size = 20),
    axis.title.y = element_text(size = 22),
    legend.position = "none",
    axis.text.x = element_text(size = 20),
    axis.ticks.x = element_blank(),
    axis.title.x = element_blank()
  ) +
  scale_x_discrete(labels = custom_labels)  +
  ylab("Jaccard dissimilarity across populations")
p <- plot + theme(rect = element_rect(fill = "transparent"))
ggsave("MUSMUS/MUSMUS_figures/MainlandIsland_aerotolerance.png", height=6, width=14)

# permutational Wilcoxon rank sum tests to see if levels are significantly different
test <- same[which(same$Project1=="Island" | same$Project1=="Mainland"),]
an <- test[which(test$aerotolerance2=="Anaerobic"),]
a <- test[which(test$aerotolerance2=="Aerotolerant"),]

# Test for difference
u.obs<-wilcox.test(Distance ~ Project1, data=an)
u.obs # W = 3143128, p-value = 2.655e-07
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"Distance"
for (i in 1:Nperm) {
  d<-an[,!names(an) %in% lose]
  d$dist.rand<-sample(an$Distance, length(an$Distance), replace=F)
  x<-wilcox.test(dist.rand ~ Project1, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat>u.obs$statistic, 1, 0)
p<-mean(perm1$result)
pval<-2*p
pval # 0

mean_A<-mean(an$Distance[an$Project1=="Island"]) # 0.9059054
mean_B<-mean(an$Distance[an$Project1=="Mainland"]) # 0.8933874
diff<-mean_A-mean_B # 0.01251802

# Test for difference
u.obs<-wilcox.test(Distance ~ Project1, data=a)
u.obs # W = 3728550, p-value < 2.2e-16
u.obs.stat <- u.obs$statistic

#do permutation test
Nperm<-1000
perm1<-data.frame(permutation=1:Nperm, u.stat=NA)
head(perm1) 

lose<-"Distance"
for (i in 1:Nperm) {
  d<-a[,!names(a) %in% lose]
  d$dist.rand<-sample(a$Distance, length(a$Distance), replace=F)
  x<-wilcox.test(dist.rand ~ Project1, data=d)
  perm1$u.stat[i]<-x$statistic
}

head(perm1)
hist(perm1$u.stat)
abline(v=u.obs$statistic, col="red", lwd=2, lty=2)

perm1$result<-ifelse(perm1$u.stat>u.obs$statistic, 1, 0)
p<-mean(perm1$result)
pval<-2*p
pval # 0

mean_A<-mean(a$Distance[a$Project1=="Island"]) # 0.8827124
mean_B<-mean(a$Distance[a$Project1=="Mainland"]) # 0.8341616
diff<-mean_A-mean_B # 0.04855081

```


